# The main entry point of your workflow.
# After configuring, running snakemake -n in a clone of this repository should successfully execute a dry-run of the workflow.


configfile: "config/config.yaml"
include: "rules/common.smk"
include: "rules/BptMotifSeach.smk"
include: "rules/ChemCLIP.smk"
include: "rules/RE_Alignment.smk"
include: "rules/RE_CallEditingSites.smk"
include: "rules/RE_FilterEditingSites.smk"
include: "rules/pangolin.smk"
include: "rules/RNASeqProcessing.smk"
include: "rules/ProcessTitrationSeries.smk"
include: "rules/Process_chRNA.smk"



wildcard_constraints:
    chrom = "|".join(autosomes),
    # sample = "[A-Z][A-Z0-9]*-[1-9]"

localrules:
    all,
    pangolin,
    SplitClinVarVcfByChrom,
    DE_testing,
    GatherEditings,
    CopyAndMergeFastq_chRNA

# Instead of using samples.tsv file, a quick/dirty way to apply rules over grob of existing files is this:
# IDS, = glob_wildcards("Path/To/{MyFileNames}.bed.gz")

rule all:
    input:
        # expand("Alignments/STAR_Align/{sample}/Aligned.sortedByCoord.out.bam.bai", sample= samples.index),
        # "SplicingAnalysis/leafcutter/clustering/autosomes/leafcutter_perind.counts.gz",
        "DoseResponseData/LCL/TidySplicingDoseData.txt.gz",
        expand("SplicingAnalysis/leafcutter/differential_splicing/{treatment}_effect_sizes.txt", treatment = NonControlTreatments),
        "featureCounts/Counts.txt",
        "SplicingAnalysis/leafcutter/JuncfilesMerged.annotated.basic.bed.gz",
        "ClinVar/PangolinResults.tsv.gz",
        "DE_testing/Results.txt.gz",
        # expand("SplicingAnalysis/juncbedfiles/{sample}.bed.gz", sample=samples.index)
        "SplicingAnalysis/leafcutter/clustering_permissive/autosomes/leafcutter_perind.counts.gz",
        "SplicingAnalysis/leafcutter/JuncfilesMerged.annotated.basic.bed.5ss.tab.gz",
        "SplicingAnalysis/FullSpliceSiteAnnotations/JuncfilesMerged.annotated.basic.bed.5ss.tab.gz",
        "../output/ClinVar.PangolinResults.tsv.gz",
        rules.GatherChRNAResults.input

        # expand("Alignments/STAR_Align/{sample}/Aligned.sortedByCoord.out.bam",sample=AllSamples),
        # 'RE_Results/EditLevel/GatheredEditingLevels.bed'

        # expand("ClinVar/ByChrom/{chrom}.vcf", chrom=autosomes),
        # expand("ClinVar/ByChrom/{chrom}.pangolin.vcf", chrom=autosomes)
        # The first rule should define the default target files
        # Subsequent target rules can be specified below. They should start with all_*.

