<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>20220922_Explore_SubparRNAseq</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">small molecular splicing RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">20220922_Explore_SubparRNAseq</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#intro">Intro</a></li>
<li><a href="#analysis">Analysis</a></li>
<li><a href="#conclusion">Conclusion</a></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>20211209_JingxinRNAseq/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed19900924code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(19900924)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed19900924code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(19900924)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/yangili1/bjf79/20211209_JingxinRNAseq/code/bigwigs/unstranded/(.+?).bw
</td>
<td style="text-align:left;">
../code/bigwigs/unstranded/(.+?).bw
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunSmallMoleculeRNASeqtree9a23a3ece39b84f39a7d5946b11d334a61e7276etargetblank9a23a3ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/bfairkun/SmallMoleculeRNASeq/tree/9a23a3ece39b84f39a7d5946b11d334a61e7276e" target="_blank">9a23a3e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunSmallMoleculeRNASeqtree9a23a3ece39b84f39a7d5946b11d334a61e7276etargetblank9a23a3ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/bfairkun/SmallMoleculeRNASeq/tree/9a23a3ece39b84f39a7d5946b11d334a61e7276e" target="_blank">9a23a3e</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/20220707_TitrationSeries_DE_testing.nb.html
    Ignored:    code/%
    Ignored:    code/.DS_Store
    Ignored:    code/._.DS_Store
    Ignored:    code/._DOCK7.pdf
    Ignored:    code/._DOCK7_DMSO1.pdf
    Ignored:    code/._DOCK7_SM2_1.pdf
    Ignored:    code/._FKTN_DMSO_1.pdf
    Ignored:    code/._FKTN_SM2_1.pdf
    Ignored:    code/._MAPT.pdf
    Ignored:    code/._PKD1_DMSO_1.pdf
    Ignored:    code/._PKD1_SM2_1.pdf
    Ignored:    code/.snakemake/
    Ignored:    code/1KG_HighCoverageCalls.samplelist.txt
    Ignored:    code/5ssSeqs.tab
    Ignored:    code/Alignments/
    Ignored:    code/ChemCLIP/
    Ignored:    code/ClinVar/
    Ignored:    code/DE_testing/
    Ignored:    code/DE_tests.mat.counts.gz
    Ignored:    code/DE_tests.txt.gz
    Ignored:    code/DoseResponseData/
    Ignored:    code/Fastq/
    Ignored:    code/FastqFastp/
    Ignored:    code/FragLenths/
    Ignored:    code/Meme/
    Ignored:    code/Multiqc/
    Ignored:    code/OMIM/
    Ignored:    code/OldBigWigs/
    Ignored:    code/PhyloP/
    Ignored:    code/QC/
    Ignored:    code/ReferenceGenomes/
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TracksSession
    Ignored:    code/bigwigs/
    Ignored:    code/featureCounts/
    Ignored:    code/geena/
    Ignored:    code/hg38ToMm39.over.chain.gz
    Ignored:    code/igv_session.template.xml
    Ignored:    code/igv_session.xml
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/scratch/
    Ignored:    code/test.txt.gz
    Ignored:    code/testPlottingWithMyScript.ForJingxin.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin2.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin3.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin4.sh
    Ignored:    code/testPlottingWithMyScript.sh
    Ignored:    data/._Hijikata_TableS1_41598_2017_8902_MOESM2_ESM.xls
    Ignored:    data/._Hijikata_TableS2_41598_2017_8902_MOESM3_ESM.xls
    Ignored:    output/._PioritizedIntronTargets.pdf

Unstaged changes:
    Modified:   analysis/20220922_ExploreSubparRNAseq.Rmd
    Modified:   analysis/20220923_ExploreSpecificityEstimates.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/20220922_ExploreSubparRNAseq.Rmd</code>) and HTML (<code>docs/20220922_ExploreSubparRNAseq.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/9a23a3ece39b84f39a7d5946b11d334a61e7276e/analysis/20220922_ExploreSubparRNAseq.Rmd" target="_blank">9a23a3e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-09-26
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/SmallMoleculeRNASeq/9a23a3ece39b84f39a7d5946b11d334a61e7276e/docs/20220922_ExploreSubparRNAseq.html" target="_blank">9a23a3e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-09-26
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/272cd7602a7f90fe4cf35c8919e39787aa3c8d38/analysis/20220922_ExploreSubparRNAseq.Rmd" target="_blank">272cd76</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-09-26
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/SmallMoleculeRNASeq/272cd7602a7f90fe4cf35c8919e39787aa3c8d38/docs/20220922_ExploreSubparRNAseq.html" target="_blank">272cd76</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-09-26
</td>
<td>
update site
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>Jingxin’s lab has made many compounds (ie &gt;50) from one or two scaffolds, and is interseted in using RNA-seq to readout splicing effects. Obviously at this scale, with limited resources, and with low prior knowledge about their splice-modulating activity, it doesn’t make sense to do a full titration experiment with RNA-seq for each compound. We are more likely to settle with a single dose, at a smaller read depth, and try to learn what we can about the activity and specificity of these compounds. Here I will explore our existing data just to gain some intuitions on the power of RNA-seq, keeping in mind what it would be like if we only had one replicate at lower read coverage. Eventually I may do a more careful power analysis, involving sub-sampling reads from our existing data, and re-analyzing them to simulate potential study designs.</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<p>First let’s read in library size in terms of mapped reads.</p>
<pre class="r"><code>library(tidyverse)
library(gplots)
library(RColorBrewer)

counts &lt;- read_tsv(&quot;../output/QC/ReadCountsAndJunctionsPerSamples.tsv&quot;, col_names = c(&quot;fn&quot;, &quot;ChromosomalReads&quot;))

sample.list &lt;- read_tsv(&quot;../code/bigwigs/BigwigList.tsv&quot;,
                        col_names = c(&quot;SampleName&quot;, &quot;bigwig&quot;, &quot;group&quot;, &quot;strand&quot;)) %&gt;%
  filter(strand==&quot;.&quot;) %&gt;%
  dplyr::select(-strand) %&gt;%
  mutate(old.sample.name = str_replace(bigwig, &quot;/project2/yangili1/bjf79/20211209_JingxinRNAseq/code/bigwigs/unstranded/(.+?).bw&quot;, &quot;\\1&quot;)) %&gt;%
  separate(SampleName, into=c(&quot;treatment&quot;, &quot;dose.nM&quot;, &quot;cell.type&quot;, &quot;libType&quot;, &quot;rep&quot;), convert=T, remove=F, sep=&quot;_&quot;) %&gt;%
  left_join(
    read_tsv(&quot;../code/bigwigs/BigwigList.groups.tsv&quot;, col_names = c(&quot;group&quot;, &quot;color&quot;, &quot;bed&quot;, &quot;supergroup&quot;)),
    by=&quot;group&quot;
  )

TreatmentColorsForLabels &lt;- sample.list %&gt;%
  group_by(treatment) %&gt;%
  filter(dose.nM == max(dose.nM) | treatment == &quot;DMSO&quot;) %&gt;%
  ungroup() %&gt;%
  distinct(treatment, .keep_all=T) %&gt;%
  arrange(dose.nM) %&gt;%
  mutate(vjust=row_number()*1.2)

TreatmentColorsForLabelsKey &lt;- TreatmentColorsForLabels %&gt;%
  filter(!treatment==&quot;DMSO&quot;) %&gt;%
  dplyr::select(treatment, color) %&gt;% deframe()

TreatmentColorsLabels.Layer &lt;- geom_text(
    data = TreatmentColorsForLabels, 
    aes(label=treatment, color=color, vjust=vjust),
    y=Inf, x=Inf, hjust=1.05
  )

counts.plot.dat &lt;- counts %&gt;%
  mutate(old.sample.name = str_replace(fn, &quot;Alignments/STAR_Align/(.+?)/Aligned.sortedByCoord.out.bam&quot;, &quot;\\1&quot;)) %&gt;%
  inner_join(sample.list) %&gt;%
  arrange(cell.type, libType, desc(treatment==&quot;DMSO&quot;), treatment, dose.nM, rep) %&gt;%
  mutate(SampleName = factor(SampleName, levels=SampleName)) %&gt;%
  mutate(Experiment = case_when(
    libType == &quot;chRNA&quot; ~ &quot;nascent RNA profiling&quot;,
    cell.type == &quot;LCL&quot; &amp; libType == &quot;polyA&quot; ~ &quot;dose response experiment&quot;,
    cell.type == &quot;Fibroblast&quot; ~ &quot;Initial experiment in fibroblast&quot;
  )) %&gt;%
  mutate(label = paste0(treatment,&quot;; &quot;, dose.nM))

counts.plot.labels &lt;- counts.plot.dat %&gt;%
  dplyr::select(SampleName, label) %&gt;% deframe()

ggplot(counts.plot.dat, aes(x=SampleName, y=ChromosomalReads/2E6, fill=color)) +
  geom_col() +
  scale_fill_identity() +
  scale_x_discrete(name=&quot;Sample; treatment_nanomolar-dose&quot;, label=counts.plot.labels) +
  facet_grid(cols = vars(Experiment), scales=&quot;free&quot;, space = &quot;free_x&quot;) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size=5)) +
  theme(strip.text.x = element_text(size = 8)) +
  labs(title=&quot;RNA-seq datasets&quot;, y=&quot;Read count (M)&quot;)</code></pre>
<p><img src="figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/272cd7602a7f90fe4cf35c8919e39787aa3c8d38/docs/figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-1-1.png" target="_blank">272cd76</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Now let’s read in the table of a few hundred GA-GT splice sites that I previously shared with Jingxin, along with some other splice count tables… I want to see how rare these GA-GT splice sites are in our data, to get a sense how many reads we need to detect them.</p>
<pre class="r"><code>GA.GT.Introns &lt;- read_tsv(&quot;../output/EC50Estimtes.FromPSI.txt.gz&quot;)


all.samples.PSI &lt;- read_tsv(&quot;../code/SplicingAnalysis/leafcutter_all_samples/PSI.table.bed.gz&quot;)
all.samples.junccounts &lt;- read_tsv(&quot;../code/SplicingAnalysis/leafcutter_all_samples/JuncCounts.table.bed.gz&quot;)

all.samples.5ss &lt;- read_tsv(&quot;../code/SplicingAnalysis/FullSpliceSiteAnnotations/JuncfilesMerged.annotated.basic.bed.5ss.tab.gz&quot;, col_names = c(&quot;intron&quot;, &quot;seq&quot;, &quot;score&quot;)) %&gt;%
  mutate(intron = str_replace(intron, &quot;^(.+?)::.+$&quot;, &quot;\\1&quot;)) %&gt;%
  separate(intron, into=c(&quot;chrom&quot;, &quot;start&quot;, &quot;stop&quot;, &quot;strand&quot;), sep=&quot;_&quot;, convert=T, remove=F)

all.samples.intron.annotations &lt;- read_tsv(&quot;../code/SplicingAnalysis/FullSpliceSiteAnnotations/JuncfilesMerged.annotated.basic.bed.gz&quot;)

All.GA.GT &lt;- all.samples.5ss %&gt;%
  filter(str_detect(seq, &quot;^\\w{2}GAGT&quot;)) %&gt;%
  inner_join(all.samples.intron.annotations %&gt;%
               rename(stop=end) %&gt;%
               dplyr::select(-score)) %&gt;%
  left_join(
    GA.GT.Introns %&gt;%
      dplyr::select(chrom=`#Chrom`, start, stop=end, strand=strand.y, LowerLimit, UpperLimit, UpstreamSpliceAcceptor, spearman.coef.Branaplam, spearman.coef.C2C5,spearman.coef.Risdiplam)
  ) %&gt;%
  mutate(GAGT.set = case_when(
    !is.na(LowerLimit) ~ &quot;Juncs modelled in LCL&quot;,
    !is.na(spearman.coef.Branaplam) ~ &quot;Juncs counted in LCL&quot;,
    TRUE ~ &quot;Other juncs observed across datasets&quot;
  )) %&gt;%
  inner_join(
    all.samples.junccounts %&gt;%
      dplyr::select(junc, contains(&quot;LCL&quot;)) %&gt;%
      dplyr::select(junc, contains(&quot;polyA&quot;)) %&gt;%
      mutate(intron = str_replace(junc, &quot;^(chr.+?):(.+?):(.+?):clu.+?([+-])$&quot;, &quot;\\1_\\2_\\3_\\4&quot;))
  )

GA.GT.CPM.Plot.dat &lt;- All.GA.GT %&gt;%
  dplyr::select(intron, GAGT.set, contains(&quot;LCL&quot;)) %&gt;%
  add_count(GAGT.set) %&gt;%
  gather(&quot;SampleName&quot;, &quot;JuncCounts&quot;, contains(&quot;LCL&quot;)) %&gt;%
  group_by(SampleName, n, GAGT.set) %&gt;%
  summarise(sumJuncCounts = sum(JuncCounts)) %&gt;%
  inner_join(counts.plot.dat) %&gt;%
  mutate(CPM=sumJuncCounts/(ChromosomalReads/1E6)) %&gt;%
  mutate(FacetLabel = paste0(GAGT.set, &quot;;n=&quot;,n)) %&gt;%
  arrange(cell.type, libType, desc(treatment==&quot;DMSO&quot;), treatment, dose.nM, rep, GAGT.set)

GA.GT.CPM.Plot.dat$SampleName &lt;- factor(GA.GT.CPM.Plot.dat$SampleName, levels=unique(GA.GT.CPM.Plot.dat$SampleName))

ggplot(GA.GT.CPM.Plot.dat, aes(x=SampleName, y=CPM, fill=color, color=color)) +
  geom_col() +
  scale_fill_identity() +
  scale_color_identity() +
  scale_x_discrete(name=&quot;Sample; treatment_nanomolar-dose&quot;, label=counts.plot.labels) +
  facet_wrap(~FacetLabel, scales=&quot;free_y&quot;) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size=5)) +
  theme(strip.text.x = element_text(size = 6)) +
  labs(title=&quot;DetectionOf GA|GT introns in aggregate&quot;, y=&quot;CountsPerMillion&quot;)</code></pre>
<p><img src="figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/272cd7602a7f90fe4cf35c8919e39787aa3c8d38/docs/figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-2-1.png" target="_blank">272cd76</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, aggregating the set of 600 or so junctions that I modelled I see clear dose-dependent differences between DMSO and treatments, with CountsPerMillions (expression of junction reads) in the hundreds… Even with a single sample at a fraction of the read depth, i think this would be easily detectable.</p>
<p>Perhaps another way to consider what it would be like to analyze data with just one replicate, is to look at delta-PSI, based on either 1 sample, or the average of 3 samples, from the fibroblast data…</p>
<pre class="r"><code>cluster_sig_files &lt;- list.files(&quot;../code/SplicingAnalysis/leafcutter/differential_splicing&quot;, &quot;*_cluster_significance.txt&quot;, full.names = T)
effect_sizes_files &lt;- list.files(&quot;../code/SplicingAnalysis/leafcutter/differential_splicing&quot;, &quot;*_effect_sizes.txt&quot;, full.names = T)
treatments &lt;- str_replace(cluster_sig_files, &quot;.+/(.+?)_cluster_significance.txt&quot;, &quot;\\1&quot;)

cluster.sig &lt;- map(cluster_sig_files, read_tsv) %&gt;%
  set_names(cluster_sig_files) %&gt;%
  bind_rows(.id=&quot;f&quot;) %&gt;%
  mutate(treatment = str_replace(f, &quot;.+/(.+?)_cluster_significance.txt&quot;, &quot;\\1&quot;)) %&gt;%
  select(-f)

effect_sizes &lt;- map(effect_sizes_files, read_tsv) %&gt;%
  set_names(effect_sizes_files) %&gt;%
  bind_rows(.id=&quot;f&quot;) %&gt;%
  mutate(treatment = str_replace(f, &quot;.+/(.+?)_effect_sizes.txt&quot;, &quot;\\1&quot;)) %&gt;%
  select(-f) %&gt;%
  unite(&quot;psi_treatment&quot;, treatments, sep=&quot; &quot;, na.rm=T) %&gt;%
  mutate(psi_treatment = as.numeric(psi_treatment),
         cluster = str_replace(intron, &quot;(.+?:).+:(.+?)&quot;, &quot;\\1\\2&quot;))


#Based on all introns
all.samples.PSI %&gt;%
  dplyr::select(1:6, contains(&quot;Fibroblast&quot;)) %&gt;%
  dplyr::select(-c(1:6)) %&gt;%
  drop_na() %&gt;%
  as.matrix() %&gt;%
  cor() %&gt;%
  heatmap.2(trace=&#39;none&#39;)</code></pre>
<p><img src="figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/272cd7602a7f90fe4cf35c8919e39787aa3c8d38/docs/figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-3-1.png" target="_blank">272cd76</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#Based on all GA|GT introns
all.samples.PSI %&gt;%
  dplyr::select(1:6, contains(&quot;Fibroblast&quot;)) %&gt;%
  inner_join(dplyr::select(All.GA.GT, `#Chrom`=chrom, start, end=stop, strand)) %&gt;%
  dplyr::select(-c(1:6)) %&gt;%
  drop_na() %&gt;%
  as.matrix() %&gt;%
  cor() %&gt;%
  heatmap.2(trace=&#39;none&#39;)</code></pre>
<p><img src="figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/272cd7602a7f90fe4cf35c8919e39787aa3c8d38/docs/figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-3-2.png" target="_blank">272cd76</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Lastly, I think way to plot it that Yang is interested in is to average the 3 replicates for treatment, calculate delta-psi (from DMSO averaged across three replicates), and compare to the delta-psi from just a single replicate. Let’s include one GA-GT introns for this…</p>
<pre class="r"><code>DMSO.mean.PSI &lt;- all.samples.PSI %&gt;%
  dplyr::select(1:6, contains(&quot;Fibroblast&quot;)) %&gt;%
  inner_join(dplyr::select(All.GA.GT, `#Chrom`=chrom, start, end=stop, strand)) %&gt;%
  drop_na() %&gt;%
  dplyr::select(junc, contains(&quot;Fibroblast&quot;)) %&gt;%
  gather(&quot;SampleName&quot;, &quot;PSI&quot;, contains(&quot;Fibroblast&quot;)) %&gt;%
  inner_join(sample.list) %&gt;%
  filter(treatment == &quot;DMSO&quot;) %&gt;%
  group_by(junc) %&gt;%
  summarise(meanPSI.DMSO = mean(PSI)) %&gt;%
  ungroup()

treament.mean.PSI &lt;- all.samples.PSI %&gt;%
  dplyr::select(1:6, contains(&quot;Fibroblast&quot;)) %&gt;%
  inner_join(dplyr::select(All.GA.GT, `#Chrom`=chrom, start, end=stop, strand)) %&gt;%
  drop_na() %&gt;%
  dplyr::select(junc, contains(&quot;Fibroblast&quot;)) %&gt;%
  gather(&quot;SampleName&quot;, &quot;PSI&quot;, contains(&quot;Fibroblast&quot;)) %&gt;%
  inner_join(sample.list) %&gt;%
  filter(!treatment == &quot;DMSO&quot;) %&gt;%
  group_by(treatment, junc) %&gt;%
  summarise(PSI = mean(PSI)) %&gt;%
  ungroup() %&gt;%
  mutate(SampleName = paste(treatment, &quot;mean&quot;, sep=&quot;_&quot;)) %&gt;%
  mutate(PSI.Calculation = &quot;Average across 3 treatment replicates&quot;)
  
Compare3vs1Replicate.Dat.To.Plot &lt;- all.samples.PSI %&gt;%
  dplyr::select(1:6, contains(&quot;Fibroblast&quot;)) %&gt;%
  inner_join(dplyr::select(All.GA.GT, `#Chrom`=chrom, start, end=stop, strand)) %&gt;%
  drop_na() %&gt;%
  dplyr::select(junc, contains(&quot;Fibroblast&quot;)) %&gt;%
  gather(&quot;SampleName&quot;, &quot;PSI&quot;, contains(&quot;Fibroblast&quot;)) %&gt;%
  inner_join(sample.list) %&gt;%
  filter(!treatment == &quot;DMSO&quot;) %&gt;%
  dplyr::select(treatment, SampleName, junc, PSI) %&gt;%
  mutate(PSI.Calculation = &quot;Single treatment replicate&quot;) %&gt;%
  bind_rows(treament.mean.PSI) %&gt;%
  inner_join(DMSO.mean.PSI) %&gt;%
  mutate(DeltaPSI = PSI-meanPSI.DMSO)


ggplot(Compare3vs1Replicate.Dat.To.Plot, aes(x=DeltaPSI, color=treatment)) +
  stat_ecdf(aes(linetype=PSI.Calculation)) +
  facet_wrap(~treatment) +
  theme_bw() +
  coord_cartesian(xlim=c(-10, 10)) +
  labs(y=&quot;ecdf&quot;, title=&#39;effects of GA|GT introns&#39;)</code></pre>
<p><img src="figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/9a23a3ece39b84f39a7d5946b11d334a61e7276e/docs/figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-4-1.png" target="_blank">9a23a3e</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-09-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/272cd7602a7f90fe4cf35c8919e39787aa3c8d38/docs/figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-4-1.png" target="_blank">272cd76</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2022-09-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, so the mean genomewide distribution (plotted as ecdf) of the delta-PSI of GA.GT introns looks visually indistinguishable whether you average 3 replicates or if you just use one replicate.</p>
<p>Let’s now plot these DeltaPSIs again (comparing deltaPSI calculated from one replicate, versus average from three replicates) as a correlation matrix, with clustering the rows and columns…</p>
<pre class="r"><code>Compare3vs1Replicate.Dat.To.Plot.As.matrix &lt;- Compare3vs1Replicate.Dat.To.Plot %&gt;%
  dplyr::select(SampleName, junc, DeltaPSI) %&gt;%
  pivot_wider(names_from = &quot;SampleName&quot;, values_from=&quot;DeltaPSI&quot;) %&gt;%
  column_to_rownames(&quot;junc&quot;)

ConvertToColorVector &lt;- function(Vector, PalletteString=&quot;Set1&quot;){
  ConversionKey &lt;- setNames(brewer.pal(length(unique(Vector)), PalletteString), unique(Vector))[1:length(unique(Vector))]
  ColorVector &lt;- recode(Vector, !!!ConversionKey)
  return( list(Key=ConversionKey, ColorVector=ColorVector))
}

Rowcols &lt;- colnames(Compare3vs1Replicate.Dat.To.Plot.As.matrix) %&gt;%
  str_replace(&quot;^(.+?)_.+&quot;, &quot;\\1&quot;) %&gt;% ConvertToColorVector()
Colcols &lt;- colnames(Compare3vs1Replicate.Dat.To.Plot.As.matrix) %&gt;%
  str_detect(&quot;mean&quot;) %&gt;% as.character() %&gt;% ConvertToColorVector(PalletteString=&quot;Set2&quot;)

# Row colors are treatments
# Column colors are 1 or 3 replicates
cor(Compare3vs1Replicate.Dat.To.Plot.As.matrix) %&gt;%
  heatmap.2(trace=&#39;none&#39;, ColSideColors = Colcols$ColorVector, RowSideColors = Rowcols$ColorVector)</code></pre>
<p><img src="figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Let’s plot deltaPSI from 1 vs 3 averaged replicates for some individual introns. A random sample of GA|GT introns. I always find looking at a random sample of introns to be helpful, to gain some intuition.</p>
<pre class="r"><code>sample_n_of &lt;- function(data, size, ...) {
  dots &lt;- quos(...)
  
  group_ids &lt;- data %&gt;% 
    group_by(!!! dots) %&gt;% 
    group_indices()
  
  sampled_groups &lt;- sample(unique(group_ids), size)
  
  data %&gt;% 
    filter(group_ids %in% sampled_groups)
}

set.seed(0)
Compare3vs1Replicate.Dat.To.Plot %&gt;%
  sample_n_of(20, junc) %&gt;%
  ggplot(aes(y=DeltaPSI, x=SampleName, color=PSI.Calculation, fill=treatment)) +
  geom_col() +
  facet_wrap(~junc,scales=&quot;free_y&quot;) +
  theme_bw() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  labs(title=&quot;20 random GA|GT introns&quot;, y=&quot;DeltaPSI (versus DMSO from 3 replicates)&quot;, color=&quot;Treatment PSI from&quot;)</code></pre>
<p><img src="figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Hmm.. so as expected, individual intron measurements can be quite noisy, and I still haven’t figured out how or if I will want to do any differential splicing tests at the individual intron level.</p>
<p>Let’s go back to thinking about characterizing samples on a global level… I wonder if it will be useful to plot new samples in a previously determined priniciple component space that is more interpretable. For example, let’s go back to the dose response experiment, calculate principal components, and overlay the fibroblast samples in the same principle component axes. Keep in mind the experiments were done in different cell types, so I’m not sure what to expect here, but let’s just see…</p>
<pre class="r"><code>#First compute principle components from dose response data
pca.results &lt;- all.samples.PSI %&gt;%
  inner_join(
    dplyr::select(All.GA.GT, `#Chrom`=chrom, start, end=stop, strand)
    ) %&gt;%
  drop_na() %&gt;%
  dplyr::select(1:6, contains(&quot;LCL&quot;)) %&gt;%
  dplyr::select(1:6, contains(&quot;polyA&quot;)) %&gt;%
  dplyr::select(junc, contains(&quot;LCL&quot;)) %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% prcomp(scale=T)

summary(pca.results)</code></pre>
<pre><code>Importance of components:
                           PC1    PC2     PC3     PC4     PC5     PC6     PC7
Standard deviation     17.1445 8.3535 6.58435 4.88941 4.21117 3.98381 3.44426
Proportion of Variance  0.5148 0.1222 0.07593 0.04187 0.03106 0.02779 0.02078
Cumulative Proportion   0.5148 0.6370 0.71290 0.75477 0.78583 0.81362 0.83440
                           PC8     PC9    PC10    PC11   PC12    PC13    PC14
Standard deviation     3.29972 3.02333 2.72737 2.68984 2.4602 2.40520 2.34934
Proportion of Variance 0.01907 0.01601 0.01303 0.01267 0.0106 0.01013 0.00967
Cumulative Proportion  0.85347 0.86948 0.88250 0.89518 0.9058 0.91591 0.92557
                          PC15    PC16   PC17    PC18    PC19    PC20    PC21
Standard deviation     2.22745 2.15526 2.0963 2.07448 1.98977 1.93936 1.85475
Proportion of Variance 0.00869 0.00814 0.0077 0.00754 0.00693 0.00659 0.00602
Cumulative Proportion  0.93426 0.94240 0.9501 0.95763 0.96456 0.97115 0.97717
                          PC22    PC23    PC24    PC25    PC26      PC27
Standard deviation     1.75061 1.67522 1.62255 1.52060 1.48911 1.928e-14
Proportion of Variance 0.00537 0.00491 0.00461 0.00405 0.00388 0.000e+00
Cumulative Proportion  0.98254 0.98746 0.99207 0.99612 1.00000 1.000e+00</code></pre>
<pre class="r"><code># Then plot original samples
pca.results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;SampleName&quot;) %&gt;%
  dplyr::select(SampleName, PC1, PC2, PC3) %&gt;%
  left_join(sample.list, by=&quot;SampleName&quot;) %&gt;%
  ggplot(aes(x=PC1, y=PC2, color=color)) +
  TreatmentColorsLabels.Layer +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using all GA|GT introns&quot;, x=&quot;PC1 (52% variance explained)&quot;, y=&quot;PC2 (15% variance explained)&quot;)</code></pre>
<p><img src="figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now let’s try plotting the fibroblast data in the old PC space. From visually looking at a few loci in the genome browser, I already know that SM2 is a lot like branaplam, so I’ll look for that effect…</p>
<pre class="r"><code>FirboblastDat &lt;- all.samples.PSI %&gt;%
  inner_join(
    dplyr::select(All.GA.GT, `#Chrom`=chrom, start, end=stop, strand)
    ) %&gt;%
  drop_na() %&gt;%
  dplyr::select(junc, contains(&quot;Fibroblast&quot;)) %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  scale() %&gt;% t()

Fibroblastdat.projected &lt;- predict(pca.results, FirboblastDat)

TreatmentColorsForLabels.FibroblastColorsAdded &lt;-
sample.list %&gt;%
  group_by(treatment) %&gt;%
  filter(dose.nM == max(dose.nM) | treatment == &quot;DMSO&quot; | cell.type==&quot;Fibroblast&quot;) %&gt;%
  ungroup() %&gt;%
  distinct(treatment, .keep_all=T) %&gt;%
  arrange(dose.nM) %&gt;%
  mutate(vjust=row_number()*1.2)

TreatmentColorsLabels.Layer.FibroblastColorsAdded &lt;- geom_text(
    data = TreatmentColorsForLabels.FibroblastColorsAdded, 
    aes(label=treatment, color=color, vjust=vjust),
    y=Inf, x=Inf, hjust=1.05
  )

bind_rows(
    Fibroblastdat.projected %&gt;% as.data.frame(),
    pca.results$x %&gt;% as.data.frame()
  ) %&gt;%
  rownames_to_column(&quot;SampleName&quot;) %&gt;%
  dplyr::select(SampleName, PC1, PC2, PC3) %&gt;%
  left_join(sample.list, by=&quot;SampleName&quot;) %&gt;%
  ggplot(aes(x=PC1, y=PC2, color=color, shape=cell.type)) +
  geom_point(size=3) +
  scale_color_identity() +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  theme_bw() +
  labs(title = &quot;PCA using all GA|GT introns&quot;, x=&quot;PC1 (52% variance explained)&quot;, y=&quot;PC2 (15% variance explained)&quot;)</code></pre>
<p><img src="figure/20220922_ExploreSubparRNAseq.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, I rather like looking at the data this way, and each point is a single sample, and each point is placed nicely in a place I expect and can somewhat interpret in these two dimensions.</p>
</div>
<div id="conclusion" class="section level2">
<h2>Conclusion</h2>
<p>though I haven’t worked out the details of how I would want to perform an analysis with one RNA-seq replicate, I’m confident there is enough genome-wide information to at the very least discriminate “active” from inactive compounds from 1 replicate…</p>
<p>We might not need to do a formal statistical test (ie leafcutter) for each splice event to get a decent profile of each treatment. I could probably consider doing simpler things like looking at meta-intron features (the sum of junction counts many GA|GT introns), or if I wanted to get a broad sense of specificity, plotting things as I have done in the same PC space as previously calculated will at least distinguish the branaplam-vs-risdiplam dimension. Though, a better understanding of specificity (particularly for new dimensions that might not easily be described in the risdiplam-branaplam axis) may require more unbiased analysis which will no doubt benefit from first doing splice-event-level statistical tests.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RColorBrewer_1.1-2 gplots_3.0.1.1     forcats_0.4.0      stringr_1.4.0     
 [5] dplyr_1.0.9        purrr_0.3.4        readr_1.3.1        tidyr_1.2.0       
 [9] tibble_3.1.7       ggplot2_3.3.6      tidyverse_1.3.0   

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5         lubridate_1.7.4    gtools_3.9.2.2     assertthat_0.2.1  
 [5] rprojroot_2.0.2    digest_0.6.20      utf8_1.1.4         R6_2.4.0          
 [9] cellranger_1.1.0   backports_1.4.1    reprex_0.3.0       evaluate_0.15     
[13] httr_1.4.4         highr_0.9          pillar_1.7.0       rlang_1.0.5       
[17] readxl_1.3.1       gdata_2.18.0       rstudioapi_0.14    whisker_0.3-2     
[21] rmarkdown_1.13     labeling_0.3       munsell_0.5.0      broom_1.0.0       
[25] compiler_3.6.1     httpuv_1.5.1       modelr_0.1.8       xfun_0.31         
[29] pkgconfig_2.0.2    htmltools_0.5.3    tidyselect_1.1.2   workflowr_1.6.2   
[33] fansi_0.4.0        crayon_1.3.4       dbplyr_1.4.2       withr_2.5.0       
[37] later_0.8.0        bitops_1.0-6       grid_3.6.1         jsonlite_1.6      
[41] gtable_0.3.0       lifecycle_1.0.1    DBI_1.1.0          git2r_0.26.1      
[45] magrittr_1.5       scales_1.1.0       KernSmooth_2.23-15 cli_3.3.0         
[49] stringi_1.4.3      farver_2.1.0       fs_1.5.2           promises_1.0.1    
[53] xml2_1.3.2         ellipsis_0.3.2     generics_0.1.3     vctrs_0.4.1       
[57] tools_3.6.1        glue_1.6.2         hms_0.5.3          fastmap_1.1.0     
[61] yaml_2.2.0         colorspace_1.4-1   caTools_1.17.1.2   rvest_0.3.5       
[65] knitr_1.39         haven_2.3.1       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
