<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-02-21" />

<title>browse exp of 52</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">small molecular splicing RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">browse exp of 52</h1>
<h4 class="date">2023-02-21</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#intro" id="toc-intro">Intro</a>
<ul>
<li><a href="#reads-per-sample" id="toc-reads-per-sample">Reads per
sample</a></li>
<li><a href="#gene-expression-pca" id="toc-gene-expression-pca">Gene
expression PCA</a></li>
<li><a href="#verify-genome-wide-ga-gt-splice-site-induction"
id="toc-verify-genome-wide-ga-gt-splice-site-induction">Verify
genome-wide GA-GT splice site induction</a></li>
<li><a href="#splicing-correlation-matrix-and-pca"
id="toc-splicing-correlation-matrix-and-pca">Splicing correlation matrix
and PCA</a></li>
<li><a href="#leafcutter-differential-splicing"
id="toc-leafcutter-differential-splicing">Leafcutter differential
splicing</a></li>
</ul></li>
<li><a href="#investigation-of-the-c05-d08-and-b12"
id="toc-investigation-of-the-c05-d08-and-b12">Investigation of the C05,
D08, and B12</a>
<ul>
<li><a href="#pca-on-subset-of-scaffolds"
id="toc-pca-on-subset-of-scaffolds">PCA on subset of scaffolds</a></li>
</ul></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong>
<code>20211209_JingxinRNAseq/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed19900924code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(19900924)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed19900924code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(19900924)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/yangili1/bjf79/20211209_JingxinRNAseq/code/bigwigs/unstranded/(.+?).bw
</td>
<td style="text-align:left;">
../code/bigwigs/unstranded/(.+?).bw
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunSmallMoleculeRNASeqtreee996344ad0d4841a485c9f6f500e259daacdc96btargetblanke996344a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/tree/e996344ad0d4841a485c9f6f500e259daacdc96b" target="_blank">e996344</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunSmallMoleculeRNASeqtreee996344ad0d4841a485c9f6f500e259daacdc96btargetblanke996344a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/tree/e996344ad0d4841a485c9f6f500e259daacdc96b" target="_blank">e996344</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/20220707_TitrationSeries_DE_testing.nb.html
    Ignored:    code/.DS_Store
    Ignored:    code/._DOCK7.pdf
    Ignored:    code/._DOCK7_DMSO1.pdf
    Ignored:    code/._DOCK7_SM2_1.pdf
    Ignored:    code/._FKTN_DMSO_1.pdf
    Ignored:    code/._FKTN_SM2_1.pdf
    Ignored:    code/._MAPT.pdf
    Ignored:    code/._PKD1_DMSO_1.pdf
    Ignored:    code/._PKD1_SM2_1.pdf
    Ignored:    code/.snakemake/
    Ignored:    code/1KG_HighCoverageCalls.samplelist.txt
    Ignored:    code/5ssSeqs.tab
    Ignored:    code/Alignments/
    Ignored:    code/ChemCLIP/
    Ignored:    code/ClinVar/
    Ignored:    code/DE_testing/
    Ignored:    code/DE_tests.mat.counts.gz
    Ignored:    code/DE_tests.txt.gz
    Ignored:    code/DataNotToCommit/
    Ignored:    code/DoseResponseData/
    Ignored:    code/Fastq/
    Ignored:    code/FastqFastp/
    Ignored:    code/FragLenths/
    Ignored:    code/Meme/
    Ignored:    code/Multiqc/
    Ignored:    code/OMIM/
    Ignored:    code/OldBigWigs/
    Ignored:    code/PhyloP/
    Ignored:    code/QC/
    Ignored:    code/ReferenceGenomes/
    Ignored:    code/Session.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TracksSession
    Ignored:    code/bigwigs/
    Ignored:    code/featureCounts/
    Ignored:    code/figs/
    Ignored:    code/geena/
    Ignored:    code/hg38ToMm39.over.chain.gz
    Ignored:    code/igv_session.template.xml
    Ignored:    code/igv_session.xml
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/rstudio-server.job
    Ignored:    code/scratch/
    Ignored:    code/test.txt.gz
    Ignored:    code/testPlottingWithMyScript.ForJingxin.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin2.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin3.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin4.sh
    Ignored:    code/testPlottingWithMyScript.sh
    Ignored:    data/~$52CompoundsTempPlateLayoutForPipettingConvenience.xlsx
    Ignored:    output/._PioritizedIntronTargets.pdf

Unstaged changes:
    Modified:   analysis/20220221_ExploreExpOf52.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/20220221_ExploreExpOf52.Rmd</code>) and HTML
(<code>docs/20220221_ExploreExpOf52.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/analysis/20220221_ExploreExpOf52.Rmd" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
<td>
update 52exp nb
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/SmallMoleculeRNASeq/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/20220221_ExploreExpOf52.html" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
<td>
update 52exp nb
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/analysis/20220221_ExploreExpOf52.Rmd" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
<td>
update exp52 notebook
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/SmallMoleculeRNASeq/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/20220221_ExploreExpOf52.html" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
<td>
update exp52 notebook
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/analysis/20220221_ExploreExpOf52.Rmd" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/SmallMoleculeRNASeq/a8c9152c8020689603208f678ef53d6864f0077c/docs/20220221_ExploreExpOf52.html" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
<td>
update site
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/94cef290271f7226c6cf85aaf1a91754bbecad15/analysis/20220221_ExploreExpOf52.Rmd" target="_blank">94cef29</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-22
</td>
<td>
added Rmd for 52Exp
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)

library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(RColorBrewer)
library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(edgeR)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code>library(qvalue)
library(Heatplus)
library(gplots)</code></pre>
<pre><code>
Attaching package: &#39;gplots&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    lowess</code></pre>
<pre class="r"><code>library(ggrepel)
library(knitr)

# define some useful funcs
sample_n_of &lt;- function(data, size, ...) {
  dots &lt;- quos(...)
  
  group_ids &lt;- data %&gt;% 
    group_by(!!! dots) %&gt;% 
    group_indices()
  
  sampled_groups &lt;- sample(unique(group_ids), size)
  
  data %&gt;% 
    filter(group_ids %in% sampled_groups)
}


# Set theme
theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family=&quot;Helvetica&quot;)))

# I use layer a lot, to rotate long x-axis labels
Rotate_x_labels &lt;- theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

#test plot
ggplot(mtcars, aes(x=mpg, y=cyl)) +
  geom_point()</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/setup-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-setup-1">
Past versions of setup-1.png
</button>
</p>
<div id="fig-setup-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/setup-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>Jingxin’s lab has synthesized (or maybe contracted out synthesis of)
hundred(s) branaplam/risdiplam derivatives. They screened them for
splice-modifying activity using a SMN2 minigene I believe, resulting in
52 compounds that they send Gabi and I for further experiments: We
recieved the 52 compounds in a 96 well plate, and in subsequent plots we
are naming them just by the well position they were shipped in (ie A01,
A02, …, E04). The molar concentration of each molecule was determined by
Jingxin’s lab and shipped at 500x concentration. I think molar
concentration is based on EC90 of SMN2 splice modifying activity in
their initial screen. We grew LCLs cells to a density of approximately
1M cells per mL, and split into seperate flasks each containing 7mL: 2
seperate replicate flasks for each molecule treatment, and 6 flasks of
DMSO control (total of 110 samples), and applied 14uL of treatment. The
following day, cells were collected by centrifugation, supernate was
discarded and stored in 1mL of Trizol at -20C for futher processing. We
extracted RNA by phase seperation (Trizol manufacturer’s instructions).
The top phase was mixed with an equal volume of ethanol, and applied to
Zymo5 nucleic-acid purification columns. After two washes, we performed
on-column DNAse I treatment, followed by an extra application of binding
buffer and two more washes, and finally elution in water. Qubit
quantification was used to quantify RNA before using <a
href="https://www.neb.com/products/e7760-nebnext-ultra-ii-directional-rna-library-prep-kit-for-illumina#Product%20Information">NEBNext
ultra directional ii</a> RNAseq kits with NEB’s polyA capture kit. There
are a couple optional steps in the kit preparation: We fragmented the
RNA for only5 minutes, and performed size selection (step 6.3) to
preferentially obtain longer insert sizes. The libraries were intially
sequenced on a MiSeq (~20M reads total), to determine optimal re-pooling
volumes (aiming for equal representation of each sample) and to confirm
long insert sizes to justify the 150+150 paired end sequencing that we
ended up doing on the NovaSeq S4 (~10B reads total). In practice, I
noticed that one sample had very very low number of reads, and will
likely have to be dropped from analysis (see below). Also, sample
pooling volumes weren’t as perfect as I would have hoped, but I think
they are acceptable. see below.</p>
<p>In the remainder of this notebook, I will explore this data is ways
similar to my previous notebooks regarding Jingxin’s RNAseq - analyses
that I can do quickly. The pre-processing that I have done leading up to
this notebook includes read trimming (<a
href="https://academic.oup.com/bioinformatics/article/34/17/i884/5093234">fastp</a>),
alignment with STAR (I used single pass mode, but perhaps before
publication it might be worth aligning all samples and re-aligning
samples for STAR’s “two-pass” procedure for most sensitive
identification of unannotated junctions), gene counting with
featureCounts, differential expression analysis with edgeR (a separate
contrast for each treatment, comparing to the same 6 DMSO replicates ),
splicing quantification and differential testing with leafcutter (again,
comparing each treatment to 6 DMSO replicates).</p>
<div id="reads-per-sample" class="section level3">
<h3>Reads per sample</h3>
<p>Let’s just consider protein coding genes, as written out in my
reformatted featureCounts output, which I will read in here. I will
occasionally read in data from previous experiments too for
comparison.</p>
<pre class="r"><code># read in sample metadata


Metadata.ExpOf52 &lt;- read_tsv(&quot;../code/config/samples.52MoleculeExperiment.tsv&quot;) %&gt;%
  mutate(cell.type = &quot;LCL&quot;, libType=&quot;polyA&quot;, rep=BioRep, old.sample.name=SampleID, dose.nM=NA) %&gt;%
  mutate(treatment = if_else(Treatment==&quot;DMSO&quot;, &quot;DMSO&quot;, paste0(&quot;W&quot;, Treatment))) %&gt;%
  dplyr::select(treatment, cell.type, dose.nM, libType, rep, old.sample.name) %&gt;%
  mutate(SampleName = paste(treatment, dose.nM, cell.type, libType, rep, sep = &quot;_&quot;))

Metadata.PreviousExperiments &lt;- read_tsv(&quot;../code/bigwigs/BigwigList.tsv&quot;,col_names = c(&quot;SampleName&quot;, &quot;bigwig&quot;, &quot;group&quot;, &quot;strand&quot;)) %&gt;%
  filter(strand==&quot;.&quot;) %&gt;%
  dplyr::select(-strand) %&gt;%
  mutate(old.sample.name = str_replace(bigwig, &quot;/project2/yangili1/bjf79/20211209_JingxinRNAseq/code/bigwigs/unstranded/(.+?).bw&quot;, &quot;\\1&quot;)) %&gt;%
  separate(SampleName, into=c(&quot;treatment&quot;, &quot;dose.nM&quot;, &quot;cell.type&quot;, &quot;libType&quot;, &quot;rep&quot;), convert=T, remove=F, sep=&quot;_&quot;) %&gt;%
  left_join(
    read_tsv(&quot;../code/bigwigs/BigwigList.groups.tsv&quot;, col_names = c(&quot;group&quot;, &quot;color&quot;, &quot;bed&quot;, &quot;supergroup&quot;)),
    by=&quot;group&quot;
  )

FullMetadata &lt;- bind_rows(Metadata.ExpOf52, Metadata.PreviousExperiments) %&gt;%
  mutate(Experiment = case_when(
    cell.type == &quot;Fibroblast&quot; ~ &quot;Single high dose fibroblast&quot;,
    startsWith(old.sample.name, &quot;TitrationExp&quot;) ~ &quot;Dose response titration&quot;,
    startsWith(old.sample.name, &quot;chRNA&quot;) ~ &quot;nascent RNA profiling&quot;,
    startsWith(old.sample.name, &quot;NewMolecule&quot;) ~ &quot;Single high dose LCL&quot;
  )) %&gt;%
  mutate(color = case_when(
    treatment == &quot;DMSO&quot; ~ &quot;#969696&quot;,
    Experiment == &quot;Single high dose LCL&quot; ~ &quot;#252525&quot;,
    TRUE ~ color
  )) %&gt;%
  mutate(leafcutter.name = str_replace_all(old.sample.name, &quot;-&quot;, &quot;.&quot;))



gene.counts &lt;- read_tsv(&quot;../code/DE_testing/ExpOf52_Counts.mat.txt.gz&quot;) %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  DGEList() %&gt;%
  calcNormFactors()

counts.plot.dat &lt;- gene.counts$samples %&gt;%
  rownames_to_column(&quot;old.sample.name&quot;) %&gt;%
  inner_join(FullMetadata, by=&quot;old.sample.name&quot;) %&gt;%
  mutate(dose.nM = case_when(
    treatment == &quot;DMSO&quot; ~ &quot;NA&quot;,
    cell.type == &quot;Fibroblast&quot; ~ &quot;CC50 dose&quot;,
    Experiment == &quot;Single high dose LCL&quot; ~ &quot;SMN_EC90 dose&quot;,
    TRUE ~ as.character(dose.nM)
  )) %&gt;%
  mutate(label = dose.nM) %&gt;%
  arrange(Experiment, treatment, dose.nM)

counts.plot.labels &lt;- counts.plot.dat %&gt;%
  dplyr::select(old.sample.name, label) %&gt;% deframe()

counts.plot.dat %&gt;%
  filter(Experiment==&quot;Single high dose LCL&quot;)</code></pre>
<pre><code>      old.sample.name group.x  lib.size norm.factors treatment cell.type
1   NewMolecule.E05-1       1 135983671    0.9401592      DMSO       LCL
2   NewMolecule.E06-2       1 317779082    0.6907965      DMSO       LCL
3   NewMolecule.E07-3       1  72297626    0.9693137      DMSO       LCL
4   NewMolecule.E05-4       1 129001545    0.8532237      DMSO       LCL
5   NewMolecule.E06-5       1 182905676    0.9465600      DMSO       LCL
6   NewMolecule.E07-6       1 126772689    0.7745714      DMSO       LCL
7   NewMolecule.A01-1       1 183728666    0.9614895      WA01       LCL
8   NewMolecule.A01-2       1 198632154    0.6962443      WA01       LCL
9   NewMolecule.A02-1       1 312182548    0.9826652      WA02       LCL
10  NewMolecule.A02-2       1 183395930    0.9897056      WA02       LCL
11  NewMolecule.A03-1       1 177704482    0.9400929      WA03       LCL
12  NewMolecule.A03-2       1 249934473    0.9561665      WA03       LCL
13  NewMolecule.A04-1       1 174653379    0.9222987      WA04       LCL
14  NewMolecule.A04-2       1 548033015    0.7589615      WA04       LCL
15  NewMolecule.A05-1       1 178435056    0.8614118      WA05       LCL
16  NewMolecule.A05-2       1 153251808    0.7788587      WA05       LCL
17  NewMolecule.A06-1       1 190563753    1.0080036      WA06       LCL
18  NewMolecule.A06-2       1 151314210    0.6000523      WA06       LCL
19  NewMolecule.A07-1       1 123503847    0.7953700      WA07       LCL
20  NewMolecule.A07-2       1 163742739    0.7842479      WA07       LCL
21  NewMolecule.A08-1       1 128532357    0.9945259      WA08       LCL
22  NewMolecule.A08-2       1 191110810    0.8467907      WA08       LCL
23  NewMolecule.A09-1       1 250617437    0.9637614      WA09       LCL
24  NewMolecule.A09-2       1 198092385    0.8870051      WA09       LCL
25  NewMolecule.A10-1       1 233224920    0.9467818      WA10       LCL
26  NewMolecule.A10-2       1 125874155    0.8707373      WA10       LCL
27  NewMolecule.A11-1       1 151421732    0.7724431      WA11       LCL
28  NewMolecule.A11-2       1 159203721    0.6239473      WA11       LCL
29  NewMolecule.A12-1       1 242979912    0.8638066      WA12       LCL
30  NewMolecule.A12-2       1 197958602    0.6814641      WA12       LCL
31  NewMolecule.B01-1       1 176245134    0.8257129      WB01       LCL
32  NewMolecule.B01-2       1 144824255    0.7994645      WB01       LCL
33  NewMolecule.B02-1       1 159560647    0.9525037      WB02       LCL
34  NewMolecule.B02-2       1 181332941    0.8442828      WB02       LCL
35  NewMolecule.B03-1       1 501440283    0.9212623      WB03       LCL
36  NewMolecule.B03-2       1 161562099    0.8031914      WB03       LCL
37  NewMolecule.B04-1       1 363641016    0.9203852      WB04       LCL
38  NewMolecule.B04-2       1 262007104    0.9186215      WB04       LCL
39  NewMolecule.B05-1       1 134091835    0.9803464      WB05       LCL
40  NewMolecule.B05-2       1 129596579    0.9316498      WB05       LCL
41  NewMolecule.B06-1       1 159743971    0.9776352      WB06       LCL
42  NewMolecule.B06-2       1  84886553    0.9970722      WB06       LCL
43  NewMolecule.B07-1       1 160357588    0.8486932      WB07       LCL
44  NewMolecule.B07-2       1 137933937    0.7173181      WB07       LCL
45  NewMolecule.B08-1       1 172609974    1.0109620      WB08       LCL
46  NewMolecule.B08-2       1  82813127    0.9109179      WB08       LCL
47  NewMolecule.B09-1       1 178767085    0.9536717      WB09       LCL
48  NewMolecule.B09-2       1 264104038    0.8378014      WB09       LCL
49  NewMolecule.B10-1       1 238413032    0.8317911      WB10       LCL
50  NewMolecule.B10-2       1 110987864    0.8812357      WB10       LCL
51  NewMolecule.B11-1       1 174607137    0.8259890      WB11       LCL
52  NewMolecule.B11-2       1 168465700    0.7937547      WB11       LCL
53  NewMolecule.B12-1       1 181271838    0.9129708      WB12       LCL
54  NewMolecule.B12-2       1 172779925    0.6964442      WB12       LCL
55  NewMolecule.C01-1       1 244342093    0.9665939      WC01       LCL
56  NewMolecule.C01-2       1 138058013    0.6168776      WC01       LCL
57  NewMolecule.C02-1       1 154946142    0.9809795      WC02       LCL
58  NewMolecule.C02-2       1 187610831    0.7841767      WC02       LCL
59  NewMolecule.C03-1       1 165446125    0.8161006      WC03       LCL
60  NewMolecule.C03-2       1 174266943    0.7833399      WC03       LCL
61  NewMolecule.C04-1       1 148019761    0.9639406      WC04       LCL
62  NewMolecule.C04-2       1   4025684    0.8837338      WC04       LCL
63  NewMolecule.C05-1       1 165455144    0.8796405      WC05       LCL
64  NewMolecule.C05-2       1 174718878    0.7735656      WC05       LCL
65  NewMolecule.C06-1       1 140277168    0.9390193      WC06       LCL
66  NewMolecule.C06-2       1 168324527    0.7628401      WC06       LCL
67  NewMolecule.C07-1       1 261153647    0.9823486      WC07       LCL
68  NewMolecule.C07-2       1 143163993    0.8017761      WC07       LCL
69  NewMolecule.C08-1       1 147676792    0.9694391      WC08       LCL
70  NewMolecule.C08-2       1 108832959    0.9102188      WC08       LCL
71  NewMolecule.C09-1       1 408007760    0.9806656      WC09       LCL
72  NewMolecule.C09-2       1 173089550    0.8327315      WC09       LCL
73  NewMolecule.C10-1       1 167604989    0.9559976      WC10       LCL
74  NewMolecule.C10-2       1 137455261    0.8389724      WC10       LCL
75  NewMolecule.C11-1       1 187805337    0.8355439      WC11       LCL
76  NewMolecule.C11-2       1 177066236    0.7906897      WC11       LCL
77  NewMolecule.C12-1       1 171294673    0.8092100      WC12       LCL
78  NewMolecule.C12-2       1 207161226    0.9439126      WC12       LCL
79  NewMolecule.D01-1       1 148474840    0.8973409      WD01       LCL
80  NewMolecule.D01-2       1 176919098    0.8387012      WD01       LCL
81  NewMolecule.D02-1       1 182860841    0.9258999      WD02       LCL
82  NewMolecule.D02-2       1 186819768    0.9105906      WD02       LCL
83  NewMolecule.D03-1       1 131745383    0.8897832      WD03       LCL
84  NewMolecule.D03-2       1 186120074    0.8657665      WD03       LCL
85  NewMolecule.D04-1       1 147445482    0.9188042      WD04       LCL
86  NewMolecule.D04-2       1 170662427    0.8789052      WD04       LCL
87  NewMolecule.D05-1       1 109693197    0.9849873      WD05       LCL
88  NewMolecule.D05-2       1  55187567    0.9279879      WD05       LCL
89  NewMolecule.D06-1       1 122174618    0.9893905      WD06       LCL
90  NewMolecule.D06-2       1 211272103    0.7724558      WD06       LCL
91  NewMolecule.D07-1       1 186303795    0.9672297      WD07       LCL
92  NewMolecule.D07-2       1 167009833    0.7753588      WD07       LCL
93  NewMolecule.D08-1       1 183462792    1.0004535      WD08       LCL
94  NewMolecule.D08-2       1 146479099    1.0033560      WD08       LCL
95  NewMolecule.D09-1       1 268002864    0.9646019      WD09       LCL
96  NewMolecule.D09-2       1  91945124    0.9215091      WD09       LCL
97  NewMolecule.D10-1       1 278370909    0.9590191      WD10       LCL
98  NewMolecule.D10-2       1 116412533    0.8802938      WD10       LCL
99  NewMolecule.D11-1       1 175024946    0.9210652      WD11       LCL
100 NewMolecule.D11-2       1 106706050    0.8053986      WD11       LCL
101 NewMolecule.D12-1       1  82466922    0.8892412      WD12       LCL
102 NewMolecule.D12-2       1 243296548    0.7777244      WD12       LCL
103 NewMolecule.E01-1       1 123440553    0.9716875      WE01       LCL
104 NewMolecule.E01-2       1 128461448    0.7058736      WE01       LCL
105 NewMolecule.E02-1       1 440121482    0.9524480      WE02       LCL
106 NewMolecule.E02-2       1 127160501    0.7928684      WE02       LCL
107 NewMolecule.E03-1       1 128754472    0.9062433      WE03       LCL
108 NewMolecule.E03-2       1 185922881    0.8547261      WE03       LCL
109 NewMolecule.E04-1       1 468869018    0.9580410      WE04       LCL
110 NewMolecule.E04-2       1 168588670    0.9200381      WE04       LCL
          dose.nM libType rep          SampleName bigwig group.y   color  bed
1              NA   polyA   1 DMSO_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #969696 &lt;NA&gt;
2              NA   polyA   2 DMSO_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #969696 &lt;NA&gt;
3              NA   polyA   3 DMSO_NA_LCL_polyA_3   &lt;NA&gt;    &lt;NA&gt; #969696 &lt;NA&gt;
4              NA   polyA   4 DMSO_NA_LCL_polyA_4   &lt;NA&gt;    &lt;NA&gt; #969696 &lt;NA&gt;
5              NA   polyA   5 DMSO_NA_LCL_polyA_5   &lt;NA&gt;    &lt;NA&gt; #969696 &lt;NA&gt;
6              NA   polyA   6 DMSO_NA_LCL_polyA_6   &lt;NA&gt;    &lt;NA&gt; #969696 &lt;NA&gt;
7   SMN_EC90 dose   polyA   1 WA01_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
8   SMN_EC90 dose   polyA   2 WA01_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
9   SMN_EC90 dose   polyA   1 WA02_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
10  SMN_EC90 dose   polyA   2 WA02_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
11  SMN_EC90 dose   polyA   1 WA03_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
12  SMN_EC90 dose   polyA   2 WA03_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
13  SMN_EC90 dose   polyA   1 WA04_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
14  SMN_EC90 dose   polyA   2 WA04_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
15  SMN_EC90 dose   polyA   1 WA05_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
16  SMN_EC90 dose   polyA   2 WA05_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
17  SMN_EC90 dose   polyA   1 WA06_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
18  SMN_EC90 dose   polyA   2 WA06_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
19  SMN_EC90 dose   polyA   1 WA07_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
20  SMN_EC90 dose   polyA   2 WA07_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
21  SMN_EC90 dose   polyA   1 WA08_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
22  SMN_EC90 dose   polyA   2 WA08_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
23  SMN_EC90 dose   polyA   1 WA09_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
24  SMN_EC90 dose   polyA   2 WA09_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
25  SMN_EC90 dose   polyA   1 WA10_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
26  SMN_EC90 dose   polyA   2 WA10_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
27  SMN_EC90 dose   polyA   1 WA11_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
28  SMN_EC90 dose   polyA   2 WA11_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
29  SMN_EC90 dose   polyA   1 WA12_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
30  SMN_EC90 dose   polyA   2 WA12_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
31  SMN_EC90 dose   polyA   1 WB01_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
32  SMN_EC90 dose   polyA   2 WB01_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
33  SMN_EC90 dose   polyA   1 WB02_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
34  SMN_EC90 dose   polyA   2 WB02_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
35  SMN_EC90 dose   polyA   1 WB03_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
36  SMN_EC90 dose   polyA   2 WB03_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
37  SMN_EC90 dose   polyA   1 WB04_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
38  SMN_EC90 dose   polyA   2 WB04_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
39  SMN_EC90 dose   polyA   1 WB05_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
40  SMN_EC90 dose   polyA   2 WB05_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
41  SMN_EC90 dose   polyA   1 WB06_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
42  SMN_EC90 dose   polyA   2 WB06_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
43  SMN_EC90 dose   polyA   1 WB07_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
44  SMN_EC90 dose   polyA   2 WB07_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
45  SMN_EC90 dose   polyA   1 WB08_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
46  SMN_EC90 dose   polyA   2 WB08_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
47  SMN_EC90 dose   polyA   1 WB09_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
48  SMN_EC90 dose   polyA   2 WB09_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
49  SMN_EC90 dose   polyA   1 WB10_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
50  SMN_EC90 dose   polyA   2 WB10_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
51  SMN_EC90 dose   polyA   1 WB11_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
52  SMN_EC90 dose   polyA   2 WB11_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
53  SMN_EC90 dose   polyA   1 WB12_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
54  SMN_EC90 dose   polyA   2 WB12_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
55  SMN_EC90 dose   polyA   1 WC01_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
56  SMN_EC90 dose   polyA   2 WC01_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
57  SMN_EC90 dose   polyA   1 WC02_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
58  SMN_EC90 dose   polyA   2 WC02_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
59  SMN_EC90 dose   polyA   1 WC03_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
60  SMN_EC90 dose   polyA   2 WC03_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
61  SMN_EC90 dose   polyA   1 WC04_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
62  SMN_EC90 dose   polyA   2 WC04_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
63  SMN_EC90 dose   polyA   1 WC05_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
64  SMN_EC90 dose   polyA   2 WC05_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
65  SMN_EC90 dose   polyA   1 WC06_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
66  SMN_EC90 dose   polyA   2 WC06_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
67  SMN_EC90 dose   polyA   1 WC07_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
68  SMN_EC90 dose   polyA   2 WC07_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
69  SMN_EC90 dose   polyA   1 WC08_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
70  SMN_EC90 dose   polyA   2 WC08_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
71  SMN_EC90 dose   polyA   1 WC09_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
72  SMN_EC90 dose   polyA   2 WC09_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
73  SMN_EC90 dose   polyA   1 WC10_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
74  SMN_EC90 dose   polyA   2 WC10_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
75  SMN_EC90 dose   polyA   1 WC11_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
76  SMN_EC90 dose   polyA   2 WC11_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
77  SMN_EC90 dose   polyA   1 WC12_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
78  SMN_EC90 dose   polyA   2 WC12_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
79  SMN_EC90 dose   polyA   1 WD01_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
80  SMN_EC90 dose   polyA   2 WD01_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
81  SMN_EC90 dose   polyA   1 WD02_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
82  SMN_EC90 dose   polyA   2 WD02_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
83  SMN_EC90 dose   polyA   1 WD03_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
84  SMN_EC90 dose   polyA   2 WD03_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
85  SMN_EC90 dose   polyA   1 WD04_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
86  SMN_EC90 dose   polyA   2 WD04_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
87  SMN_EC90 dose   polyA   1 WD05_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
88  SMN_EC90 dose   polyA   2 WD05_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
89  SMN_EC90 dose   polyA   1 WD06_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
90  SMN_EC90 dose   polyA   2 WD06_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
91  SMN_EC90 dose   polyA   1 WD07_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
92  SMN_EC90 dose   polyA   2 WD07_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
93  SMN_EC90 dose   polyA   1 WD08_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
94  SMN_EC90 dose   polyA   2 WD08_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
95  SMN_EC90 dose   polyA   1 WD09_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
96  SMN_EC90 dose   polyA   2 WD09_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
97  SMN_EC90 dose   polyA   1 WD10_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
98  SMN_EC90 dose   polyA   2 WD10_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
99  SMN_EC90 dose   polyA   1 WD11_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
100 SMN_EC90 dose   polyA   2 WD11_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
101 SMN_EC90 dose   polyA   1 WD12_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
102 SMN_EC90 dose   polyA   2 WD12_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
103 SMN_EC90 dose   polyA   1 WE01_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
104 SMN_EC90 dose   polyA   2 WE01_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
105 SMN_EC90 dose   polyA   1 WE02_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
106 SMN_EC90 dose   polyA   2 WE02_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
107 SMN_EC90 dose   polyA   1 WE03_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
108 SMN_EC90 dose   polyA   2 WE03_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
109 SMN_EC90 dose   polyA   1 WE04_NA_LCL_polyA_1   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
110 SMN_EC90 dose   polyA   2 WE04_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
    supergroup           Experiment   leafcutter.name         label
1         &lt;NA&gt; Single high dose LCL NewMolecule.E05.1            NA
2         &lt;NA&gt; Single high dose LCL NewMolecule.E06.2            NA
3         &lt;NA&gt; Single high dose LCL NewMolecule.E07.3            NA
4         &lt;NA&gt; Single high dose LCL NewMolecule.E05.4            NA
5         &lt;NA&gt; Single high dose LCL NewMolecule.E06.5            NA
6         &lt;NA&gt; Single high dose LCL NewMolecule.E07.6            NA
7         &lt;NA&gt; Single high dose LCL NewMolecule.A01.1 SMN_EC90 dose
8         &lt;NA&gt; Single high dose LCL NewMolecule.A01.2 SMN_EC90 dose
9         &lt;NA&gt; Single high dose LCL NewMolecule.A02.1 SMN_EC90 dose
10        &lt;NA&gt; Single high dose LCL NewMolecule.A02.2 SMN_EC90 dose
11        &lt;NA&gt; Single high dose LCL NewMolecule.A03.1 SMN_EC90 dose
12        &lt;NA&gt; Single high dose LCL NewMolecule.A03.2 SMN_EC90 dose
13        &lt;NA&gt; Single high dose LCL NewMolecule.A04.1 SMN_EC90 dose
14        &lt;NA&gt; Single high dose LCL NewMolecule.A04.2 SMN_EC90 dose
15        &lt;NA&gt; Single high dose LCL NewMolecule.A05.1 SMN_EC90 dose
16        &lt;NA&gt; Single high dose LCL NewMolecule.A05.2 SMN_EC90 dose
17        &lt;NA&gt; Single high dose LCL NewMolecule.A06.1 SMN_EC90 dose
18        &lt;NA&gt; Single high dose LCL NewMolecule.A06.2 SMN_EC90 dose
19        &lt;NA&gt; Single high dose LCL NewMolecule.A07.1 SMN_EC90 dose
20        &lt;NA&gt; Single high dose LCL NewMolecule.A07.2 SMN_EC90 dose
21        &lt;NA&gt; Single high dose LCL NewMolecule.A08.1 SMN_EC90 dose
22        &lt;NA&gt; Single high dose LCL NewMolecule.A08.2 SMN_EC90 dose
23        &lt;NA&gt; Single high dose LCL NewMolecule.A09.1 SMN_EC90 dose
24        &lt;NA&gt; Single high dose LCL NewMolecule.A09.2 SMN_EC90 dose
25        &lt;NA&gt; Single high dose LCL NewMolecule.A10.1 SMN_EC90 dose
26        &lt;NA&gt; Single high dose LCL NewMolecule.A10.2 SMN_EC90 dose
27        &lt;NA&gt; Single high dose LCL NewMolecule.A11.1 SMN_EC90 dose
28        &lt;NA&gt; Single high dose LCL NewMolecule.A11.2 SMN_EC90 dose
29        &lt;NA&gt; Single high dose LCL NewMolecule.A12.1 SMN_EC90 dose
30        &lt;NA&gt; Single high dose LCL NewMolecule.A12.2 SMN_EC90 dose
31        &lt;NA&gt; Single high dose LCL NewMolecule.B01.1 SMN_EC90 dose
32        &lt;NA&gt; Single high dose LCL NewMolecule.B01.2 SMN_EC90 dose
33        &lt;NA&gt; Single high dose LCL NewMolecule.B02.1 SMN_EC90 dose
34        &lt;NA&gt; Single high dose LCL NewMolecule.B02.2 SMN_EC90 dose
35        &lt;NA&gt; Single high dose LCL NewMolecule.B03.1 SMN_EC90 dose
36        &lt;NA&gt; Single high dose LCL NewMolecule.B03.2 SMN_EC90 dose
37        &lt;NA&gt; Single high dose LCL NewMolecule.B04.1 SMN_EC90 dose
38        &lt;NA&gt; Single high dose LCL NewMolecule.B04.2 SMN_EC90 dose
39        &lt;NA&gt; Single high dose LCL NewMolecule.B05.1 SMN_EC90 dose
40        &lt;NA&gt; Single high dose LCL NewMolecule.B05.2 SMN_EC90 dose
41        &lt;NA&gt; Single high dose LCL NewMolecule.B06.1 SMN_EC90 dose
42        &lt;NA&gt; Single high dose LCL NewMolecule.B06.2 SMN_EC90 dose
43        &lt;NA&gt; Single high dose LCL NewMolecule.B07.1 SMN_EC90 dose
44        &lt;NA&gt; Single high dose LCL NewMolecule.B07.2 SMN_EC90 dose
45        &lt;NA&gt; Single high dose LCL NewMolecule.B08.1 SMN_EC90 dose
46        &lt;NA&gt; Single high dose LCL NewMolecule.B08.2 SMN_EC90 dose
47        &lt;NA&gt; Single high dose LCL NewMolecule.B09.1 SMN_EC90 dose
48        &lt;NA&gt; Single high dose LCL NewMolecule.B09.2 SMN_EC90 dose
49        &lt;NA&gt; Single high dose LCL NewMolecule.B10.1 SMN_EC90 dose
50        &lt;NA&gt; Single high dose LCL NewMolecule.B10.2 SMN_EC90 dose
51        &lt;NA&gt; Single high dose LCL NewMolecule.B11.1 SMN_EC90 dose
52        &lt;NA&gt; Single high dose LCL NewMolecule.B11.2 SMN_EC90 dose
53        &lt;NA&gt; Single high dose LCL NewMolecule.B12.1 SMN_EC90 dose
54        &lt;NA&gt; Single high dose LCL NewMolecule.B12.2 SMN_EC90 dose
55        &lt;NA&gt; Single high dose LCL NewMolecule.C01.1 SMN_EC90 dose
56        &lt;NA&gt; Single high dose LCL NewMolecule.C01.2 SMN_EC90 dose
57        &lt;NA&gt; Single high dose LCL NewMolecule.C02.1 SMN_EC90 dose
58        &lt;NA&gt; Single high dose LCL NewMolecule.C02.2 SMN_EC90 dose
59        &lt;NA&gt; Single high dose LCL NewMolecule.C03.1 SMN_EC90 dose
60        &lt;NA&gt; Single high dose LCL NewMolecule.C03.2 SMN_EC90 dose
61        &lt;NA&gt; Single high dose LCL NewMolecule.C04.1 SMN_EC90 dose
62        &lt;NA&gt; Single high dose LCL NewMolecule.C04.2 SMN_EC90 dose
63        &lt;NA&gt; Single high dose LCL NewMolecule.C05.1 SMN_EC90 dose
64        &lt;NA&gt; Single high dose LCL NewMolecule.C05.2 SMN_EC90 dose
65        &lt;NA&gt; Single high dose LCL NewMolecule.C06.1 SMN_EC90 dose
66        &lt;NA&gt; Single high dose LCL NewMolecule.C06.2 SMN_EC90 dose
67        &lt;NA&gt; Single high dose LCL NewMolecule.C07.1 SMN_EC90 dose
68        &lt;NA&gt; Single high dose LCL NewMolecule.C07.2 SMN_EC90 dose
69        &lt;NA&gt; Single high dose LCL NewMolecule.C08.1 SMN_EC90 dose
70        &lt;NA&gt; Single high dose LCL NewMolecule.C08.2 SMN_EC90 dose
71        &lt;NA&gt; Single high dose LCL NewMolecule.C09.1 SMN_EC90 dose
72        &lt;NA&gt; Single high dose LCL NewMolecule.C09.2 SMN_EC90 dose
73        &lt;NA&gt; Single high dose LCL NewMolecule.C10.1 SMN_EC90 dose
74        &lt;NA&gt; Single high dose LCL NewMolecule.C10.2 SMN_EC90 dose
75        &lt;NA&gt; Single high dose LCL NewMolecule.C11.1 SMN_EC90 dose
76        &lt;NA&gt; Single high dose LCL NewMolecule.C11.2 SMN_EC90 dose
77        &lt;NA&gt; Single high dose LCL NewMolecule.C12.1 SMN_EC90 dose
78        &lt;NA&gt; Single high dose LCL NewMolecule.C12.2 SMN_EC90 dose
79        &lt;NA&gt; Single high dose LCL NewMolecule.D01.1 SMN_EC90 dose
80        &lt;NA&gt; Single high dose LCL NewMolecule.D01.2 SMN_EC90 dose
81        &lt;NA&gt; Single high dose LCL NewMolecule.D02.1 SMN_EC90 dose
82        &lt;NA&gt; Single high dose LCL NewMolecule.D02.2 SMN_EC90 dose
83        &lt;NA&gt; Single high dose LCL NewMolecule.D03.1 SMN_EC90 dose
84        &lt;NA&gt; Single high dose LCL NewMolecule.D03.2 SMN_EC90 dose
85        &lt;NA&gt; Single high dose LCL NewMolecule.D04.1 SMN_EC90 dose
86        &lt;NA&gt; Single high dose LCL NewMolecule.D04.2 SMN_EC90 dose
87        &lt;NA&gt; Single high dose LCL NewMolecule.D05.1 SMN_EC90 dose
88        &lt;NA&gt; Single high dose LCL NewMolecule.D05.2 SMN_EC90 dose
89        &lt;NA&gt; Single high dose LCL NewMolecule.D06.1 SMN_EC90 dose
90        &lt;NA&gt; Single high dose LCL NewMolecule.D06.2 SMN_EC90 dose
91        &lt;NA&gt; Single high dose LCL NewMolecule.D07.1 SMN_EC90 dose
92        &lt;NA&gt; Single high dose LCL NewMolecule.D07.2 SMN_EC90 dose
93        &lt;NA&gt; Single high dose LCL NewMolecule.D08.1 SMN_EC90 dose
94        &lt;NA&gt; Single high dose LCL NewMolecule.D08.2 SMN_EC90 dose
95        &lt;NA&gt; Single high dose LCL NewMolecule.D09.1 SMN_EC90 dose
96        &lt;NA&gt; Single high dose LCL NewMolecule.D09.2 SMN_EC90 dose
97        &lt;NA&gt; Single high dose LCL NewMolecule.D10.1 SMN_EC90 dose
98        &lt;NA&gt; Single high dose LCL NewMolecule.D10.2 SMN_EC90 dose
99        &lt;NA&gt; Single high dose LCL NewMolecule.D11.1 SMN_EC90 dose
100       &lt;NA&gt; Single high dose LCL NewMolecule.D11.2 SMN_EC90 dose
101       &lt;NA&gt; Single high dose LCL NewMolecule.D12.1 SMN_EC90 dose
102       &lt;NA&gt; Single high dose LCL NewMolecule.D12.2 SMN_EC90 dose
103       &lt;NA&gt; Single high dose LCL NewMolecule.E01.1 SMN_EC90 dose
104       &lt;NA&gt; Single high dose LCL NewMolecule.E01.2 SMN_EC90 dose
105       &lt;NA&gt; Single high dose LCL NewMolecule.E02.1 SMN_EC90 dose
106       &lt;NA&gt; Single high dose LCL NewMolecule.E02.2 SMN_EC90 dose
107       &lt;NA&gt; Single high dose LCL NewMolecule.E03.1 SMN_EC90 dose
108       &lt;NA&gt; Single high dose LCL NewMolecule.E03.2 SMN_EC90 dose
109       &lt;NA&gt; Single high dose LCL NewMolecule.E04.1 SMN_EC90 dose
110       &lt;NA&gt; Single high dose LCL NewMolecule.E04.2 SMN_EC90 dose</code></pre>
<pre class="r"><code>ReadsPerDataset &lt;- ggplot(counts.plot.dat, aes(x=old.sample.name, y=lib.size/2E6, fill=color)) +
  geom_col() +
  scale_fill_identity() +
  scale_x_discrete(name=&quot;dose (nanomolar)&quot;, label=counts.plot.labels) +
  scale_y_continuous(expand=c(0,0)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size=3)) +
  theme(strip.text.x = element_text(size = 8)) +
  facet_grid(~Experiment, scales = &quot;free_x&quot;, space = &quot;free_x&quot;, labeller = label_wrap_gen(15)) +
  labs(title=&quot;RNA-seq datasets&quot;, y=&quot;Read count (M)&quot;)
ReadsPerDataset</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-1-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>So for the most part, these libraries were sequenced deeper than any
of our previous experiments with Jingxin. Let’s make note of that one
outlier sample that was clearly not sequenced well. This sample will
probably have to be excluded from further analysis…</p>
<pre class="r"><code>counts.plot.dat %&gt;%
  arrange(lib.size) %&gt;%
  head(1)</code></pre>
<pre><code>    old.sample.name group.x lib.size norm.factors treatment cell.type
1 NewMolecule.C04-2       1  4025684    0.8837338      WC04       LCL
        dose.nM libType rep          SampleName bigwig group.y   color  bed
1 SMN_EC90 dose   polyA   2 WC04_NA_LCL_polyA_2   &lt;NA&gt;    &lt;NA&gt; #252525 &lt;NA&gt;
  supergroup           Experiment   leafcutter.name         label
1       &lt;NA&gt; Single high dose LCL NewMolecule.C04.2 SMN_EC90 dose</code></pre>
<p>Ok so rep2 of the molecule in well C04 will probably have to be
excluded. 4M reads isn’t enough to get much out of, and I sort of worry
whether those 4M reads are even from the corresponding library versus
barcode contamination from other samples on the lane…</p>
</div>
<div id="gene-expression-pca" class="section level3">
<h3>Gene expression PCA</h3>
<p>First let’s perform PCA with all samples, including fibroblasts. I
know this isn’t necessarily the most interpretable, but I am just
curious what the first few PCs will look like…</p>
<pre class="r"><code>SamplesToInclude &lt;- FullMetadata %&gt;%
  pull(old.sample.name)

CPM &lt;- gene.counts %&gt;%
  cpm(log=T, prior.count=T) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Geneid&quot;) %&gt;%
  dplyr::select(Geneid, all_of(SamplesToInclude))

Top14K_ExpressedGenes &lt;- (CPM %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  apply(1, mean) %&gt;%
  sort(decreasing=T))[1:14000] %&gt;%
  names()

pca.results &lt;- CPM %&gt;%
  filter(Geneid %in% Top14K_ExpressedGenes) %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% prcomp(scale=T)

summary(pca.results)</code></pre>
<pre><code>Importance of components:
                           PC1     PC2     PC3      PC4      PC5      PC6
Standard deviation     72.6439 62.2561 42.4165 30.06520 18.23340 14.95023
Proportion of Variance  0.3769  0.2768  0.1285  0.06457  0.02375  0.01596
Cumulative Proportion   0.3769  0.6538  0.7823  0.84686  0.87061  0.88657
                            PC7      PC8      PC9    PC10    PC11    PC12
Standard deviation     12.10732 11.51993 10.52554 8.64035 8.05303 7.52829
Proportion of Variance  0.01047  0.00948  0.00791 0.00533 0.00463 0.00405
Cumulative Proportion   0.89704  0.90652  0.91443 0.91977 0.92440 0.92845
                          PC13    PC14    PC15    PC16    PC17    PC18   PC19
Standard deviation     7.43240 6.07546 5.90323 5.71783 5.45541 4.70763 4.5829
Proportion of Variance 0.00395 0.00264 0.00249 0.00234 0.00213 0.00158 0.0015
Cumulative Proportion  0.93239 0.93503 0.93752 0.93985 0.94198 0.94356 0.9451
                          PC20   PC21    PC22    PC23    PC24    PC25    PC26
Standard deviation     4.34842 4.2642 4.18983 4.05111 3.70776 3.66428 3.52860
Proportion of Variance 0.00135 0.0013 0.00125 0.00117 0.00098 0.00096 0.00089
Cumulative Proportion  0.94641 0.9477 0.94897 0.95014 0.95112 0.95208 0.95297
                          PC27    PC28    PC29    PC30    PC31    PC32   PC33
Standard deviation     3.46988 3.42074 3.30835 3.25718 3.20878 3.19963 3.1289
Proportion of Variance 0.00086 0.00084 0.00078 0.00076 0.00074 0.00073 0.0007
Cumulative Proportion  0.95383 0.95466 0.95545 0.95620 0.95694 0.95767 0.9584
                          PC34    PC35    PC36    PC37    PC38    PC39    PC40
Standard deviation     3.09510 3.07249 3.04599 3.02764 2.96070 2.91136 2.87777
Proportion of Variance 0.00068 0.00067 0.00066 0.00065 0.00063 0.00061 0.00059
Cumulative Proportion  0.95905 0.95973 0.96039 0.96105 0.96167 0.96228 0.96287
                          PC41    PC42    PC43    PC44    PC45    PC46    PC47
Standard deviation     2.85418 2.81059 2.76061 2.75270 2.74102 2.70975 2.66896
Proportion of Variance 0.00058 0.00056 0.00054 0.00054 0.00054 0.00052 0.00051
Cumulative Proportion  0.96345 0.96402 0.96456 0.96510 0.96564 0.96616 0.96667
                         PC48    PC49    PC50    PC51    PC52    PC53    PC54
Standard deviation     2.6481 2.62696 2.59712 2.56767 2.55087 2.53274 2.49007
Proportion of Variance 0.0005 0.00049 0.00048 0.00047 0.00046 0.00046 0.00044
Cumulative Proportion  0.9672 0.96767 0.96815 0.96862 0.96908 0.96954 0.96998
                          PC55    PC56    PC57    PC58    PC59    PC60    PC61
Standard deviation     2.48213 2.47583 2.46033 2.44854 2.41859 2.40764 2.38683
Proportion of Variance 0.00044 0.00044 0.00043 0.00043 0.00042 0.00041 0.00041
Cumulative Proportion  0.97042 0.97086 0.97129 0.97172 0.97214 0.97255 0.97296
                         PC62    PC63    PC64    PC65    PC66    PC67    PC68
Standard deviation     2.3676 2.34748 2.33774 2.32323 2.30991 2.29735 2.28203
Proportion of Variance 0.0004 0.00039 0.00039 0.00039 0.00038 0.00038 0.00037
Cumulative Proportion  0.9734 0.97375 0.97415 0.97453 0.97491 0.97529 0.97566
                          PC69    PC70    PC71    PC72    PC73    PC74    PC75
Standard deviation     2.26430 2.25797 2.25197 2.23527 2.21472 2.20971 2.19590
Proportion of Variance 0.00037 0.00036 0.00036 0.00036 0.00035 0.00035 0.00034
Cumulative Proportion  0.97603 0.97639 0.97675 0.97711 0.97746 0.97781 0.97815
                          PC76    PC77    PC78    PC79    PC80    PC81    PC82
Standard deviation     2.19124 2.16918 2.15949 2.14600 2.13137 2.11768 2.10960
Proportion of Variance 0.00034 0.00034 0.00033 0.00033 0.00032 0.00032 0.00032
Cumulative Proportion  0.97850 0.97883 0.97917 0.97950 0.97982 0.98014 0.98046
                          PC83    PC84    PC85    PC86    PC87   PC88   PC89
Standard deviation     2.10420 2.09067 2.07932 2.07158 2.06890 2.0552 2.0510
Proportion of Variance 0.00032 0.00031 0.00031 0.00031 0.00031 0.0003 0.0003
Cumulative Proportion  0.98077 0.98109 0.98140 0.98170 0.98201 0.9823 0.9826
                         PC90    PC91    PC92    PC93    PC94    PC95    PC96
Standard deviation     2.0325 2.02628 2.02095 2.01505 2.01453 2.00140 1.99207
Proportion of Variance 0.0003 0.00029 0.00029 0.00029 0.00029 0.00029 0.00028
Cumulative Proportion  0.9829 0.98320 0.98349 0.98378 0.98407 0.98436 0.98464
                          PC97    PC98    PC99   PC100   PC101   PC102   PC103
Standard deviation     1.97879 1.96978 1.96142 1.95843 1.94605 1.93103 1.92697
Proportion of Variance 0.00028 0.00028 0.00027 0.00027 0.00027 0.00027 0.00027
Cumulative Proportion  0.98492 0.98520 0.98547 0.98574 0.98602 0.98628 0.98655
                         PC104   PC105   PC106   PC107   PC108   PC109   PC110
Standard deviation     1.91522 1.90976 1.90458 1.90223 1.89465 1.89031 1.88708
Proportion of Variance 0.00026 0.00026 0.00026 0.00026 0.00026 0.00026 0.00025
Cumulative Proportion  0.98681 0.98707 0.98733 0.98759 0.98784 0.98810 0.98835
                         PC111   PC112   PC113   PC114   PC115   PC116   PC117
Standard deviation     1.86688 1.86214 1.85424 1.85032 1.83938 1.82714 1.82318
Proportion of Variance 0.00025 0.00025 0.00025 0.00024 0.00024 0.00024 0.00024
Cumulative Proportion  0.98860 0.98885 0.98910 0.98934 0.98958 0.98982 0.99006
                         PC118   PC119   PC120   PC121   PC122   PC123   PC124
Standard deviation     1.80790 1.80509 1.79744 1.78928 1.77623 1.77575 1.76304
Proportion of Variance 0.00023 0.00023 0.00023 0.00023 0.00023 0.00023 0.00022
Cumulative Proportion  0.99029 0.99052 0.99075 0.99098 0.99121 0.99143 0.99166
                         PC125   PC126   PC127   PC128   PC129   PC130   PC131
Standard deviation     1.75581 1.74527 1.74072 1.73710 1.72480 1.71605 1.71063
Proportion of Variance 0.00022 0.00022 0.00022 0.00022 0.00021 0.00021 0.00021
Cumulative Proportion  0.99188 0.99209 0.99231 0.99253 0.99274 0.99295 0.99316
                         PC132   PC133  PC134  PC135  PC136  PC137   PC138
Standard deviation     1.70481 1.69772 1.6895 1.6757 1.6641 1.6591 1.64981
Proportion of Variance 0.00021 0.00021 0.0002 0.0002 0.0002 0.0002 0.00019
Cumulative Proportion  0.99336 0.99357 0.9938 0.9940 0.9942 0.9944 0.99456
                         PC139   PC140   PC141   PC142   PC143   PC144   PC145
Standard deviation     1.64177 1.63931 1.62764 1.61481 1.61010 1.60438 1.59635
Proportion of Variance 0.00019 0.00019 0.00019 0.00019 0.00019 0.00018 0.00018
Cumulative Proportion  0.99476 0.99495 0.99514 0.99532 0.99551 0.99569 0.99587
                         PC146   PC147   PC148   PC149   PC150   PC151   PC152
Standard deviation     1.59185 1.58410 1.57662 1.56689 1.55334 1.55064 1.54392
Proportion of Variance 0.00018 0.00018 0.00018 0.00018 0.00017 0.00017 0.00017
Cumulative Proportion  0.99606 0.99624 0.99641 0.99659 0.99676 0.99693 0.99710
                         PC153   PC154   PC155   PC156   PC157   PC158   PC159
Standard deviation     1.52924 1.52064 1.51619 1.50377 1.49740 1.48478 1.47533
Proportion of Variance 0.00017 0.00017 0.00016 0.00016 0.00016 0.00016 0.00016
Cumulative Proportion  0.99727 0.99743 0.99760 0.99776 0.99792 0.99808 0.99823
                         PC160   PC161   PC162   PC163   PC164   PC165   PC166
Standard deviation     1.46910 1.45457 1.44128 1.42765 1.41422 1.39923 1.38427
Proportion of Variance 0.00015 0.00015 0.00015 0.00015 0.00014 0.00014 0.00014
Cumulative Proportion  0.99839 0.99854 0.99869 0.99883 0.99898 0.99912 0.99925
                         PC167   PC168   PC169   PC170   PC171   PC172
Standard deviation     1.37542 1.37181 1.35146 1.31545 1.27766 1.22693
Proportion of Variance 0.00014 0.00013 0.00013 0.00012 0.00012 0.00011
Cumulative Proportion  0.99939 0.99952 0.99965 0.99978 0.99989 1.00000
                           PC173
Standard deviation     1.839e-14
Proportion of Variance 0.000e+00
Cumulative Proportion  1.000e+00</code></pre>
<pre class="r"><code>pca.results.to.plot &lt;- pca.results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;old.sample.name&quot;) %&gt;%
  dplyr::select(old.sample.name, PC1:PC6) %&gt;%
  left_join(FullMetadata, by=&quot;old.sample.name&quot;)

ggplot(pca.results.to.plot, aes(x=PC1, y=PC2, color=color, shape=Experiment)) +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using top 14K genes\nAll samples&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-3-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC3, y=PC4, color=color, shape=Experiment)) +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using top 14K genes\nAll samples&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-3-2.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC5, y=PC6, color=color, shape=Experiment)) +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using top 14K genes\nAll samples&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-3">
Past versions of unnamed-chunk-3-3.png
</button>
</p>
<div id="fig-unnamed-chunk-3-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-3-3.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Let’s repeat but just include the dose titration experiment samples
and this recent experiment of 52 molecules…</p>
<pre class="r"><code>SamplesToInclude &lt;- FullMetadata %&gt;%
  filter(Experiment %in% c(&quot;Single high dose LCL&quot;, &quot;Dose response titration&quot;)) %&gt;%
  pull(old.sample.name)

CPM &lt;- gene.counts %&gt;%
  cpm(log=T, prior.count=T) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Geneid&quot;) %&gt;%
  dplyr::select(Geneid, all_of(SamplesToInclude))

Top14K_ExpressedGenes &lt;- (CPM %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  apply(1, mean) %&gt;%
  sort(decreasing=T))[1:14000] %&gt;%
  names()

pca.results &lt;- CPM %&gt;%
  filter(Geneid %in% Top14K_ExpressedGenes) %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% prcomp(scale=T)

summary(pca.results)</code></pre>
<pre><code>Importance of components:
                           PC1     PC2      PC3      PC4      PC5      PC6
Standard deviation     83.1894 55.0561 24.77276 24.28695 19.94844 14.70160
Proportion of Variance  0.4943  0.2165  0.04383  0.04213  0.02842  0.01544
Cumulative Proportion   0.4943  0.7108  0.75467  0.79680  0.82522  0.84066
                            PC7     PC8      PC9    PC10    PC11    PC12
Standard deviation     13.00748 11.7139 10.86657 9.32844 8.28935 7.53691
Proportion of Variance  0.01209  0.0098  0.00843 0.00622 0.00491 0.00406
Cumulative Proportion   0.85275  0.8626  0.87098 0.87720 0.88211 0.88616
                          PC13    PC14    PC15    PC16    PC17    PC18   PC19
Standard deviation     7.24574 6.92730 6.29126 5.97731 5.69311 5.48199 5.2932
Proportion of Variance 0.00375 0.00343 0.00283 0.00255 0.00232 0.00215 0.0020
Cumulative Proportion  0.88991 0.89334 0.89617 0.89872 0.90104 0.90318 0.9052
                          PC20    PC21    PC22    PC23    PC24    PC25    PC26
Standard deviation     5.04198 4.97192 4.91997 4.79399 4.69313 4.61243 4.55639
Proportion of Variance 0.00182 0.00177 0.00173 0.00164 0.00157 0.00152 0.00148
Cumulative Proportion  0.90700 0.90877 0.91050 0.91214 0.91371 0.91523 0.91671
                          PC27    PC28   PC29    PC30    PC31    PC32    PC33
Standard deviation     4.54571 4.46480 4.4291 4.38057 4.30252 4.24036 4.19862
Proportion of Variance 0.00148 0.00142 0.0014 0.00137 0.00132 0.00128 0.00126
Cumulative Proportion  0.91819 0.91961 0.9210 0.92238 0.92371 0.92499 0.92625
                          PC34    PC35   PC36    PC37    PC38    PC39    PC40
Standard deviation     4.17229 4.12458 4.0955 4.07422 4.06222 3.98905 3.96266
Proportion of Variance 0.00124 0.00122 0.0012 0.00119 0.00118 0.00114 0.00112
Cumulative Proportion  0.92749 0.92871 0.9299 0.93109 0.93227 0.93341 0.93453
                         PC41    PC42    PC43    PC44    PC45    PC46    PC47
Standard deviation     3.9299 3.89363 3.89227 3.82138 3.81024 3.77689 3.75136
Proportion of Variance 0.0011 0.00108 0.00108 0.00104 0.00104 0.00102 0.00101
Cumulative Proportion  0.9356 0.93672 0.93780 0.93884 0.93988 0.94090 0.94190
                          PC48    PC49    PC50    PC51    PC52    PC53    PC54
Standard deviation     3.72866 3.72341 3.69228 3.66300 3.65323 3.62103 3.60460
Proportion of Variance 0.00099 0.00099 0.00097 0.00096 0.00095 0.00094 0.00093
Cumulative Proportion  0.94289 0.94388 0.94486 0.94582 0.94677 0.94771 0.94864
                          PC55    PC56   PC57    PC58    PC59    PC60    PC61
Standard deviation     3.58833 3.57787 3.5416 3.51746 3.50854 3.48229 3.46762
Proportion of Variance 0.00092 0.00091 0.0009 0.00088 0.00088 0.00087 0.00086
Cumulative Proportion  0.94955 0.95047 0.9514 0.95225 0.95313 0.95399 0.95485
                          PC62    PC63    PC64    PC65    PC66    PC67   PC68
Standard deviation     3.45692 3.43346 3.42045 3.39272 3.38289 3.36342 3.3474
Proportion of Variance 0.00085 0.00084 0.00084 0.00082 0.00082 0.00081 0.0008
Cumulative Proportion  0.95571 0.95655 0.95738 0.95821 0.95902 0.95983 0.9606
                          PC69    PC70    PC71    PC72    PC73    PC74    PC75
Standard deviation     3.32739 3.31967 3.30262 3.28465 3.28166 3.25750 3.23148
Proportion of Variance 0.00079 0.00079 0.00078 0.00077 0.00077 0.00076 0.00075
Cumulative Proportion  0.96142 0.96221 0.96299 0.96376 0.96453 0.96529 0.96603
                          PC76    PC77    PC78    PC79    PC80    PC81    PC82
Standard deviation     3.22940 3.21138 3.19718 3.18893 3.17672 3.16148 3.14200
Proportion of Variance 0.00074 0.00074 0.00073 0.00073 0.00072 0.00071 0.00071
Cumulative Proportion  0.96678 0.96751 0.96825 0.96897 0.96969 0.97041 0.97111
                         PC83   PC84    PC85    PC86    PC87    PC88    PC89
Standard deviation     3.1396 3.1281 3.11272 3.09873 3.07685 3.03933 3.03570
Proportion of Variance 0.0007 0.0007 0.00069 0.00069 0.00068 0.00066 0.00066
Cumulative Proportion  0.9718 0.9725 0.97321 0.97389 0.97457 0.97523 0.97589
                          PC90    PC91    PC92    PC93    PC94    PC95    PC96
Standard deviation     3.02075 3.01671 2.99872 2.98609 2.97603 2.95891 2.94129
Proportion of Variance 0.00065 0.00065 0.00064 0.00064 0.00063 0.00063 0.00062
Cumulative Proportion  0.97654 0.97719 0.97783 0.97847 0.97910 0.97973 0.98034
                          PC97    PC98   PC99   PC100   PC101   PC102   PC103
Standard deviation     2.91912 2.91648 2.8908 2.87791 2.86451 2.84156 2.83717
Proportion of Variance 0.00061 0.00061 0.0006 0.00059 0.00059 0.00058 0.00057
Cumulative Proportion  0.98095 0.98156 0.9822 0.98275 0.98333 0.98391 0.98449
                         PC104   PC105   PC106   PC107   PC108   PC109   PC110
Standard deviation     2.82141 2.79989 2.77712 2.76955 2.75960 2.74038 2.73493
Proportion of Variance 0.00057 0.00056 0.00055 0.00055 0.00054 0.00054 0.00053
Cumulative Proportion  0.98505 0.98561 0.98617 0.98671 0.98726 0.98779 0.98833
                         PC111   PC112   PC113   PC114   PC115  PC116  PC117
Standard deviation     2.72667 2.71699 2.69161 2.67425 2.67144 2.6408 2.6329
Proportion of Variance 0.00053 0.00053 0.00052 0.00051 0.00051 0.0005 0.0005
Cumulative Proportion  0.98886 0.98939 0.98990 0.99041 0.99092 0.9914 0.9919
                         PC118   PC119   PC120   PC121   PC122   PC123   PC124
Standard deviation     2.62226 2.60273 2.57230 2.56833 2.53999 2.53069 2.50099
Proportion of Variance 0.00049 0.00048 0.00047 0.00047 0.00046 0.00046 0.00045
Cumulative Proportion  0.99241 0.99289 0.99337 0.99384 0.99430 0.99475 0.99520
                         PC125   PC126   PC127   PC128   PC129   PC130  PC131
Standard deviation     2.48410 2.47870 2.46989 2.41798 2.40858 2.39057 2.3799
Proportion of Variance 0.00044 0.00044 0.00044 0.00042 0.00041 0.00041 0.0004
Cumulative Proportion  0.99564 0.99608 0.99652 0.99693 0.99735 0.99776 0.9982
                        PC132   PC133   PC134   PC135   PC136     PC137
Standard deviation     2.3681 2.33293 2.28534 2.21770 2.13166 3.325e-14
Proportion of Variance 0.0004 0.00039 0.00037 0.00035 0.00032 0.000e+00
Cumulative Proportion  0.9986 0.99895 0.99932 0.99968 1.00000 1.000e+00</code></pre>
<pre class="r"><code>pca.results.to.plot &lt;- pca.results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;old.sample.name&quot;) %&gt;%
  dplyr::select(old.sample.name, PC1:PC6) %&gt;%
  left_join(FullMetadata, by=&quot;old.sample.name&quot;)

ggplot(pca.results.to.plot, aes(x=PC1, y=PC2, color=color, shape=Experiment)) +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using top 14K genes\nTitration experiment and ExpOf52&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-4-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC3, y=PC4, color=color, shape=Experiment)) +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using top 14K genes\nTitration experiment and ExpOf52&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-4-2.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC5, y=PC6, color=color, shape=Experiment)) +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using top 14K genes\nTitration experiment and ExpOf52&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-3">
Past versions of unnamed-chunk-4-3.png
</button>
</p>
<div id="fig-unnamed-chunk-4-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-4-3.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>pca.results.to.plot %&gt;%
  filter(PC5&gt;200)</code></pre>
<pre><code>    old.sample.name      PC1      PC2      PC3      PC4      PC5      PC6
1 NewMolecule.C04-2 78.41428 41.81794 7.094914 17.13685 217.0802 48.79238
  treatment cell.type dose.nM libType rep          SampleName bigwig group
1      WC04       LCL      NA   polyA   2 WC04_NA_LCL_polyA_2   &lt;NA&gt;  &lt;NA&gt;
    color  bed supergroup           Experiment   leafcutter.name
1 #252525 &lt;NA&gt;       &lt;NA&gt; Single high dose LCL NewMolecule.C04.2</code></pre>
<p>Ok… some interesting patterns. That one outlier in PC5 is the sample
with low read depth that I want to exclude. It seems like the lab/batch
effects are quite strong, the first two PCs have nothing to do with
dose. (Lighter gray points are the DMSO controls in both experiments)..
Maybe PC3 is related to this? But it’s not going to be so
straightforward to integrate this experiment of 52 data with the
previous dose titration experiment also done in LCLs. Well at least
previously when I perform PCA based on splicing junction excision ratios
these batch and tissue effects seem to dissappear in the PCA, so maybe
those splicing comparisons might be more fair without more careful
integration. Now let’s try PCA again but now just looking at the samples
in this recent experiment of 52 molecules, to check at least that the
DMSO samples cluster together…</p>
<pre class="r"><code>SamplesToInclude &lt;- FullMetadata %&gt;%
  filter(Experiment %in% c(&quot;Single high dose LCL&quot;)) %&gt;%
  filter(!old.sample.name == &quot;NewMolecule.C04-2&quot;) %&gt;%
  pull(old.sample.name)

CPM &lt;- gene.counts %&gt;%
  cpm(log=T, prior.count=T) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Geneid&quot;) %&gt;%
  dplyr::select(Geneid, all_of(SamplesToInclude))

Top14K_ExpressedGenes &lt;- (CPM %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  apply(1, mean) %&gt;%
  sort(decreasing=T))[1:14000] %&gt;%
  names()

pca.results &lt;- CPM %&gt;%
  filter(Geneid %in% Top14K_ExpressedGenes) %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% prcomp(scale=T)

summary(pca.results)</code></pre>
<pre><code>Importance of components:
                          PC1      PC2      PC3     PC4      PC5      PC6
Standard deviation     90.959 35.44498 25.33393 18.9319 16.70679 13.71600
Proportion of Variance  0.591  0.08974  0.04584  0.0256  0.01994  0.01344
Cumulative Proportion   0.591  0.68071  0.72655  0.7521  0.77209  0.78553
                           PC7      PC8      PC9    PC10    PC11    PC12
Standard deviation     13.2829 11.50160 10.55358 9.41743 8.79989 8.34584
Proportion of Variance  0.0126  0.00945  0.00796 0.00633 0.00553 0.00498
Cumulative Proportion   0.7981  0.80758  0.81553 0.82187 0.82740 0.83238
                          PC13    PC14    PC15   PC16    PC17    PC18    PC19
Standard deviation     8.08116 7.48997 7.28621 7.0990 6.92833 6.66918 6.61082
Proportion of Variance 0.00466 0.00401 0.00379 0.0036 0.00343 0.00318 0.00312
Cumulative Proportion  0.83704 0.84105 0.84484 0.8484 0.85187 0.85505 0.85817
                          PC20    PC21    PC22    PC23    PC24    PC25    PC26
Standard deviation     6.51037 6.40642 6.33377 6.27073 6.12692 6.07714 5.94719
Proportion of Variance 0.00303 0.00293 0.00287 0.00281 0.00268 0.00264 0.00253
Cumulative Proportion  0.86119 0.86413 0.86699 0.86980 0.87248 0.87512 0.87765
                          PC27    PC28    PC29    PC30    PC31    PC32   PC33
Standard deviation     5.87452 5.82229 5.78341 5.75898 5.68407 5.60317 5.5522
Proportion of Variance 0.00247 0.00242 0.00239 0.00237 0.00231 0.00224 0.0022
Cumulative Proportion  0.88011 0.88253 0.88492 0.88729 0.88960 0.89184 0.8940
                          PC34    PC35    PC36    PC37    PC38    PC39    PC40
Standard deviation     5.53457 5.40381 5.40268 5.35630 5.33855 5.27676 5.24385
Proportion of Variance 0.00219 0.00209 0.00208 0.00205 0.00204 0.00199 0.00196
Cumulative Proportion  0.89623 0.89832 0.90040 0.90245 0.90449 0.90647 0.90844
                          PC41    PC42    PC43    PC44    PC45    PC46    PC47
Standard deviation     5.21234 5.20952 5.18056 5.13364 5.12455 5.08510 5.05057
Proportion of Variance 0.00194 0.00194 0.00192 0.00188 0.00188 0.00185 0.00182
Cumulative Proportion  0.91038 0.91232 0.91424 0.91612 0.91799 0.91984 0.92166
                          PC48    PC49    PC50    PC51    PC52    PC53    PC54
Standard deviation     5.02761 4.96764 4.91994 4.90446 4.85925 4.81311 4.80058
Proportion of Variance 0.00181 0.00176 0.00173 0.00172 0.00169 0.00165 0.00165
Cumulative Proportion  0.92347 0.92523 0.92696 0.92868 0.93036 0.93202 0.93367
                          PC55    PC56    PC57    PC58    PC59    PC60    PC61
Standard deviation     4.78097 4.76876 4.74355 4.70452 4.67266 4.66706 4.64767
Proportion of Variance 0.00163 0.00162 0.00161 0.00158 0.00156 0.00156 0.00154
Cumulative Proportion  0.93530 0.93692 0.93853 0.94011 0.94167 0.94323 0.94477
                          PC62    PC63    PC64    PC65    PC66    PC67    PC68
Standard deviation     4.63563 4.59928 4.56367 4.54641 4.53988 4.52220 4.49649
Proportion of Variance 0.00153 0.00151 0.00149 0.00148 0.00147 0.00146 0.00144
Cumulative Proportion  0.94630 0.94781 0.94930 0.95078 0.95225 0.95371 0.95516
                          PC69    PC70    PC71   PC72    PC73    PC74    PC75
Standard deviation     4.48580 4.46001 4.43564 4.4250 4.39833 4.35304 4.33560
Proportion of Variance 0.00144 0.00142 0.00141 0.0014 0.00138 0.00135 0.00134
Cumulative Proportion  0.95659 0.95801 0.95942 0.9608 0.96220 0.96355 0.96490
                          PC76   PC77   PC78    PC79    PC80    PC81    PC82
Standard deviation     4.28467 4.2674 4.2613 4.23460 4.21085 4.19227 4.17845
Proportion of Variance 0.00131 0.0013 0.0013 0.00128 0.00127 0.00126 0.00125
Cumulative Proportion  0.96621 0.9675 0.9688 0.97009 0.97135 0.97261 0.97385
                          PC83    PC84   PC85    PC86    PC87    PC88    PC89
Standard deviation     4.14368 4.14020 4.1010 4.07135 4.05284 3.99964 3.95561
Proportion of Variance 0.00123 0.00122 0.0012 0.00118 0.00117 0.00114 0.00112
Cumulative Proportion  0.97508 0.97631 0.9775 0.97869 0.97986 0.98101 0.98212
                          PC90    PC91    PC92    PC93    PC94    PC95    PC96
Standard deviation     3.93752 3.90476 3.89645 3.86114 3.84045 3.82990 3.81073
Proportion of Variance 0.00111 0.00109 0.00108 0.00106 0.00105 0.00105 0.00104
Cumulative Proportion  0.98323 0.98432 0.98541 0.98647 0.98752 0.98857 0.98961
                          PC97    PC98    PC99   PC100  PC101   PC102   PC103
Standard deviation     3.75910 3.71712 3.68479 3.61557 3.5502 3.51461 3.48465
Proportion of Variance 0.00101 0.00099 0.00097 0.00093 0.0009 0.00088 0.00087
Cumulative Proportion  0.99062 0.99160 0.99257 0.99351 0.9944 0.99529 0.99616
                         PC104   PC105   PC106   PC107   PC108     PC109
Standard deviation     3.44532 3.39627 3.26300 3.17868 3.10257 4.622e-14
Proportion of Variance 0.00085 0.00082 0.00076 0.00072 0.00069 0.000e+00
Cumulative Proportion  0.99701 0.99783 0.99859 0.99931 1.00000 1.000e+00</code></pre>
<pre class="r"><code>pca.results.to.plot &lt;- pca.results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;old.sample.name&quot;) %&gt;%
  dplyr::select(old.sample.name, PC1:PC6) %&gt;%
  left_join(FullMetadata, by=&quot;old.sample.name&quot;)

ggplot(pca.results.to.plot, aes(x=PC1, y=PC2, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using top 14K genes\nExpOf52&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-5-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC3, y=PC4, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using top 14K genes\nExpOf52&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-2">
Past versions of unnamed-chunk-5-2.png
</button>
</p>
<div id="fig-unnamed-chunk-5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-5-2.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC5, y=PC6, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using top 14K genes\nExpOf52&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-3">
Past versions of unnamed-chunk-5-3.png
</button>
</p>
<div id="fig-unnamed-chunk-5-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-5-3.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, even though I am looking at only the first 6 PCs, it seems pretty
clear that the DMSO samples don’t cleanly cluster together. Perhaps some
of these molecules weren’t very active.</p>
<p>Now just for curiosity, let’s just include the fibroblast and the
dose titration experiment.</p>
<pre class="r"><code>SamplesToInclude &lt;- FullMetadata %&gt;%
  filter(Experiment %in% c(&quot;Dose response titration&quot;, &quot;Single high dose fibroblast&quot;)) %&gt;%
  pull(old.sample.name)

CPM &lt;- gene.counts %&gt;%
  cpm(log=T, prior.count=T) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Geneid&quot;) %&gt;%
  dplyr::select(Geneid, all_of(SamplesToInclude))

Top14K_ExpressedGenes &lt;- (CPM %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  apply(1, mean) %&gt;%
  sort(decreasing=T))[1:14000] %&gt;%
  names()

pca.results &lt;- CPM %&gt;%
  filter(Geneid %in% Top14K_ExpressedGenes) %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% prcomp(scale=T)

summary(pca.results)</code></pre>
<pre><code>Importance of components:
                           PC1     PC2      PC3      PC4      PC5      PC6
Standard deviation     95.5166 43.8246 29.44168 20.92836 16.18447 14.81627
Proportion of Variance  0.6517  0.1372  0.06192  0.03129  0.01871  0.01568
Cumulative Proportion   0.6517  0.7889  0.85077  0.88206  0.90077  0.91645
                            PC7      PC8     PC9    PC10    PC11    PC12   PC13
Standard deviation     12.69624 10.41019 9.97714 8.53642 8.01215 6.54003 6.1490
Proportion of Variance  0.01151  0.00774 0.00711 0.00521 0.00459 0.00306 0.0027
Cumulative Proportion   0.92796  0.93570 0.94281 0.94802 0.95260 0.95566 0.9584
                          PC14    PC15    PC16    PC17    PC18    PC19    PC20
Standard deviation     5.81417 5.60050 5.26490 5.17174 5.10546 5.04332 4.95929
Proportion of Variance 0.00241 0.00224 0.00198 0.00191 0.00186 0.00182 0.00176
Cumulative Proportion  0.96078 0.96302 0.96500 0.96691 0.96877 0.97058 0.97234
                          PC21    PC22    PC23    PC24    PC25    PC26    PC27
Standard deviation     4.92047 4.86882 4.79843 4.69269 4.66221 4.60350 4.56525
Proportion of Variance 0.00173 0.00169 0.00164 0.00157 0.00155 0.00151 0.00149
Cumulative Proportion  0.97407 0.97576 0.97741 0.97898 0.98053 0.98205 0.98354
                          PC28    PC29    PC30   PC31    PC32   PC33    PC34
Standard deviation     4.38148 4.37470 4.29248 4.2711 4.13309 4.1052 4.06510
Proportion of Variance 0.00137 0.00137 0.00132 0.0013 0.00122 0.0012 0.00118
Cumulative Proportion  0.98491 0.98627 0.98759 0.9889 0.99011 0.9913 0.99250
                          PC35    PC36    PC37    PC38    PC39    PC40    PC41
Standard deviation     4.04592 3.98561 3.94879 3.89471 3.83228 3.76342 3.62769
Proportion of Variance 0.00117 0.00113 0.00111 0.00108 0.00105 0.00101 0.00094
Cumulative Proportion  0.99367 0.99480 0.99592 0.99700 0.99805 0.99906 1.00000
                            PC42
Standard deviation     4.242e-14
Proportion of Variance 0.000e+00
Cumulative Proportion  1.000e+00</code></pre>
<pre class="r"><code>pca.results.to.plot &lt;- pca.results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;old.sample.name&quot;) %&gt;%
  dplyr::select(old.sample.name, PC1:PC6) %&gt;%
  left_join(FullMetadata, by=&quot;old.sample.name&quot;)

ggplot(pca.results.to.plot, aes(x=PC1, y=PC2, color=color, shape=Experiment)) +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using top 14K genes\nTitrationExp and FibroblastExp&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-6-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC3, y=PC4, color=color, shape=Experiment)) +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using top 14K genes\nTitrationExp and FibroblastExp&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-2">
Past versions of unnamed-chunk-6-2.png
</button>
</p>
<div id="fig-unnamed-chunk-6-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-6-2.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC5, y=PC6, color=color, shape=Experiment)) +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using top 14K genes\nTitrationExp and FibroblastExp&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-3">
Past versions of unnamed-chunk-6-3.png
</button>
</p>
<div id="fig-unnamed-chunk-6-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-6-3.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Wow, so in that, the second PC seems to pretty obviously reflect
dose. I am quite surprised we don’t see something so obvious with the
new samples… Somewhat concerning. Of course, perhaps it’s not too
surprising considering this experiment just has so many more samples,
only a few of which are DMSO, that maybe I shouldn’t expect some dose
effect in the first few PCs… One other thing I have done that was
helpful last time was to perform PCA on the titration series then
project new samples in the same PC space. I’ll try that later…</p>
<p>Now Let’s check for some splicing effects. Maybe that will be more
illuminating…</p>
</div>
<div id="verify-genome-wide-ga-gt-splice-site-induction"
class="section level3">
<h3>Verify genome-wide GA-GT splice site induction</h3>
<p>Let’s start by checking for global enrichment of GA|GT juncs… If this
doesn’t show an effect I’ll be very skeptical about the usefulness of
this data.</p>
<pre class="r"><code>leafcutter.sig &lt;- Sys.glob(&quot;../code/SplicingAnalysis/leafcutter/differential_splicing/ExpOf52_*_cluster_significance.txt&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/SplicingAnalysis/leafcutter/differential_splicing/ExpOf52_(.+?)_cluster_significance.txt&quot;, &quot;W\\1&quot;)) %&gt;%
  lapply(fread) %&gt;%
  bind_rows(.id=&quot;treatment&quot;)

leafcutter.effects &lt;- Sys.glob(&quot;../code/SplicingAnalysis/leafcutter/differential_splicing/ExpOf52_*_effect_sizes.txt&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/SplicingAnalysis/leafcutter/differential_splicing/ExpOf52_(.+?)_effect_sizes.txt&quot;, &quot;W\\1&quot;)) %&gt;%
  lapply(fread, col.names=c(&quot;intron&quot;, &quot;logef&quot;, &quot;treatment_PSI&quot;, &quot;DMSO_PSI&quot;, &quot;deltapsi&quot;)) %&gt;%
  bind_rows(.id=&quot;treatment&quot;) %&gt;%
  mutate(name = str_replace(intron, &quot;(.+?):(.+?):(.+?):clu.+([+-])&quot;, &quot;\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(deltapsi = deltapsi * -1, logef=logef*-1) %&gt;%
  mutate(cluster = str_replace(intron, &quot;(.+?):.+?:.+?:(clu.+[+-])&quot;, &quot;\\1:\\2&quot;))


Introns &lt;- read_tsv(&quot;../code/SplicingAnalysis/FullSpliceSiteAnnotations/JuncfilesMerged.annotated.basic.bed.5ss.tab.gz&quot;, col_names = c(&quot;name&quot;, &quot;seq&quot;, &quot;score&quot;)) %&gt;%
  separate(name, into=c(&quot;name&quot;, &quot;pos&quot;), sep = &quot;::&quot;)

Introns.annotations &lt;- read_tsv(&quot;../code/SplicingAnalysis/FullSpliceSiteAnnotations/JuncfilesMerged.annotated.basic.bed.gz&quot;) %&gt;%
  mutate(name = paste(chrom, start, end, strand, sep = &quot;_&quot;)) %&gt;%
  left_join(Introns, by=&quot;name&quot;)


leafcutter.effects %&gt;%
  inner_join(
    Introns.annotations %&gt;%
    mutate(IsGAGT = if_else(str_detect(seq, &quot;^\\w\\wGAGT&quot;), &quot;GA-GT&quot;, &quot;not GA-GT&quot;)),
    by=&quot;name&quot;) %&gt;%
  # count(IsGAGT, treatment)
  ggplot(aes(x=logef, group=treatment)) +
  stat_ecdf() +
  coord_cartesian(xlim=c(-2,2)) +
  facet_wrap(~IsGAGT) +
  theme_bw() +
  labs(title=&quot;Distribution of log(FC) for all tested introns\nExpOf52&quot;, y=&quot;ecdf&quot;, x=&quot;Splicing log(FC)&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-7-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok so I see some subtle general up-regulation of GA-GT introns…
That’s reassuring. But the degree to which all samples have about the
same general effect size is surprising to me… I expected some of the
compounds to be less active… Even though we normalized concentration
based on SMN2 minigene EC90, I suspected a bit of noise in that estimate
anyway. Let’s compare this to the analogous plot from the previous
fibroblast dataset where I also did differential splicing with
leafcutter to estimate splicing log(FC)…</p>
<pre class="r"><code>leafcutter.effects.fibroblasts &lt;- list.files(path=&quot;../code/SplicingAnalysis/leafcutter/differential_splicing&quot;, pattern=&quot;^[A-Z]+[0-9]*_effect_sizes.txt&quot;, full.names=T) %&gt;%
  setNames(str_replace(., &quot;../code/SplicingAnalysis/leafcutter/differential_splicing/(.+?)_effect_sizes.txt&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(fread, col.names=c(&quot;intron&quot;, &quot;logef&quot;, &quot;treatment_PSI&quot;, &quot;DMSO_PSI&quot;, &quot;deltapsi&quot;)) %&gt;%
  bind_rows(.id=&quot;treatment&quot;) %&gt;%
  mutate(name = str_replace(intron, &quot;(.+?):(.+?):(.+?):clu.+([+-])&quot;, &quot;\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(deltapsi = deltapsi * -1, logef=logef*-1) %&gt;%
  mutate(cluster = str_replace(intron, &quot;(.+?):.+?:.+?:(clu.+[+-])&quot;, &quot;\\1:\\2&quot;))

leafcutter.sig.fibroblasts &lt;-list.files(path=&quot;../code/SplicingAnalysis/leafcutter/differential_splicing&quot;, pattern=&quot;^[A-Z]+[0-9]*_cluster_significance.txt&quot;, full.names=T) %&gt;%
  setNames(str_replace(., &quot;../code/SplicingAnalysis/leafcutter/differential_splicing/(.+?)_cluster_significance.txt&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(fread) %&gt;%
  bind_rows(.id=&quot;treatment&quot;)


Fibroblast.colors &lt;- FullMetadata %&gt;%
  filter(cell.type==&quot;Fibroblast&quot;) %&gt;%
  distinct(treatment, .keep_all=T) %&gt;%
  dplyr::select(treatment, color) %&gt;%
  deframe()

leafcutter.effects.fibroblasts %&gt;%
  inner_join(
    Introns.annotations %&gt;%
    mutate(IsGAGT = if_else(str_detect(seq, &quot;^\\w\\wGAGT&quot;), &quot;GA-GT&quot;, &quot;not GA-GT&quot;)),
    by=&quot;name&quot;) %&gt;%
  # count(IsGAGT, treatment)
  ggplot(aes(x=logef, color=treatment)) +
  stat_ecdf() +
  scale_color_manual(values=Fibroblast.colors) +
  coord_cartesian(xlim=c(-2,2)) +
  facet_wrap(~IsGAGT) +
  theme_bw() +
  labs(title=&quot;Distribution of log(FC) for all tested introns\nFibroblast Exp&quot;, y=&quot;ecdf&quot;, x=&quot;Splicing log(FC)&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-8-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, so maybe the degree of these 52 treatments are roughly on the
order of the CUOME treatment in fibroblast data. Not too strong. I’m
still not totally sure this is even real…</p>
<p>Ok, so now let’s doing PCA just using GA|GT juncs… I think this
should really work. Let’s start by including all samples. If I don’t
really see DMSO control pop out from the rest, I will be a bit
worried.</p>
</div>
<div id="splicing-correlation-matrix-and-pca" class="section level3">
<h3>Splicing correlation matrix and PCA</h3>
<p>Actually before doing PCA, to get a broad overview of results, let’s
calculate the correlation coefficient across samples based on splicing.
For sake of quickly getting an overview, and picking reasonable GA-GT
introns (that aren’t too noisy to measure) let’s just consider the
500ish GA-GT introns I previously modelled. These are a pretty ‘good’
set of differentially spliced GA-GT introns based on
risdiplam,branaplam,C2C5…</p>
<pre class="r"><code>#Read in splice junction count table

PSI &lt;- fread(&quot;../code/SplicingAnalysis/leafcutter_all_samples/leafcutter_perind_numers.bed.gz&quot;, sep = &#39;\t&#39; ) %&gt;%
  dplyr::select(-&quot;NewMolecule.C04.2&quot;)

Modelled.introns &lt;- read_tsv(&quot;../output/EC50Estimtes.FromPSI.txt.gz&quot;) %&gt;%
  dplyr::select(`#Chrom`, start, end, strand=strand.y)


colnames(PSI)</code></pre>
<pre><code>  [1] &quot;#Chrom&quot;             &quot;start&quot;              &quot;end&quot;               
  [4] &quot;junc&quot;               &quot;gid&quot;                &quot;strand&quot;            
  [7] &quot;A10.1&quot;              &quot;A10.2&quot;              &quot;A10.3&quot;             
 [10] &quot;C2.1&quot;               &quot;C2.2&quot;               &quot;C2.3&quot;              
 [13] &quot;CUOME.1&quot;            &quot;CUOME.2&quot;            &quot;CUOME.3&quot;           
 [16] &quot;DMSO.1&quot;             &quot;DMSO.2&quot;             &quot;DMSO.3&quot;            
 [19] &quot;SM2.1&quot;              &quot;SM2.2&quot;              &quot;SM2.3&quot;             
 [22] &quot;TitrationExpBran.1&quot; &quot;TitrationExpBran.2&quot; &quot;TitrationExpBran.3&quot;
 [25] &quot;TitrationExpBran.4&quot; &quot;TitrationExpBran.5&quot; &quot;TitrationExpBran.6&quot;
 [28] &quot;TitrationExpBran.7&quot; &quot;TitrationExpBran.8&quot; &quot;TitrationExpC2C5.1&quot;
 [31] &quot;TitrationExpC2C5.2&quot; &quot;TitrationExpC2C5.3&quot; &quot;TitrationExpC2C5.4&quot;
 [34] &quot;TitrationExpC2C5.5&quot; &quot;TitrationExpC2C5.6&quot; &quot;TitrationExpC2C5.7&quot;
 [37] &quot;TitrationExpC2C5.8&quot; &quot;TitrationExpDMSO.1&quot; &quot;TitrationExpDMSO.2&quot;
 [40] &quot;TitrationExpDMSO.3&quot; &quot;TitrationExpRis.1&quot;  &quot;TitrationExpRis.2&quot; 
 [43] &quot;TitrationExpRis.3&quot;  &quot;TitrationExpRis.4&quot;  &quot;TitrationExpRis.5&quot; 
 [46] &quot;TitrationExpRis.6&quot;  &quot;TitrationExpRis.7&quot;  &quot;TitrationExpRis.8&quot; 
 [49] &quot;chRNA_1&quot;            &quot;chRNA_2&quot;            &quot;chRNA_3&quot;           
 [52] &quot;chRNA_4&quot;            &quot;chRNA_5&quot;            &quot;chRNA_6&quot;           
 [55] &quot;chRNA_7&quot;            &quot;chRNA_8&quot;            &quot;chRNA_9&quot;           
 [58] &quot;chRNA_10&quot;           &quot;chRNA_11&quot;           &quot;chRNA_12&quot;          
 [61] &quot;chRNA_13&quot;           &quot;chRNA_14&quot;           &quot;chRNA_15&quot;          
 [64] &quot;chRNA_16&quot;           &quot;chRNA_17&quot;           &quot;chRNA_18&quot;          
 [67] &quot;chRNA_19&quot;           &quot;chRNA_20&quot;           &quot;chRNA_21&quot;          
 [70] &quot;NewMolecule.A01.1&quot;  &quot;NewMolecule.A02.1&quot;  &quot;NewMolecule.A03.1&quot; 
 [73] &quot;NewMolecule.A04.1&quot;  &quot;NewMolecule.A05.1&quot;  &quot;NewMolecule.A06.1&quot; 
 [76] &quot;NewMolecule.A07.1&quot;  &quot;NewMolecule.A08.1&quot;  &quot;NewMolecule.A09.1&quot; 
 [79] &quot;NewMolecule.A10.1&quot;  &quot;NewMolecule.A11.1&quot;  &quot;NewMolecule.A12.1&quot; 
 [82] &quot;NewMolecule.B01.1&quot;  &quot;NewMolecule.B02.1&quot;  &quot;NewMolecule.B03.1&quot; 
 [85] &quot;NewMolecule.B04.1&quot;  &quot;NewMolecule.B05.1&quot;  &quot;NewMolecule.B06.1&quot; 
 [88] &quot;NewMolecule.B07.1&quot;  &quot;NewMolecule.B08.1&quot;  &quot;NewMolecule.B09.1&quot; 
 [91] &quot;NewMolecule.B10.1&quot;  &quot;NewMolecule.B11.1&quot;  &quot;NewMolecule.B12.1&quot; 
 [94] &quot;NewMolecule.C01.1&quot;  &quot;NewMolecule.C02.1&quot;  &quot;NewMolecule.C03.1&quot; 
 [97] &quot;NewMolecule.C04.1&quot;  &quot;NewMolecule.C05.1&quot;  &quot;NewMolecule.C06.1&quot; 
[100] &quot;NewMolecule.C07.1&quot;  &quot;NewMolecule.C08.1&quot;  &quot;NewMolecule.C09.1&quot; 
[103] &quot;NewMolecule.C10.1&quot;  &quot;NewMolecule.C11.1&quot;  &quot;NewMolecule.C12.1&quot; 
[106] &quot;NewMolecule.D01.1&quot;  &quot;NewMolecule.D02.1&quot;  &quot;NewMolecule.D03.1&quot; 
[109] &quot;NewMolecule.D04.1&quot;  &quot;NewMolecule.D05.1&quot;  &quot;NewMolecule.D06.1&quot; 
[112] &quot;NewMolecule.D07.1&quot;  &quot;NewMolecule.D08.1&quot;  &quot;NewMolecule.D09.1&quot; 
[115] &quot;NewMolecule.D10.1&quot;  &quot;NewMolecule.D11.1&quot;  &quot;NewMolecule.D12.1&quot; 
[118] &quot;NewMolecule.E01.1&quot;  &quot;NewMolecule.E02.1&quot;  &quot;NewMolecule.E03.1&quot; 
[121] &quot;NewMolecule.E04.1&quot;  &quot;NewMolecule.E05.1&quot;  &quot;NewMolecule.E06.2&quot; 
[124] &quot;NewMolecule.E07.3&quot;  &quot;NewMolecule.A01.2&quot;  &quot;NewMolecule.A02.2&quot; 
[127] &quot;NewMolecule.A03.2&quot;  &quot;NewMolecule.A04.2&quot;  &quot;NewMolecule.A05.2&quot; 
[130] &quot;NewMolecule.A06.2&quot;  &quot;NewMolecule.A07.2&quot;  &quot;NewMolecule.A08.2&quot; 
[133] &quot;NewMolecule.A09.2&quot;  &quot;NewMolecule.A10.2&quot;  &quot;NewMolecule.A11.2&quot; 
[136] &quot;NewMolecule.A12.2&quot;  &quot;NewMolecule.B01.2&quot;  &quot;NewMolecule.B02.2&quot; 
[139] &quot;NewMolecule.B03.2&quot;  &quot;NewMolecule.B04.2&quot;  &quot;NewMolecule.B05.2&quot; 
[142] &quot;NewMolecule.B06.2&quot;  &quot;NewMolecule.B07.2&quot;  &quot;NewMolecule.B08.2&quot; 
[145] &quot;NewMolecule.B09.2&quot;  &quot;NewMolecule.B10.2&quot;  &quot;NewMolecule.B11.2&quot; 
[148] &quot;NewMolecule.B12.2&quot;  &quot;NewMolecule.C01.2&quot;  &quot;NewMolecule.C02.2&quot; 
[151] &quot;NewMolecule.C03.2&quot;  &quot;NewMolecule.C05.2&quot;  &quot;NewMolecule.C06.2&quot; 
[154] &quot;NewMolecule.C07.2&quot;  &quot;NewMolecule.C08.2&quot;  &quot;NewMolecule.C09.2&quot; 
[157] &quot;NewMolecule.C10.2&quot;  &quot;NewMolecule.C11.2&quot;  &quot;NewMolecule.C12.2&quot; 
[160] &quot;NewMolecule.D01.2&quot;  &quot;NewMolecule.D02.2&quot;  &quot;NewMolecule.D03.2&quot; 
[163] &quot;NewMolecule.D04.2&quot;  &quot;NewMolecule.D05.2&quot;  &quot;NewMolecule.D06.2&quot; 
[166] &quot;NewMolecule.D07.2&quot;  &quot;NewMolecule.D08.2&quot;  &quot;NewMolecule.D09.2&quot; 
[169] &quot;NewMolecule.D10.2&quot;  &quot;NewMolecule.D11.2&quot;  &quot;NewMolecule.D12.2&quot; 
[172] &quot;NewMolecule.E01.2&quot;  &quot;NewMolecule.E02.2&quot;  &quot;NewMolecule.E03.2&quot; 
[175] &quot;NewMolecule.E04.2&quot;  &quot;NewMolecule.E05.4&quot;  &quot;NewMolecule.E06.5&quot; 
[178] &quot;NewMolecule.E07.6&quot; </code></pre>
<pre class="r"><code>PSI.GAGT.Only &lt;- PSI %&gt;%
  inner_join(Modelled.introns) %&gt;%
  dplyr::select(junc, A10.1:NewMolecule.E07.6) %&gt;%
  drop_na()


PSI.GAGT.Only.cormat &lt;- PSI.GAGT.Only %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  cor(method=&#39;s&#39;)

heatmap.2(PSI.GAGT.Only.cormat, trace=&#39;none&#39;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-9-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>And now let’s do PCA based on these introns…</p>
<pre class="r"><code>pca.results &lt;- PSI.GAGT.Only %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% prcomp(scale=T)

summary(pca.results)</code></pre>
<pre><code>Importance of components:
                           PC1     PC2     PC3     PC4     PC5     PC6     PC7
Standard deviation     17.8068 10.8573 8.17051 7.30330 5.95972 5.17196 4.98289
Proportion of Variance  0.2909  0.1081 0.06125 0.04893 0.03259 0.02454 0.02278
Cumulative Proportion   0.2909  0.3991 0.46029 0.50923 0.54181 0.56635 0.58913
                           PC8     PC9    PC10    PC11    PC12   PC13    PC14
Standard deviation     4.65906 4.15658 4.07768 3.91171 3.76570 3.7066 3.51788
Proportion of Variance 0.01991 0.01585 0.01525 0.01404 0.01301 0.0126 0.01135
Cumulative Proportion  0.60905 0.62490 0.64015 0.65419 0.66720 0.6798 0.69116
                          PC15    PC16    PC17    PC18    PC19    PC20    PC21
Standard deviation     3.35281 3.25740 3.16906 3.12639 3.09122 3.02097 3.00494
Proportion of Variance 0.01031 0.00973 0.00921 0.00897 0.00877 0.00837 0.00828
Cumulative Proportion  0.70147 0.71121 0.72042 0.72939 0.73815 0.74653 0.75481
                          PC22    PC23    PC24    PC25    PC26    PC27    PC28
Standard deviation     2.91815 2.83474 2.79655 2.72807 2.67897 2.57623 2.56316
Proportion of Variance 0.00781 0.00737 0.00717 0.00683 0.00658 0.00609 0.00603
Cumulative Proportion  0.76262 0.77000 0.77717 0.78400 0.79058 0.79667 0.80270
                         PC29    PC30    PC31    PC32    PC33    PC34    PC35
Standard deviation     2.4933 2.42762 2.38281 2.32648 2.22686 2.14178 2.12131
Proportion of Variance 0.0057 0.00541 0.00521 0.00497 0.00455 0.00421 0.00413
Cumulative Proportion  0.8084 0.81381 0.81902 0.82398 0.82853 0.83274 0.83687
                          PC36    PC37    PC38    PC39    PC40    PC41   PC42
Standard deviation     2.11105 2.07792 2.04445 1.99915 1.96822 1.95962 1.8972
Proportion of Variance 0.00409 0.00396 0.00383 0.00367 0.00355 0.00352 0.0033
Cumulative Proportion  0.84096 0.84492 0.84876 0.85242 0.85598 0.85950 0.8628
                          PC43   PC44    PC45   PC46    PC47    PC48    PC49
Standard deviation     1.87969 1.8689 1.82108 1.8094 1.77040 1.72243 1.70644
Proportion of Variance 0.00324 0.0032 0.00304 0.0030 0.00288 0.00272 0.00267
Cumulative Proportion  0.86604 0.8692 0.87229 0.8753 0.87817 0.88089 0.88356
                          PC50    PC51    PC52    PC53    PC54    PC55   PC56
Standard deviation     1.70530 1.65973 1.63623 1.60037 1.59820 1.58595 1.5491
Proportion of Variance 0.00267 0.00253 0.00246 0.00235 0.00234 0.00231 0.0022
Cumulative Proportion  0.88623 0.88876 0.89121 0.89356 0.89591 0.89821 0.9004
                          PC57    PC58    PC59    PC60    PC61   PC62    PC63
Standard deviation     1.52812 1.50637 1.49055 1.46312 1.44493 1.4382 1.41790
Proportion of Variance 0.00214 0.00208 0.00204 0.00196 0.00192 0.0019 0.00184
Cumulative Proportion  0.90256 0.90464 0.90668 0.90864 0.91056 0.9124 0.91430
                          PC64    PC65   PC66    PC67    PC68    PC69    PC70
Standard deviation     1.40272 1.38000 1.3627 1.35727 1.33489 1.32463 1.30675
Proportion of Variance 0.00181 0.00175 0.0017 0.00169 0.00163 0.00161 0.00157
Cumulative Proportion  0.91610 0.91785 0.9196 0.92125 0.92288 0.92449 0.92606
                          PC71    PC72    PC73    PC74    PC75    PC76    PC77
Standard deviation     1.29953 1.29335 1.28311 1.26960 1.25385 1.24148 1.23204
Proportion of Variance 0.00155 0.00153 0.00151 0.00148 0.00144 0.00141 0.00139
Cumulative Proportion  0.92761 0.92914 0.93065 0.93213 0.93357 0.93499 0.93638
                          PC78    PC79    PC80   PC81    PC82    PC83    PC84
Standard deviation     1.21811 1.20481 1.20317 1.1894 1.17549 1.17356 1.15990
Proportion of Variance 0.00136 0.00133 0.00133 0.0013 0.00127 0.00126 0.00123
Cumulative Proportion  0.93774 0.93907 0.94040 0.9417 0.94297 0.94423 0.94546
                          PC85   PC86    PC87    PC88    PC89    PC90    PC91
Standard deviation     1.15034 1.1443 1.13021 1.12064 1.11255 1.10493 1.10202
Proportion of Variance 0.00121 0.0012 0.00117 0.00115 0.00114 0.00112 0.00111
Cumulative Proportion  0.94668 0.9479 0.94905 0.95020 0.95134 0.95246 0.95357
                          PC92    PC93    PC94    PC95    PC96    PC97    PC98
Standard deviation     1.09981 1.07942 1.06509 1.05582 1.05330 1.03539 1.02657
Proportion of Variance 0.00111 0.00107 0.00104 0.00102 0.00102 0.00098 0.00097
Cumulative Proportion  0.95468 0.95575 0.95679 0.95781 0.95883 0.95982 0.96078
                          PC99   PC100   PC101  PC102   PC103   PC104   PC105
Standard deviation     1.01193 1.01054 1.00006 0.9921 0.98595 0.98306 0.97870
Proportion of Variance 0.00094 0.00094 0.00092 0.0009 0.00089 0.00089 0.00088
Cumulative Proportion  0.96172 0.96266 0.96358 0.9645 0.96537 0.96626 0.96714
                         PC106   PC107   PC108  PC109  PC110   PC111   PC112
Standard deviation     0.96883 0.96234 0.95920 0.9351 0.9323 0.92778 0.91778
Proportion of Variance 0.00086 0.00085 0.00084 0.0008 0.0008 0.00079 0.00077
Cumulative Proportion  0.96800 0.96885 0.96969 0.9705 0.9713 0.97208 0.97285
                         PC113   PC114   PC115  PC116  PC117   PC118   PC119
Standard deviation     0.91312 0.89464 0.88541 0.8759 0.8726 0.86852 0.86330
Proportion of Variance 0.00076 0.00073 0.00072 0.0007 0.0007 0.00069 0.00068
Cumulative Proportion  0.97362 0.97435 0.97507 0.9758 0.9765 0.97717 0.97785
                         PC120   PC121   PC122   PC123   PC124   PC125   PC126
Standard deviation     0.85787 0.85264 0.84570 0.84076 0.82717 0.82139 0.82103
Proportion of Variance 0.00068 0.00067 0.00066 0.00065 0.00063 0.00062 0.00062
Cumulative Proportion  0.97853 0.97919 0.97985 0.98050 0.98112 0.98174 0.98236
                         PC127   PC128   PC129   PC130   PC131   PC132   PC133
Standard deviation     0.81732 0.80003 0.79463 0.79015 0.77880 0.77161 0.76729
Proportion of Variance 0.00061 0.00059 0.00058 0.00057 0.00056 0.00055 0.00054
Cumulative Proportion  0.98297 0.98356 0.98414 0.98471 0.98527 0.98582 0.98636
                         PC134   PC135   PC136   PC137   PC138   PC139   PC140
Standard deviation     0.75267 0.74685 0.74292 0.73295 0.72225 0.71640 0.71031
Proportion of Variance 0.00052 0.00051 0.00051 0.00049 0.00048 0.00047 0.00046
Cumulative Proportion  0.98688 0.98739 0.98789 0.98839 0.98887 0.98934 0.98980
                         PC141   PC142   PC143   PC144   PC145   PC146  PC147
Standard deviation     0.70834 0.69835 0.69170 0.68614 0.68132 0.66946 0.6641
Proportion of Variance 0.00046 0.00045 0.00044 0.00043 0.00043 0.00041 0.0004
Cumulative Proportion  0.99026 0.99071 0.99115 0.99158 0.99200 0.99242 0.9928
                        PC148   PC149   PC150   PC151   PC152   PC153   PC154
Standard deviation     0.6576 0.65180 0.64783 0.63299 0.62971 0.62458 0.61835
Proportion of Variance 0.0004 0.00039 0.00039 0.00037 0.00036 0.00036 0.00035
Cumulative Proportion  0.9932 0.99361 0.99399 0.99436 0.99472 0.99508 0.99543
                         PC155   PC156   PC157   PC158   PC159   PC160   PC161
Standard deviation     0.61050 0.60903 0.58801 0.58311 0.57708 0.56122 0.55556
Proportion of Variance 0.00034 0.00034 0.00032 0.00031 0.00031 0.00029 0.00028
Cumulative Proportion  0.99577 0.99611 0.99643 0.99674 0.99705 0.99734 0.99762
                         PC162   PC163   PC164   PC165   PC166   PC167   PC168
Standard deviation     0.54262 0.53499 0.53018 0.52517 0.51671 0.51438 0.50128
Proportion of Variance 0.00027 0.00026 0.00026 0.00025 0.00024 0.00024 0.00023
Cumulative Proportion  0.99789 0.99815 0.99841 0.99866 0.99891 0.99915 0.99938
                         PC169   PC170   PC171     PC172
Standard deviation     0.48872 0.48374 0.44743 1.373e-14
Proportion of Variance 0.00022 0.00021 0.00018 0.000e+00
Cumulative Proportion  0.99960 0.99982 1.00000 1.000e+00</code></pre>
<pre class="r"><code>pca.results.to.plot &lt;- pca.results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;leafcutter.name&quot;) %&gt;%
  dplyr::select(leafcutter.name, PC1:PC6) %&gt;%
  left_join(FullMetadata, by=&quot;leafcutter.name&quot;)

ggplot(pca.results.to.plot, aes(x=PC1, y=PC2, color=color, shape=Experiment)) +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 1090 GAGT introns\nAll samples&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-10-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC3, y=PC4, color=color, shape=Experiment)) +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 1090 GAGT introns\nAll samples&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-2">
Past versions of unnamed-chunk-10-2.png
</button>
</p>
<div id="fig-unnamed-chunk-10-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-10-2.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC5, y=PC6, color=color, shape=Experiment)) +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 1090 GAGT introns\nAll samples&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-10-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-3">
Past versions of unnamed-chunk-10-3.png
</button>
</p>
<div id="fig-unnamed-chunk-10-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-10-3.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>That is actually kind of reassuring. The first PC is definitely
related to dose. And all DMSO samples cluster together. However, again,
the effective dosage seems a bit weak compared to fibroblast
experiments… Looks like most of the samples are roughly equivalent to
~100nM risdiplam, so still perhaps on the high end of what would be
realistic clinical concentration but I think a slightly higher dose
would have been more informative. In this particular PCA, it looks like
PC3 sepearates branaplam vs risdiplam, and it’s not totally clear how
these molecules are different along that axis… I could try, as mentioned
before just doing PCA using the dose titration experiments then
projecting the new samples in that space… But first let’s do a PCA just
using the samples from the new experiment..</p>
<pre class="r"><code>SamplesToInclude &lt;- FullMetadata %&gt;%
  filter(Experiment %in% c(&quot;Single high dose LCL&quot;)) %&gt;%
  filter(!old.sample.name == &quot;NewMolecule.C04-2&quot;) %&gt;%
  pull(leafcutter.name)

pca.results &lt;- PSI.GAGT.Only %&gt;%
  dplyr::select(junc, all_of(SamplesToInclude)) %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% prcomp(scale=T)

summary(pca.results)</code></pre>
<pre><code>Importance of components:
                           PC1     PC2     PC3     PC4     PC5     PC6     PC7
Standard deviation     17.3652 8.02663 7.63797 4.89303 4.62299 4.47221 4.29379
Proportion of Variance  0.2767 0.05911 0.05352 0.02196 0.01961 0.01835 0.01691
Cumulative Proportion   0.2767 0.33576 0.38928 0.41125 0.43085 0.44920 0.46612
                          PC8     PC9    PC10    PC11    PC12    PC13    PC14
Standard deviation     3.9200 3.84117 3.69235 3.60719 3.54779 3.47655 3.44963
Proportion of Variance 0.0141 0.01354 0.01251 0.01194 0.01155 0.01109 0.01092
Cumulative Proportion  0.4802 0.49375 0.50626 0.51820 0.52974 0.54083 0.55175
                          PC15    PC16    PC17    PC18    PC19    PC20    PC21
Standard deviation     3.33808 3.32645 3.30802 3.26244 3.24349 3.17997 3.10580
Proportion of Variance 0.01022 0.01015 0.01004 0.00976 0.00965 0.00928 0.00885
Cumulative Proportion  0.56197 0.57212 0.58216 0.59193 0.60158 0.61086 0.61971
                          PC22    PC23    PC24    PC25    PC26   PC27    PC28
Standard deviation     3.04656 3.01811 2.97941 2.92856 2.91952 2.8791 2.87597
Proportion of Variance 0.00852 0.00836 0.00814 0.00787 0.00782 0.0076 0.00759
Cumulative Proportion  0.62822 0.63658 0.64472 0.65259 0.66041 0.6680 0.67560
                          PC29    PC30    PC31    PC32    PC33    PC34   PC35
Standard deviation     2.84331 2.80557 2.77632 2.74624 2.73980 2.69767 2.6825
Proportion of Variance 0.00742 0.00722 0.00707 0.00692 0.00689 0.00668 0.0066
Cumulative Proportion  0.68302 0.69024 0.69731 0.70423 0.71112 0.71780 0.7244
                          PC36    PC37   PC38    PC39   PC40    PC41    PC42
Standard deviation     2.67372 2.66036 2.6208 2.58876 2.5790 2.55420 2.51670
Proportion of Variance 0.00656 0.00649 0.0063 0.00615 0.0061 0.00599 0.00581
Cumulative Proportion  0.73096 0.73745 0.7438 0.74990 0.7560 0.76199 0.76780
                          PC43    PC44    PC45    PC46   PC47    PC48    PC49
Standard deviation     2.50155 2.48335 2.48004 2.45997 2.4489 2.42392 2.41933
Proportion of Variance 0.00574 0.00566 0.00564 0.00555 0.0055 0.00539 0.00537
Cumulative Proportion  0.77354 0.77920 0.78484 0.79039 0.7959 0.80128 0.80665
                          PC50    PC51    PC52    PC53    PC54    PC55    PC56
Standard deviation     2.38644 2.36665 2.33817 2.31661 2.30298 2.29857 2.27662
Proportion of Variance 0.00522 0.00514 0.00502 0.00492 0.00487 0.00485 0.00476
Cumulative Proportion  0.81188 0.81702 0.82203 0.82695 0.83182 0.83667 0.84142
                          PC57    PC58    PC59   PC60    PC61    PC62    PC63
Standard deviation     2.26629 2.25224 2.22404 2.2140 2.17655 2.16350 2.15976
Proportion of Variance 0.00471 0.00465 0.00454 0.0045 0.00435 0.00429 0.00428
Cumulative Proportion  0.84613 0.85079 0.85533 0.8598 0.86417 0.86846 0.87274
                          PC64    PC65    PC66    PC67    PC68    PC69    PC70
Standard deviation     2.12927 2.10607 2.09074 2.07866 2.05863 2.05530 2.02185
Proportion of Variance 0.00416 0.00407 0.00401 0.00396 0.00389 0.00388 0.00375
Cumulative Proportion  0.87690 0.88097 0.88498 0.88895 0.89283 0.89671 0.90046
                          PC71    PC72   PC73    PC74    PC75    PC76    PC77
Standard deviation     2.00272 1.99787 1.9816 1.96398 1.94003 1.93185 1.92771
Proportion of Variance 0.00368 0.00366 0.0036 0.00354 0.00345 0.00342 0.00341
Cumulative Proportion  0.90414 0.90780 0.9114 0.91494 0.91840 0.92182 0.92523
                          PC78    PC79    PC80    PC81    PC82    PC83    PC84
Standard deviation     1.88660 1.86190 1.85577 1.83633 1.82108 1.80387 1.78728
Proportion of Variance 0.00327 0.00318 0.00316 0.00309 0.00304 0.00299 0.00293
Cumulative Proportion  0.92849 0.93168 0.93483 0.93793 0.94097 0.94396 0.94689
                          PC85    PC86    PC87    PC88    PC89    PC90    PC91
Standard deviation     1.77038 1.76254 1.73891 1.72961 1.69315 1.68904 1.67390
Proportion of Variance 0.00288 0.00285 0.00277 0.00274 0.00263 0.00262 0.00257
Cumulative Proportion  0.94976 0.95261 0.95539 0.95813 0.96076 0.96338 0.96595
                          PC92   PC93    PC94    PC95    PC96    PC97    PC98
Standard deviation     1.64617 1.6161 1.59282 1.58560 1.55184 1.52377 1.51716
Proportion of Variance 0.00249 0.0024 0.00233 0.00231 0.00221 0.00213 0.00211
Cumulative Proportion  0.96843 0.9708 0.97316 0.97547 0.97767 0.97980 0.98192
                          PC99   PC100   PC101   PC102   PC103   PC104   PC105
Standard deviation     1.49663 1.46361 1.46094 1.45131 1.43022 1.42266 1.39131
Proportion of Variance 0.00205 0.00197 0.00196 0.00193 0.00188 0.00186 0.00178
Cumulative Proportion  0.98397 0.98594 0.98789 0.98983 0.99170 0.99356 0.99534
                         PC106   PC107   PC108     PC109
Standard deviation     1.35016 1.32492 1.22667 3.724e-14
Proportion of Variance 0.00167 0.00161 0.00138 0.000e+00
Cumulative Proportion  0.99701 0.99862 1.00000 1.000e+00</code></pre>
<pre class="r"><code>pca.results.to.plot &lt;- pca.results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;leafcutter.name&quot;) %&gt;%
  dplyr::select(leafcutter.name, PC1:PC6) %&gt;%
  left_join(FullMetadata, by=&quot;leafcutter.name&quot;)

ggplot(pca.results.to.plot, aes(x=PC1, y=PC2, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 1090 GAGT introns\nExpOf52 samples&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-11-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC3, y=PC4, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 1090 GAGT introns\nExpOf52 samples&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-2">
Past versions of unnamed-chunk-11-2.png
</button>
</p>
<div id="fig-unnamed-chunk-11-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-11-2.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC5, y=PC6, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 1090 GAGT introns\nExpOf52 samples&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-11-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-3">
Past versions of unnamed-chunk-11-3.png
</button>
</p>
<div id="fig-unnamed-chunk-11-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-11-3.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, reassuringly, again DMSO samples stick out in the first PC, and I
see a few biological replicates are close together.</p>
<p>Ok, now I want to try projecting the old titration series data into
this same PCA spaace, which might help interpretability.</p>
<pre class="r"><code>SamplesToProject &lt;- FullMetadata %&gt;%
  filter(!Experiment %in% c(&quot;Single high dose LCL&quot;)) %&gt;%
  filter(!old.sample.name == &quot;NewMolecule.C04-2&quot;) %&gt;%
  pull(leafcutter.name)

pca.results.to.plot_AdditionProjections &lt;- PSI.GAGT.Only %&gt;%
  dplyr::select(junc, all_of(SamplesToProject)) %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% predict(pca.results, .) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;leafcutter.name&quot;) %&gt;%
  dplyr::select(leafcutter.name, PC1:PC6) %&gt;%
  left_join(FullMetadata, by=&quot;leafcutter.name&quot;)


TreatmentColorsForLabels.FibroblastColorsAdded &lt;-
FullMetadata %&gt;%
  group_by(treatment) %&gt;%
  filter(dose.nM == max(dose.nM) | treatment == &quot;DMSO&quot; | cell.type==&quot;Fibroblast&quot;) %&gt;%
  ungroup() %&gt;%
  distinct(treatment, .keep_all=T) %&gt;%
  arrange(dose.nM) %&gt;%
  mutate(vjust=row_number()*1.2)

TreatmentColorsLabels.Layer.FibroblastColorsAdded &lt;- geom_text(
    data = TreatmentColorsForLabels.FibroblastColorsAdded, 
    aes(label=treatment, color=color, vjust=vjust),
    y=Inf, x=Inf, hjust=1.05
  )

  
ggplot(pca.results.to.plot, aes(x=PC1, y=PC2, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  geom_point(data = pca.results.to.plot_AdditionProjections) +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 1090 GAGT introns\nExpOf52 samples&quot;, caption=&quot;Additional samples projected&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-12-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC3, y=PC4, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  geom_point(data = pca.results.to.plot_AdditionProjections) +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 1090 GAGT introns\nAll samples&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-2">
Past versions of unnamed-chunk-12-2.png
</button>
</p>
<div id="fig-unnamed-chunk-12-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-12-2.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>I think these PCA are interseting… it’s still hard to make sense of
it all… And this isn’t really the way to identify new axes of specifity
since the introns were selected precisely because of their effect in
risdiplam/branaplam/C2C5 in previous experiments… Better would be to
identify all differentially spliced introns and perform a PCA on that,
also projecting the old risdiplam/branaplam/C2C5 data in the PCA for
help with interpretation… Let’s work towards that….</p>
</div>
<div id="leafcutter-differential-splicing" class="section level3">
<h3>Leafcutter differential splicing</h3>
<p>First, let’s explore the leafcutter differential splicing results…
From previous plots, I suspect the effective dosage of these new data to
be much less than the fibroblast data, and perhaps we will get much
fewer significant differential splicing events… First Let’s check it
out, the number of differentially spliced clusters, some P value
histograms, etc.</p>
<pre class="r"><code>leafcutter.sig %&gt;%
  sample_n_of(12, treatment) %&gt;%
  filter(!is.na(p)) %&gt;%
  ggplot(aes(x=p)) +
    geom_histogram() +
    facet_wrap(~treatment, scales=&quot;free&quot;) +
  labs(caption=&quot;P value histograms for some random splicing contrasts&quot;, title=&quot;Exp of 52&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-13-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>…and compare that to the fibroblast tests…</p>
<pre class="r"><code>leafcutter.sig.fibroblasts %&gt;%
  filter(!is.na(p)) %&gt;%
  ggplot(aes(x=p)) +
    geom_histogram() +
    facet_wrap(~treatment, scales=&quot;free&quot;) +
  labs(caption=&quot;P value histograms for some random splicing contrasts&quot;, title=&quot;fibroblast experiment&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-14-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>ok, so a lot these will probably have even less significant hits than
the fibroblast CUOME contrast.</p>
<p>Let’s see how many hits at an FDR of 10%…</p>
<pre class="r"><code>bind_rows(
bind_rows(leafcutter.sig.fibroblasts, leafcutter.sig) %&gt;%
  filter(p.adjust &lt; 0.01) %&gt;%
  count(treatment) %&gt;%
  mutate(FDR = 0.01),
bind_rows(leafcutter.sig.fibroblasts, leafcutter.sig) %&gt;%
  filter(p.adjust &lt; 0.05) %&gt;%
  count(treatment) %&gt;%
  mutate(FDR = 0.05),
bind_rows(leafcutter.sig.fibroblasts, leafcutter.sig) %&gt;%
  filter(p.adjust &lt; 0.1) %&gt;%
  count(treatment) %&gt;%
  mutate(FDR = 0.1)) %&gt;%
  left_join(FullMetadata, by=&#39;treatment&#39;) %&gt;%
  ggplot(aes(x=treatment, y=n)) +
  geom_bar(stat=&quot;identity&quot;,position = &quot;identity&quot;, alpha=.15) +
  facet_grid(~Experiment, scales = &quot;free&quot;, space = &quot;free_x&quot;, labeller = label_wrap_gen(10)) +
  Rotate_x_labels +
  theme(axis.text.x = element_text(size=6)) +
  theme(strip.text.x = element_text(size = 5)) +
  labs(title=&quot;Number of leafcutter signigicant clusters&quot;, x=&quot;treatment&quot;, caption=&quot;FDR 1%, 5%, 10%&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-15-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>bind_rows(leafcutter.sig.fibroblasts, leafcutter.sig) %&gt;%
  filter(p.adjust &lt; 2) %&gt;%
  count(treatment) %&gt;%
  left_join(FullMetadata, by=&#39;treatment&#39;) %&gt;%
  ggplot(aes(x=treatment, y=n)) +
  geom_bar(stat=&quot;identity&quot;,position = &quot;identity&quot;, alpha=1) +
  facet_grid(~Experiment, scales = &quot;free&quot;, space = &quot;free_x&quot;, labeller = label_wrap_gen(10)) +
  Rotate_x_labels +
  theme(axis.text.x = element_text(size=6)) +
  theme(strip.text.x = element_text(size = 5)) +
  labs(title=&quot;Number of leafcutter tests&quot;, x=&quot;treatment&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-2">
Past versions of unnamed-chunk-15-2.png
</button>
</p>
<div id="fig-unnamed-chunk-15-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-15-2.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, most treatments have even less hits than the CUOME contrast. In
principle, sharing information across samples could help with this. I
think this might be important towards finding difference amongst samples
and identification of new specificities. Like mash could be useful. But
that takes a bit of time to implement… leafcutter doesn’t just output
standard errors, which I think are required. I suppose one hackish
solution would be to perform a contrast with all treatments combined, vs
DMSO… then identify significant clusters. To find more molecule-specific
effects, I could then look at the nominal P value from the non
aggregated treatment contrast.</p>
<p>Following this idea, I did a contrast with all molecules vs DMSO…
Here are the leafcutter results…</p>
<pre class="r"><code>leafcutter.effects.MergedContrast &lt;- read_tsv(&quot;../code/SplicingAnalysis/MergedExp52_Contrast/_effect_sizes.txt&quot;)
leafcutter.sig.MergedContrast &lt;- read_tsv(&quot;../code/SplicingAnalysis/MergedExp52_Contrast/_cluster_significance.txt&quot;)

bind_rows(
bind_rows(leafcutter.sig.fibroblasts, leafcutter.sig, leafcutter.sig.MergedContrast) %&gt;%
  filter(p.adjust &lt; 0.01) %&gt;%
  count(treatment) %&gt;%
  mutate(FDR = 0.01),
bind_rows(leafcutter.sig.fibroblasts, leafcutter.sig, leafcutter.sig.MergedContrast) %&gt;%
  filter(p.adjust &lt; 0.05) %&gt;%
  count(treatment) %&gt;%
  mutate(FDR = 0.05),
bind_rows(leafcutter.sig.fibroblasts, leafcutter.sig, leafcutter.sig.MergedContrast) %&gt;%
  filter(p.adjust &lt; 0.1) %&gt;%
  count(treatment) %&gt;%
  mutate(FDR = 0.1)) %&gt;%
  left_join(FullMetadata, by=&#39;treatment&#39;) %&gt;%
  replace_na(list(treatment=&quot;Merged&quot;, Experiment=&quot;Single high dose LCL&quot;)) %&gt;%
  ggplot(aes(x=treatment, y=n)) +
  geom_bar(stat=&quot;identity&quot;,position = &quot;identity&quot;, alpha=.15) +
  facet_grid(~Experiment, scales = &quot;free&quot;, space = &quot;free_x&quot;, labeller = label_wrap_gen(10)) +
  Rotate_x_labels +
  theme(axis.text.x = element_text(size=6)) +
  theme(strip.text.x = element_text(size = 5)) +
  labs(title=&quot;Number of leafcutter signigicant clusters&quot;, x=&quot;treatment&quot;, caption=&quot;FDR 1%, 5%, 10%&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-16-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>bind_rows(leafcutter.sig.fibroblasts, leafcutter.sig, leafcutter.sig.MergedContrast) %&gt;%
  filter(p.adjust &lt; 2) %&gt;%
  count(treatment) %&gt;%
  left_join(FullMetadata, by=&#39;treatment&#39;) %&gt;%
  replace_na(list(treatment=&quot;Merged&quot;, Experiment=&quot;Single high dose LCL&quot;)) %&gt;%
  ggplot(aes(x=treatment, y=n)) +
  geom_bar(stat=&quot;identity&quot;,position = &quot;identity&quot;, alpha=1) +
  facet_grid(~Experiment, scales = &quot;free&quot;, space = &quot;free_x&quot;, labeller = label_wrap_gen(10)) +
  Rotate_x_labels +
  theme(axis.text.x = element_text(size=6)) +
  theme(strip.text.x = element_text(size = 5)) +
  labs(title=&quot;Number of leafcutter tests&quot;, x=&quot;treatment&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-2">
Past versions of unnamed-chunk-16-2.png
</button>
</p>
<div id="fig-unnamed-chunk-16-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-16-2.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Wow. That did surprisingly little. I don’t have the time to
troubleshoot if there is a bug or if this is a legitamite result of
doing this merged contrast.</p>
<p>So now let’s try counting how many unique clusters there are amongst
all the attempted contrasts. Let’s be permissive and consider FDR
10%</p>
<pre class="r"><code>#Number of total hits (not necessarily unique)
leafcutter.sig %&gt;%
  filter(p.adjust &lt; 0.1) %&gt;% nrow()</code></pre>
<pre><code>[1] 32162</code></pre>
<pre class="r"><code>#Number of unique cluster hits amongst all hits
leafcutter.sig %&gt;%
  filter(p.adjust &lt; 0.1) %&gt;%
  distinct(cluster) %&gt;% nrow()</code></pre>
<pre><code>[1] 10803</code></pre>
<p>Ok, so a lot of the hits are unique to some contrasts, in the sense
that I get 10803/32162 <em>unique</em> clusters with FDR&lt;10%. I think
what I will do next is do the PCA, but just include introns that are
FDR&lt;10% in at least one contrast, and also consider some minimal
effect size, and also consider just GAGT introns within those
clusters.</p>
<pre class="r"><code>IntronsToInclude &lt;- leafcutter.sig %&gt;%
  filter(p.adjust &lt; 0.1) %&gt;%
  inner_join(leafcutter.effects, by=c(&quot;treatment&quot;, &quot;cluster&quot;)) %&gt;%
  filter(deltapsi &gt; 0.01) %&gt;%
  inner_join(Introns.annotations, by=&#39;name&#39;) %&gt;%
  filter(str_detect(seq, &quot;^\\w\\wGAGT&quot;)) %&gt;%
  pull(intron)



length(IntronsToInclude)</code></pre>
<pre><code>[1] 6383</code></pre>
<pre class="r"><code>PSI.IntronsToInclude &lt;- PSI %&gt;%
  filter(junc %in% IntronsToInclude) %&gt;%
  dplyr::select(junc, A10.1:NewMolecule.E07.6) %&gt;%
  drop_na()

dim(PSI.IntronsToInclude)</code></pre>
<pre><code>[1] 475 173</code></pre>
<p>Ok, so only 475 introns left… This probably won’t change anything
from my previous PCA with ~1000 GAGT introns… Well let’s follow through
anyway…</p>
<pre class="r"><code>SamplesToInclude &lt;- FullMetadata %&gt;%
  filter(Experiment %in% c(&quot;Single high dose LCL&quot;)) %&gt;%
  filter(!old.sample.name == &quot;NewMolecule.C04-2&quot;) %&gt;%
  pull(leafcutter.name)


pca.results &lt;- PSI.IntronsToInclude %&gt;%
  dplyr::select(junc, all_of(SamplesToInclude)) %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% prcomp(scale=T)

summary(pca.results)</code></pre>
<pre><code>Importance of components:
                           PC1     PC2     PC3     PC4     PC5     PC6     PC7
Standard deviation     11.6045 6.86645 5.16078 3.74247 3.39272 3.03012 2.81941
Proportion of Variance  0.2835 0.09926 0.05607 0.02949 0.02423 0.01933 0.01673
Cumulative Proportion   0.2835 0.38276 0.43883 0.46832 0.49255 0.51188 0.52862
                           PC8     PC9   PC10    PC11    PC12    PC13    PC14
Standard deviation     2.76866 2.66673 2.5319 2.46868 2.42788 2.38280 2.30959
Proportion of Variance 0.01614 0.01497 0.0135 0.01283 0.01241 0.01195 0.01123
Cumulative Proportion  0.54476 0.55973 0.5732 0.58605 0.59846 0.61042 0.62165
                          PC15    PC16    PC17   PC18   PC19    PC20    PC21
Standard deviation     2.24564 2.21327 2.18855 2.1681 2.1462 2.10843 2.08510
Proportion of Variance 0.01062 0.01031 0.01008 0.0099 0.0097 0.00936 0.00915
Cumulative Proportion  0.63226 0.64258 0.65266 0.6626 0.6723 0.68161 0.69076
                          PC22    PC23    PC24    PC25    PC26    PC27    PC28
Standard deviation     2.02730 2.01845 1.95205 1.91161 1.86509 1.84638 1.83184
Proportion of Variance 0.00865 0.00858 0.00802 0.00769 0.00732 0.00718 0.00706
Cumulative Proportion  0.69942 0.70799 0.71602 0.72371 0.73103 0.73821 0.74527
                          PC29    PC30    PC31    PC32    PC33    PC34    PC35
Standard deviation     1.82624 1.81628 1.76089 1.73452 1.69435 1.68276 1.66454
Proportion of Variance 0.00702 0.00695 0.00653 0.00633 0.00604 0.00596 0.00583
Cumulative Proportion  0.75230 0.75924 0.76577 0.77210 0.77815 0.78411 0.78994
                          PC36    PC37   PC38    PC39    PC40    PC41    PC42
Standard deviation     1.65906 1.62985 1.6161 1.59049 1.57398 1.56278 1.55085
Proportion of Variance 0.00579 0.00559 0.0055 0.00533 0.00522 0.00514 0.00506
Cumulative Proportion  0.79574 0.80133 0.8068 0.81215 0.81737 0.82251 0.82757
                          PC43    PC44    PC45    PC46    PC47    PC48    PC49
Standard deviation     1.54608 1.53399 1.53190 1.52334 1.51515 1.46995 1.45829
Proportion of Variance 0.00503 0.00495 0.00494 0.00489 0.00483 0.00455 0.00448
Cumulative Proportion  0.83260 0.83756 0.84250 0.84738 0.85222 0.85677 0.86124
                          PC50    PC51   PC52    PC53    PC54    PC55    PC56
Standard deviation     1.43807 1.41947 1.4118 1.39282 1.37367 1.36640 1.35806
Proportion of Variance 0.00435 0.00424 0.0042 0.00408 0.00397 0.00393 0.00388
Cumulative Proportion  0.86560 0.86984 0.8740 0.87812 0.88209 0.88602 0.88991
                          PC57   PC58    PC59    PC60    PC61    PC62    PC63
Standard deviation     1.34707 1.3249 1.31220 1.28503 1.27454 1.26128 1.24039
Proportion of Variance 0.00382 0.0037 0.00362 0.00348 0.00342 0.00335 0.00324
Cumulative Proportion  0.89373 0.8974 0.90105 0.90452 0.90794 0.91129 0.91453
                         PC64    PC65    PC66    PC67    PC68    PC69    PC70
Standard deviation     1.2326 1.21995 1.20051 1.18288 1.17106 1.16242 1.14475
Proportion of Variance 0.0032 0.00313 0.00303 0.00295 0.00289 0.00284 0.00276
Cumulative Proportion  0.9177 0.92086 0.92390 0.92684 0.92973 0.93257 0.93533
                          PC71    PC72    PC73    PC74    PC75    PC76    PC77
Standard deviation     1.13640 1.12952 1.10683 1.10248 1.08830 1.07329 1.06585
Proportion of Variance 0.00272 0.00269 0.00258 0.00256 0.00249 0.00243 0.00239
Cumulative Proportion  0.93805 0.94074 0.94332 0.94588 0.94837 0.95079 0.95319
                          PC78    PC79    PC80    PC81    PC82    PC83    PC84
Standard deviation     1.05289 1.03925 1.02051 1.01535 0.99384 0.96728 0.96611
Proportion of Variance 0.00233 0.00227 0.00219 0.00217 0.00208 0.00197 0.00196
Cumulative Proportion  0.95552 0.95779 0.95999 0.96216 0.96424 0.96621 0.96817
                          PC85    PC86    PC87    PC88    PC89    PC90    PC91
Standard deviation     0.95319 0.93714 0.92882 0.89236 0.88891 0.87637 0.86367
Proportion of Variance 0.00191 0.00185 0.00182 0.00168 0.00166 0.00162 0.00157
Cumulative Proportion  0.97008 0.97193 0.97375 0.97542 0.97709 0.97870 0.98028
                          PC92    PC93    PC94    PC95    PC96    PC97    PC98
Standard deviation     0.85877 0.85077 0.83298 0.81957 0.81333 0.80289 0.76496
Proportion of Variance 0.00155 0.00152 0.00146 0.00141 0.00139 0.00136 0.00123
Cumulative Proportion  0.98183 0.98335 0.98481 0.98623 0.98762 0.98898 0.99021
                         PC99   PC100   PC101   PC102   PC103   PC104   PC105
Standard deviation     0.7542 0.74544 0.71378 0.70698 0.70179 0.68516 0.66594
Proportion of Variance 0.0012 0.00117 0.00107 0.00105 0.00104 0.00099 0.00093
Cumulative Proportion  0.9914 0.99258 0.99365 0.99470 0.99574 0.99673 0.99766
                         PC106   PC107   PC108     PC109
Standard deviation     0.64548 0.59660 0.58250 8.508e-15
Proportion of Variance 0.00088 0.00075 0.00071 0.000e+00
Cumulative Proportion  0.99854 0.99929 1.00000 1.000e+00</code></pre>
<pre class="r"><code>pca.results.to.plot &lt;- pca.results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;leafcutter.name&quot;) %&gt;%
  dplyr::select(leafcutter.name, PC1:PC6) %&gt;%
  left_join(FullMetadata, by=&quot;leafcutter.name&quot;)

SamplesToProject &lt;- FullMetadata %&gt;%
  filter(!Experiment %in% c(&quot;Single high dose LCL&quot;)) %&gt;%
  filter(!old.sample.name == &quot;NewMolecule.C04-2&quot;) %&gt;%
  pull(leafcutter.name)

pca.results.to.plot_AdditionProjections &lt;- PSI.IntronsToInclude %&gt;%
  dplyr::select(junc, all_of(SamplesToProject)) %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% predict(pca.results, .) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;leafcutter.name&quot;) %&gt;%
  dplyr::select(leafcutter.name, PC1:PC6) %&gt;%
  left_join(FullMetadata, by=&quot;leafcutter.name&quot;)


TreatmentColorsForLabels.FibroblastColorsAdded &lt;-
FullMetadata %&gt;%
  group_by(treatment) %&gt;%
  filter(dose.nM == max(dose.nM) | treatment == &quot;DMSO&quot; | cell.type==&quot;Fibroblast&quot;) %&gt;%
  ungroup() %&gt;%
  distinct(treatment, .keep_all=T) %&gt;%
  arrange(dose.nM) %&gt;%
  mutate(vjust=row_number()*1.2)

TreatmentColorsLabels.Layer.FibroblastColorsAdded &lt;- geom_text(
    data = TreatmentColorsForLabels.FibroblastColorsAdded, 
    aes(label=treatment, color=color, vjust=vjust),
    y=Inf, x=Inf, hjust=1.05
  )

  
ggplot(pca.results.to.plot, aes(x=PC1, y=PC2, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  geom_point(data = pca.results.to.plot_AdditionProjections) +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 475 GAGT introns\nAll samples&quot;, caption=&quot;Introns based on being significant in any contrast&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-19-1.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC3, y=PC4, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  geom_point(data = pca.results.to.plot_AdditionProjections) +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 475 GAGT introns\nAll samples&quot;, caption=&quot;Introns based on being significant in any contrast&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-19-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-2">
Past versions of unnamed-chunk-19-2.png
</button>
</p>
<div id="fig-unnamed-chunk-19-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/a8c9152c8020689603208f678ef53d6864f0077c/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-19-2.png" target="_blank">a8c9152</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok this is actually somewhat interpretable. Again the main axis of
specificity seems to be along branaplam vs risdiplam effects, and there
is plenty of stuff in between. WD07 is actually 500nM risdiplam, which
looks fitting in relation to previous experiments. I think perhaps WC05
(SMSM32) and WD08 (4F-Py) the most interesting point I see so far… as
both replicates doesn’t neatly lie on either the risdiplam axis or the
branaplam axis. Just browsing these molecules structure that ZT shared
with me, these seem by my eye pretty structurally distinct. I would plot
the structures of these in the PCA but <code>ggimage</code> is kind of
tricky to install in my current R environment, so that may have to wait
for later</p>
<ul>
<li>update: I turned out making that plot with ggimage in a seperate
notebook that I run in a different R environment… I wrote out the data
for that plot in the block below and wrote a seperate notebook to just
read in that data and plot…</li>
</ul>
<pre class="r"><code>FullMetadata %&gt;%
  write_tsv(&quot;../output/20230223_FullMetadata.tsv&quot;)


bind_rows(pca.results.to.plot,
          pca.results.to.plot_AdditionProjections) %&gt;%
  write_tsv(&quot;../output/20230223_PCA_Projections.tsv.gz&quot;)
 

PSI.IntronsToInclude %&gt;%
  write_tsv(&quot;../output/20230223_IntronsToInclude.tsv.gz&quot;)</code></pre>
<p>Let’s make sure these three molecules don’t have very different read
depth or something…</p>
<pre class="r"><code>pca.results.to.plot %&gt;%
  inner_join(
    counts.plot.dat %&gt;%
      dplyr::select(SampleName, lib.size)
    ) %&gt;%
ggplot( aes(x=PC1, y=PC2, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment, color=lib.size)) +
  geom_point(data = pca.results.to.plot_AdditionProjections) +
  scale_color_viridis_c(trans=&#39;log10&#39;) +
  # scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 475 GAGT introns\nAll samples&quot;, caption=&quot;Introns based on being significant in any contrast&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-21-1.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, that makes me feel better these three interesting candidate
molecules aren’t just outliers in read depth.</p>
<p>Next I want to plot a heatmap of the correlation matrix based on
these introns, I wonder if any clear clustering patterns will come out
of this… Part of what I want to get out of this is to answer if these
three outliers (WD08, WC05, WB12) are similar to each other, or if they
are as distant from eachother as they are from risdiplam/branaplam.</p>
<pre class="r"><code>SamplesToInclude &lt;- FullMetadata %&gt;%
  filter(Experiment %in% c(&quot;Single high dose LCL&quot;, &quot;Dose response titration&quot;)) %&gt;%
  filter(!old.sample.name == &quot;NewMolecule.C04-2&quot;) %&gt;%
  pull(leafcutter.name)

PSI.IntronsToInclude.cormat &lt;- PSI.IntronsToInclude %&gt;%
  dplyr::select(junc, all_of(SamplesToInclude)) %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  cor(method=&#39;s&#39;)

GetColorPallete &lt;- function(MyVector, PalletteName = &quot;Dark2&quot;){
  UniqueFactors &lt;- MyVector %&gt;% unique()
  return (setNames(brewer.pal(n=length(UniqueFactors), name=PalletteName), UniqueFactors))
}



MetadataForCormat &lt;- data.frame(leafcutter.name = PSI.IntronsToInclude.cormat %&gt;%
  rownames()) %&gt;%
  left_join(FullMetadata) %&gt;%
  mutate(color = if_else(
    treatment %in% c(&quot;WD08&quot;, &quot;WC05&quot;, &quot;WB12&quot;),
    &quot;#e34a33&quot;,
    color
  )) %&gt;%
  mutate(ExperimentColor = recode(Experiment, !!!GetColorPallete(FullMetadata$Experiment)
))


  
heatmap.2(PSI.IntronsToInclude.cormat, trace=&#39;none&#39;, RowSideColors = MetadataForCormat$color, ColSideColors = MetadataForCormat$color, labRow = MetadataForCormat$treatment, cexRow=0.2, labCol = MetadataForCormat$treatment, cexCol=0.2)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-22-1.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>pdf(&quot;../code/scratch/Heatmap.pdf&quot;)
heatmap.2(PSI.IntronsToInclude.cormat, trace=&#39;none&#39;, RowSideColors = MetadataForCormat$color, ColSideColors = MetadataForCormat$color, labRow = MetadataForCormat$treatment, cexRow=0.2, labCol = MetadataForCormat$treatment, cexCol=0.2)
dev.off()</code></pre>
<p>Ok samples mostly just are clustering by effective dose, so you can’t
really see the specificity effect here like you can in the second PC
above… Well, you can see in the hierarchal clustering that both reps of
WC05, a rep of WD08, and a rep of WB12 cluster together… And far away on
another branch are the other replicates of WD08 and WB12…</p>
<p>so maybe their effects are similar? It’s just really hard to
tell.</p>
<p>As a slightly less biased way of making this PCA and possibly
reproducing the potential uniqueness of these 3 molecules, let’s try
changing the criteria for including introns to be more inclusive.</p>
<pre class="r"><code>IntronsToInclude &lt;- leafcutter.sig %&gt;%
  filter(p.adjust &lt; 0.1) %&gt;%
  inner_join(leafcutter.effects, by=c(&quot;treatment&quot;, &quot;cluster&quot;)) %&gt;%
  filter(abs(deltapsi) &gt; 0.01) %&gt;%
  distinct(name, .keep_all=T) %&gt;%
  inner_join(Introns.annotations, by=&#39;name&#39;) %&gt;%
  # filter(str_detect(seq, &quot;^\\w\\wGAGT&quot;)) %&gt;%
  pull(intron)

length(IntronsToInclude)</code></pre>
<pre><code>[1] 47065</code></pre>
<pre class="r"><code>PSI.IntronsToInclude &lt;- PSI %&gt;%
  filter(junc %in% IntronsToInclude) %&gt;%
  dplyr::select(junc, A10.1:NewMolecule.E07.6) %&gt;%
  drop_na()

dim(PSI.IntronsToInclude)</code></pre>
<pre><code>[1] 23416   173</code></pre>
<pre class="r"><code>SamplesToInclude &lt;- FullMetadata %&gt;%
  filter(Experiment %in% c(&quot;Single high dose LCL&quot;)) %&gt;%
  filter(!old.sample.name == &quot;NewMolecule.C04-2&quot;) %&gt;%
  pull(leafcutter.name)


pca.results &lt;- PSI.IntronsToInclude %&gt;%
  dplyr::select(junc, all_of(SamplesToInclude)) %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% prcomp(scale=T)


pca.results.to.plot &lt;- pca.results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;leafcutter.name&quot;) %&gt;%
  dplyr::select(leafcutter.name, PC1:PC6) %&gt;%
  left_join(FullMetadata, by=&quot;leafcutter.name&quot;) %&gt;%
  mutate(color = if_else(
    treatment %in% c(&quot;WD08&quot;, &quot;WC05&quot;, &quot;WB12&quot;),
    &quot;#e34a33&quot;,
    color
  )) 

SamplesToProject &lt;- FullMetadata %&gt;%
  filter(!Experiment %in% c(&quot;Single high dose LCL&quot;)) %&gt;%
  filter(!old.sample.name == &quot;NewMolecule.C04-2&quot;) %&gt;%
  pull(leafcutter.name)

pca.results.to.plot_AdditionProjections &lt;- PSI.IntronsToInclude %&gt;%
  dplyr::select(junc, all_of(SamplesToProject)) %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% predict(pca.results, .) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;leafcutter.name&quot;) %&gt;%
  dplyr::select(leafcutter.name, PC1:PC6) %&gt;%
  left_join(FullMetadata, by=&quot;leafcutter.name&quot;)

  
ggplot(pca.results.to.plot, aes(x=PC1, y=PC2, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  geom_point(data = pca.results.to.plot_AdditionProjections) +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 23416 introns\nAll samples&quot;, caption=&quot;Introns based on being significant in any contrast&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-24-1.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>pca.results.to.plot %&gt;%
  inner_join(
    counts.plot.dat %&gt;%
      dplyr::select(SampleName, lib.size)
    ) %&gt;%
ggplot(aes(x=PC1, y=PC2, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment, color=lib.size)) +
  geom_point(data = pca.results.to.plot_AdditionProjections) +
  scale_color_viridis_c(trans=&#39;log10&#39;) +
  # scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 23416 introns\nAll samples&quot;, caption=&quot;Introns based on being significant in any contrast. Colored by depth&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-24-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-2">
Past versions of unnamed-chunk-24-2.png
</button>
</p>
<div id="fig-unnamed-chunk-24-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-24-2.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC3, y=PC4, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  geom_point(data = pca.results.to.plot_AdditionProjections) +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 23416 introns\nAll samples&quot;, caption=&quot;Introns based on being significant in any contrast&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-24-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-3">
Past versions of unnamed-chunk-24-3.png
</button>
</p>
<div id="fig-unnamed-chunk-24-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-24-3.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC4, y=PC5, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  geom_point(data = pca.results.to.plot_AdditionProjections) +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 23416 introns\nAll samples&quot;, caption=&quot;Introns based on being significant in any contrast&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-24-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-4">
Past versions of unnamed-chunk-24-4.png
</button>
</p>
<div id="fig-unnamed-chunk-24-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-24-4.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok interesting… when you look at all significant introns with
abs(deltapsi) &gt; 0.01, the first few PCs measure hidden technical
variables that don’t seem to cluster biological replicates very well.
deeper PCs seem to work better to find the dosage axis (PC4) branaplam
vs risdiplam axis (PC5) and biological replicates still are close to one
another… But in this space, a different set of molecules starts to show
up, and perhaps on a different ‘arm’ of the data. I still see D08
sticking out.. WE02 also sticks out here along that axis, and WE02 was
previously along the branaplam axis and is structurally most like
branaplam.</p>
<p>Now I’m a bit concerned about the robustness of the PCA to interpret
it as being a distinct thing.. Let’s try changing up the intron critera
a tad a again and see how robust the PCA is…</p>
<pre class="r"><code>IntronsToInclude &lt;- leafcutter.sig %&gt;%
  filter(p.adjust &lt; 0.1) %&gt;%
  inner_join(leafcutter.effects, by=c(&quot;treatment&quot;, &quot;cluster&quot;)) %&gt;%
  filter(logef &gt; 0) %&gt;%
  distinct(name, .keep_all=T) %&gt;%
  inner_join(Introns.annotations, by=&#39;name&#39;) %&gt;%
  filter(str_detect(seq, &quot;^\\w\\wGAGT&quot;)) %&gt;%
  pull(intron)

length(IntronsToInclude)</code></pre>
<pre><code>[1] 1138</code></pre>
<pre class="r"><code>PSI.IntronsToInclude &lt;- PSI %&gt;%
  filter(junc %in% IntronsToInclude) %&gt;%
  dplyr::select(junc, A10.1:NewMolecule.E07.6) %&gt;%
  drop_na()

dim(PSI.IntronsToInclude)</code></pre>
<pre><code>[1] 632 173</code></pre>
<pre class="r"><code>SamplesToInclude &lt;- FullMetadata %&gt;%
  filter(Experiment %in% c(&quot;Single high dose LCL&quot;)) %&gt;%
  filter(!old.sample.name == &quot;NewMolecule.C04-2&quot;) %&gt;%
  pull(leafcutter.name)


pca.results &lt;- PSI.IntronsToInclude %&gt;%
  dplyr::select(junc, all_of(SamplesToInclude)) %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% prcomp(scale=T)


pca.results.to.plot &lt;- pca.results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;leafcutter.name&quot;) %&gt;%
  dplyr::select(leafcutter.name, PC1:PC6) %&gt;%
  left_join(FullMetadata, by=&quot;leafcutter.name&quot;) %&gt;%
  mutate(color = if_else(
    treatment %in% c(&quot;WD08&quot;, &quot;WC05&quot;, &quot;WB12&quot;),
    &quot;#e34a33&quot;,
    color
  )) 

SamplesToProject &lt;- FullMetadata %&gt;%
  filter(!Experiment %in% c(&quot;Single high dose LCL&quot;)) %&gt;%
  filter(!old.sample.name == &quot;NewMolecule.C04-2&quot;) %&gt;%
  pull(leafcutter.name)

pca.results.to.plot_AdditionProjections &lt;- PSI.IntronsToInclude %&gt;%
  dplyr::select(junc, all_of(SamplesToProject)) %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% predict(pca.results, .) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;leafcutter.name&quot;) %&gt;%
  dplyr::select(leafcutter.name, PC1:PC6) %&gt;%
  left_join(FullMetadata, by=&quot;leafcutter.name&quot;)

  
ggplot(pca.results.to.plot, aes(x=PC1, y=PC2, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  geom_point(data = pca.results.to.plot_AdditionProjections) +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nAll samples&quot;, caption=&quot;Introns based on being significant in any contrast&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-25-1.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>pca.results.to.plot %&gt;%
  inner_join(
    counts.plot.dat %&gt;%
      dplyr::select(SampleName, lib.size)
    ) %&gt;%
ggplot(aes(x=PC1, y=PC2, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment, color=lib.size)) +
  geom_point(data = pca.results.to.plot_AdditionProjections) +
  scale_color_viridis_c(trans=&#39;log10&#39;) +
  # scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nAll samples&quot;, caption=&quot;Introns based on being significant in any contrast. Colored by depth&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-25-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-2">
Past versions of unnamed-chunk-25-2.png
</button>
</p>
<div id="fig-unnamed-chunk-25-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-25-2.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC3, y=PC4, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  geom_point(data = pca.results.to.plot_AdditionProjections) +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nAll samples&quot;, caption=&quot;Introns based on being significant in any contrast&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-25-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-3">
Past versions of unnamed-chunk-25-3.png
</button>
</p>
<div id="fig-unnamed-chunk-25-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-25-3.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC3, y=PC4, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  # geom_point(data = pca.results.to.plot_AdditionProjections) +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nAll samples&quot;, caption=&quot;Introns based on being significant in any contrast&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-25-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-4">
Past versions of unnamed-chunk-25-4.png
</button>
</p>
<div id="fig-unnamed-chunk-25-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-25-4.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC4, y=PC5, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  geom_point(data = pca.results.to.plot_AdditionProjections) +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nAll samples&quot;, caption=&quot;Introns based on being significant in any contrast&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-25-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-5">
Past versions of unnamed-chunk-25-5.png
</button>
</p>
<div id="fig-unnamed-chunk-25-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-25-5.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(pca.results.to.plot, aes(x=PC4, y=PC5, color=color, shape=Experiment)) +
  geom_text(size=3, aes(label=treatment)) +
  # geom_point(data = pca.results.to.plot_AdditionProjections) +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nAll samples&quot;, caption=&quot;Introns based on being significant in any contrast&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-25-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-6">
Past versions of unnamed-chunk-25-6.png
</button>
</p>
<div id="fig-unnamed-chunk-25-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-25-6.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Hmm, ok much like the first PCA. Of course these are surely mostly
overlapping introns. But at least is sort of comforting. I thought
looking at deeper PCs might reveal something… Mostly reveals that
interpreting PCs is not always straightforward.</p>
<p>Also, if C05, D08, and B12 have unique properties, it is not so clear
if are similar or distinct from each other.</p>
<p>First let’s write out these PCA, for later use…</p>
<pre class="r"><code>pca.results.to.plot %&gt;%
  write_tsv(&quot;../output/20230227_PCA_Projections.tsv.gz&quot;)</code></pre>
</div>
</div>
<div id="investigation-of-the-c05-d08-and-b12" class="section level2">
<h2>Investigation of the C05, D08, and B12</h2>
<p>Let’s check the behavior of the hits (and all samples) on some
particular introns or genes… The HTT, MYB, SMN2, STAT1</p>
<p>First let’s glance at the differential expression test results (which
also include effect size estimates for these genes). Since gene
expression changes are generally less noisy to estimate than splicing,
we can also use these as proxies for risdiplam vs branaplam-specific
effects for a set of known genes.</p>
<pre class="r"><code>diff.expression &lt;- fread(&quot;../code/DE_testing/ExpOf52_Results.txt.gz&quot;) %&gt;%
  mutate(treatment = str_replace(treatment, &quot;group(.+?)&quot;, &quot;W\\1&quot;)) %&gt;%
  separate(Geneid, into=c(&quot;EnsemblID&quot;, &quot;symbol&quot;), sep=&#39;_&#39;)

diff.expression %&gt;%
  filter(symbol %in% c(&quot;STAT1&quot;, &quot;HTT&quot;, &quot;MYB&quot;, &quot;SMN2&quot;, &quot;SMN1&quot;, &quot;GALC&quot;, &quot;FOXM1&quot;)) %&gt;%
  left_join(FullMetadata) %&gt;%
  mutate(color = if_else(
    treatment %in% c(&quot;WD08&quot;, &quot;WC05&quot;, &quot;WB12&quot;),
    &quot;#e34a33&quot;,
    color
  )) %&gt;%
  mutate(Sig = case_when(
    PValue &lt; 0.001 ~ &quot;!&quot;,
    PValue &lt; 0.05 ~ &quot;*&quot;,
    TRUE ~ NA_character_
  )) %&gt;%
  ggplot(aes(x=treatment, y=logFC, fill=color)) +
  geom_col() +
  geom_text(aes(label=Sig), y=Inf, vjust=1) +
  scale_fill_identity() +
  facet_wrap(~symbol) +
  Rotate_x_labels +
  theme(axis.text.x = element_text(size=3))</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-27-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-27-1.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Hmm.. Offhand I can’t really remember which if any of these genes
(besides HTT) should be branaplam or risdiplam-specific. I think one
helpful way to plot this will be to incorporate titration series data
somehow, but that is a bit of work. Another way will be to color all
samples (some of which are clearly on the risdiplam or branaplam axis)
by the degree to which they were on those axes in PC space. Let’s come
up with a color scheme based on that idea…</p>
<p>For example, looking at the previous PCA plot…</p>
<p>If PC2 &gt; 0 then blue, ifelse PC2 &lt; 0, then green And PC2 is an
alpha fill scale.</p>
<pre class="r"><code>NewColors &lt;- pca.results.to.plot %&gt;%
  filter(Experiment == &quot;Single high dose LCL&quot;) %&gt;%
  group_by(treatment) %&gt;%
  summarise(PC1 = mean(PC1), PC2=mean(PC2)) %&gt;%
  mutate(NewColor = case_when(
    PC2 &gt; 0 ~ &quot;#08519c&quot;,
    PC2 &lt; 0 ~ &quot;#006d2c&quot;,
  ))

MinPC1 &lt;- min(NewColors$PC1)
MaxPC1 &lt;- max(NewColors$PC1)

NewColors &lt;- NewColors %&gt;%
  mutate(alpha = (PC1 - MinPC1)/(MaxPC1 - MinPC1)) %&gt;%
  mutate(NewColor = if_else(
    treatment %in% c(&quot;WD08&quot;, &quot;WC05&quot;, &quot;WB12&quot;),
    &quot;#e34a33&quot;,
    NewColor
  ))

diff.expression %&gt;%
  filter(symbol %in% c(&quot;STAT1&quot;, &quot;HTT&quot;, &quot;MYB&quot;, &quot;SMN2&quot;, &quot;SMN1&quot;, &quot;GALC&quot;, &quot;FOXM1&quot;)) %&gt;%
  left_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  mutate(Sig = case_when(
    PValue &lt; 0.001 ~ &quot;!&quot;,
    PValue &lt; 0.05 ~ &quot;*&quot;,
    TRUE ~ NA_character_
  )) %&gt;%
  ggplot(aes(x=treatment, y=logFC, fill=NewColor)) +
  geom_col(aes(alpha=alpha)) +
  geom_text(aes(label=Sig), y=Inf, vjust=1) +
  scale_fill_identity() +
  facet_wrap(~symbol) +
  Rotate_x_labels +
  theme(axis.text.x = element_text(size=6), legend.position=&#39;none&#39;) +
  labs(title=&quot;Expression changes in genes of interest&quot;, y=&quot;log2FC&quot;, caption=str_wrap(&quot;P&lt;0.05 = *\nP&lt;0.001 = !\nColor and  fill transparency based on PC1 &amp; PC2 to distinguish risdiplam/branaplam axes&quot;, 30))</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-28-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-28-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-28-1.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(&quot;../code/scratch/scratch.pdf&quot;, height=6, width=10)</code></pre>
<p>Ok, so based on these subset of genes, the red molecules have some
risdiplam-like effects and some branaplam-like effects. Let’s look more
globally at the DE results..</p>
<pre class="r"><code>XaxisColors &lt;- NewColors %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  arrange(treatment) %&gt;% pull(NewColor)


NumberDE.genes &lt;- bind_rows(
diff.expression %&gt;%
  filter(FDR &lt; 0.01) %&gt;%
  count(treatment) %&gt;%
  mutate(FDR = 0.01),
diff.expression %&gt;%
  filter(FDR &lt; 0.05) %&gt;%
  count(treatment) %&gt;%
  mutate(FDR = 0.05),
diff.expression %&gt;%
  filter(FDR &lt; 0.1) %&gt;%
  count(treatment) %&gt;%
  mutate(FDR = 0.1)) %&gt;%
  left_join(NewColors, by=&#39;treatment&#39;) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  ggplot(aes(x=treatment, y=n)) +
  geom_bar(stat=&quot;identity&quot;,position = &quot;identity&quot;, alpha=.15) +
  geom_text(y=Inf,  vjust=1, label=&quot;*&quot;, aes(color=NewColor, alpha=alpha)) +
  Rotate_x_labels +
  scale_color_identity() +
  theme(axis.text.x = element_text(size=6)) +
  theme(legend.position=&#39;none&#39;) +
  labs(title=&quot;Number of DE genes&quot;, x=&quot;treatment&quot;, caption=str_wrap(&quot;FDR 1%, 5%, 10%\nAsterisk Color for PC1/PC2 to approximate branaplam/risdiplam axes&quot;, 30))
NumberDE.genes</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-30-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-30-1.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok wow WD08 is a real outlier there. Tons of DE genes… It also had
the most leafcutter hits from this experiment. I’ll have to look more
into that later. First let’s zoom in on the y=axis to better look at the
other contrasts.</p>
<pre class="r"><code>NumberDE.genes +
  coord_cartesian(ylim=c(0,20))</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-31-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-31-1.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok so a typical contrast has more like 5-10 DE genes at a reasonable
FDR threshold. It’s also worth noting that the branaplam like molecules,
when roughly have similar amounts of differential splicing, have much
less differential expression.</p>
<p>Let me go back to WD08… I’m wondering if this sample just had a crazy
amount of read depth or something..</p>
<pre class="r"><code>counts.plot.dat %&gt;%
  filter(Experiment == &quot;Single high dose LCL&quot;) %&gt;%
  dplyr::select(SampleName, lib.size, treatment) %&gt;%
  left_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
  mutate(SampleName = fct_reorder(SampleName, PC2 * alpha)) %&gt;%
  ggplot(aes(x=SampleName, y=lib.size)) +
  geom_col(aes(fill=NewColor, alpha=alpha)) +
  Rotate_x_labels +
  scale_fill_identity() +
  theme(axis.text.x = element_text(size=6)) +
  theme(legend.position=&#39;none&#39;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-32-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-32-1.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, so those three treatments, included WD08 are not huge outliers in
terms of sequencing depth. Maybe there is something genuinely different
about it that makes it have many DE genes and splicing changes.</p>
<p>Let’s go back to looking at genomewide GA-GT activation levels…</p>
<pre class="r"><code>leafcutter.effects %&gt;%
  inner_join(
    Introns.annotations %&gt;%
    mutate(IsGAGT = if_else(str_detect(seq, &quot;^\\w\\wGAGT&quot;), &quot;GA-GT&quot;, &quot;not GA-GT&quot;)),
    by=&quot;name&quot;) %&gt;%
  group_by(IsGAGT, treatment) %&gt;%
  summarise(medLogef = median(logef)) %&gt;%
  left_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  ggplot(aes(x=treatment, y=medLogef, fill=NewColor, alpha=alpha)) +
  geom_col() +
  scale_fill_identity() +
  coord_flip() +
  facet_wrap(~IsGAGT) +
  theme(axis.text.y = element_text(size=6)) +
  theme(legend.position=&#39;none&#39;) +
  labs(title=&quot;3 treatments in question still activate GA-GT\nto similar levels as branaplam/risdiplam -like treatments&quot;, y=&quot;median splicing logFC&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-33-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-33-1.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok I have to do this for all dinucleotides…</p>
<pre class="r"><code>leafcutter.effects %&gt;%
  inner_join(
    Introns.annotations %&gt;%
    mutate(UpstreamOfDonor2BaseSeq = substr(seq, 3, 4)),
    by=&quot;name&quot;) %&gt;%
  group_by(UpstreamOfDonor2BaseSeq, treatment) %&gt;%
  summarise(medLogef = median(logef)) %&gt;%
  left_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  ggplot(aes(y=treatment, x=UpstreamOfDonor2BaseSeq, fill=medLogef)) +
  geom_raster() +
  scale_fill_viridis_c() +
  theme(axis.text.y = element_text(size=6, colour = XaxisColors))</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-34-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-34-1.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, well firstly it is nice to see preferential activation of GA
across the board. And there isn’t any other dinucleotide that pops out
in the three red samples in question. And the degree of GA-GT activation
doesn’t seem all that different from other treatments similar in PC1. I
think this is particularly interesting for WD08 which at 3-4x as many
differentially spliced clusters and ~100x as many DE genes as a typical
treatment. I’m leaning towards thinking this molecule just might elicit
non-splicing related toxic effects. Let’s count the number of
differentially spliced NNGT introns for each effect to better understand
this… I’m wondering whether those 3-4x as many differentially spliced
introns are from non GAGT introns (eg indirect effects).</p>
<pre class="r"><code># leafcutter.effects %&gt;%
#   inner_join(
#     Introns.annotations %&gt;%
#     mutate(UpstreamOfDonor2BaseSeq = substr(seq, 3, 4)),
#     by=&quot;name&quot;) %&gt;%
#   group_by(UpstreamOfDonor2BaseSeq, treatment) %&gt;%
#   summarise(medLogef = median(logef)) %&gt;%
#   left_join(NewColors) %&gt;%
#   mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
#   mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
#   ggplot(aes(y=treatment, x=UpstreamOfDonor2BaseSeq, fill=medLogef)) +
#   geom_raster() +
#   scale_fill_viridis_c() +
#   theme(axis.text.y = element_text(size=6, colour = XaxisColors))

leafcutter.effects_and_sig &lt;- leafcutter.effects %&gt;%
  inner_join(leafcutter.sig) %&gt;%
  inner_join(
    Introns.annotations %&gt;%
    mutate(UpstreamOfDonor2BaseSeq = substr(seq, 3, 4)),
    by=&quot;name&quot;)

leafcutter.effects_and_sig %&gt;%
  group_by(cluster, treatment) %&gt;%
  mutate(ContainsGAGT = any(UpstreamOfDonor2BaseSeq==&quot;GA&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(p.adjust &lt; 0.1) %&gt;%
  distinct(cluster, treatment, .keep_all=T) %&gt;%
  count(ContainsGAGT, treatment) %&gt;%
  left_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  ggplot(aes(x=treatment, y=n, fill=ContainsGAGT)) +
  geom_col(position=&quot;stack&quot;) +
  coord_flip() +
  theme(axis.text.y = element_text(size=6, colour = XaxisColors), legend.position=&#39;bottom&#39;) +
  labs(title=&quot;WD08 has many non-GAGT effects&quot;, y=&quot;Number significant clusters, FDR&lt;10%&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-1">
Past versions of unnamed-chunk-35-1.png
</button>
</p>
<div id="fig-unnamed-chunk-35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-35-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-35-1.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, so now I am sort of worried the WD08 effect might be indirect
effects. There is much more AS but much of it is not at GAGT introns,
and also there is tons of tons of DE.</p>
<p>The other two interesting molecules were WC05 and WB12. Those ones I
still don’t understand as well. Looking at the PCA again, WC05 is the
larger deparature from the rsidiplam/branaplam axes than WB12. I’ll keep
a particular eye on WC05.</p>
<p>I’m going to look at the -3 and -4 positions now. So I’ll just filter
for the GAGT introns, and make a similar plot as the heatmap I plotted
above for the -3 and -4 position …</p>
<pre class="r"><code>leafcutter.effects_and_sig %&gt;%
  filter(UpstreamOfDonor2BaseSeq == &quot;GA&quot;) %&gt;%
  mutate(NNGAGT = substr(seq, 1, 2)) %&gt;%
  group_by(NNGAGT, treatment) %&gt;%
  summarise(medLogef = median(logef)) %&gt;%
  left_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  ggplot(aes(y=treatment, x=NNGAGT, fill=medLogef)) +
  geom_raster() +
  scale_fill_viridis_c() +
  theme(axis.text.y = element_text(size=6, colour = XaxisColors))</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-36-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-36-1.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, so similar to previous Bhattacharya et al, and our dose titration
series, we can see the -4 and -3 positions distinguish the
branaplam-like molecules vs the risdiplam-like molecules. WC05 and WB12
don’t look particularly special on this plot. If anything I think WC05
may look a bit more like risdiplam in this respect (I’m looking in
particular at the AT column), consistent with its structural similarity
with risdiplam. WC04 pops out as interesting, even though it didn’t
necessarily pop out from the PCA. I do notice in the above plots that
WC04 had a lot of non GAGT diff splicing and a larger number of DE
genes. Could also be some indirect effects.</p>
<p>Let’s go back to the DE results. For now I’ll plot volcano plots of
some select molecules to see what the top genes are for some samples of
interest: WC05, WB12, and some reference samples to represent risdiplam
like effects (WB09, WB10) and branaplam like effects (WE02, WE04)</p>
<pre class="r"><code>diff.expression %&gt;%
  filter(treatment %in% c(&quot;WC05&quot;, &quot;WB12&quot;, &quot;WE04&quot;, &quot;WE02&quot;, &quot;WB09&quot;,&quot;WB10&quot;)) %&gt;%
  mutate(IsSig = FDR&lt;0.1) %&gt;%
  mutate(label = if_else(
    IsSig,
    symbol,
    NA_character_
  )) %&gt;%
  mutate(treatment = recode(treatment,
    &quot;WE02&quot; = &quot;WE02; branaplam-like&quot;,
    &quot;WE04&quot; = &quot;WE04; branaplam-like&quot;,
    &quot;WB09&quot; = &quot;WB09; risdiplam-like&quot;,
    &quot;WB10&quot; = &quot;WB10; risdiplam-like&quot;,
  )) %&gt;%
  ggplot(aes(x=logFC, y=-log10(PValue), color=IsSig)) +
  geom_point() +
  scale_color_manual(values=c(`TRUE`=&quot;red&quot;, `FALSE`=&quot;black&quot;)) +
  geom_text_repel(aes(label=label), size=3, max.overlaps=) +
  facet_wrap(~treatment, scales = &quot;free&quot;) +
  labs(&quot;DE results, focusing on WB12 and WC05&quot;, color=&quot;FDR&lt;10%&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-37-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-37-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/0e2a3601db2a0f243a1f62c7add5d1f194e074cd/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-37-1.png" target="_blank">0e2a360</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok honing in on some genes, in some ways WC05 and WB12 are more
risdiplam-like (DENND5A), and in some other ways they are more branaplam
like (CRYL1), and in some ways they might be like niether (ZNF837)…</p>
<p>Let’s plot these genes and a few others more closely…</p>
<pre class="r"><code>diff.expression %&gt;%
  filter(symbol %in% c(&quot;HTT&quot;, &quot;RCC1&quot;, &quot;CRYL1&quot;, &quot;ZNF837&quot;, &quot;DENND5A&quot;, &quot;FHOD3&quot;, &quot;STAT1&quot;, &quot;PITPNB&quot;)) %&gt;%
  left_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  mutate(Sig = case_when(
    PValue &lt; 0.001 ~ &quot;!&quot;,
    PValue &lt; 0.05 ~ &quot;*&quot;,
    TRUE ~ NA_character_
  )) %&gt;%
  ggplot(aes(x=treatment, y=logFC, fill=NewColor)) +
  geom_col(aes(alpha=alpha)) +
  geom_text(aes(label=Sig), y=Inf, vjust=1) +
  scale_fill_identity() +
  facet_wrap(~symbol) +
  Rotate_x_labels +
  theme(axis.text.x = element_text(size=5), legend.position=&#39;none&#39;) +
  labs(title=&quot;Expression changes in genes of interest&quot;, y=&quot;log2FC&quot;, caption=str_wrap(&quot;P&lt;0.05 = *\nP&lt;0.001 = !\nColor and  fill transparency based on PC1 &amp; PC2 to distinguish risdiplam/branaplam axes&quot;, 30))</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-38-1">
Past versions of unnamed-chunk-38-1.png
</button>
</p>
<div id="fig-unnamed-chunk-38-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-38-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>There are a few treatments in this plot that also catch my eye in HTT
and DENND5A… Let’s look at those closer..</p>
<pre class="r"><code>diff.expression %&gt;%
  filter(symbol %in% c(&quot;HTT&quot;, &quot;DENND5A&quot;)) %&gt;%
  left_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot; | treatment %in% c(&quot;WC10&quot;, &quot;WC11&quot;, &quot;WC12&quot;), 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  mutate(Sig = case_when(
    PValue &lt; 0.001 ~ &quot;!&quot;,
    PValue &lt; 0.05 ~ &quot;*&quot;,
    TRUE ~ NA_character_
  )) %&gt;%
  ggplot(aes(x=treatment, y=logFC, fill=NewColor)) +
  geom_col(aes(alpha=alpha)) +
  geom_text(aes(label=Sig), y=Inf, vjust=1) +
  scale_fill_identity() +
  facet_wrap(~symbol) +
  Rotate_x_labels +
  theme(axis.text.x = element_text(size=5), legend.position=&#39;none&#39;) +
  labs(title=&quot;Expression changes in genes of interest&quot;, y=&quot;log2FC&quot;, caption=str_wrap(&quot;P&lt;0.05 = *\nP&lt;0.001 = !\nColor and  fill transparency based on PC1 &amp; PC2 to distinguish risdiplam/branaplam axes. C10, C11, C12 reordered by reds&quot;, 30))</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-1">
Past versions of unnamed-chunk-39-1.png
</button>
</p>
<div id="fig-unnamed-chunk-39-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-39-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Let’s repeat but just consider more systematically any DE gene found
in C05 or B12 with FDR&lt;20%</p>
<pre class="r"><code>diff.expression %&gt;%
  filter(treatment %in% c(&quot;WC05&quot;, &quot;WB12&quot;)) %&gt;%
  filter( FDR&lt;0.2) %&gt;%
  distinct(EnsemblID) %&gt;%
  left_join(diff.expression) %&gt;%
  left_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  mutate(Sig = case_when(
    PValue &lt; 0.001 ~ &quot;!&quot;,
    PValue &lt; 0.05 ~ &quot;*&quot;,
    TRUE ~ NA_character_
  )) %&gt;%
  ggplot(aes(x=treatment, y=logFC, fill=NewColor)) +
  geom_col(aes(alpha=alpha)) +
  geom_text(aes(label=Sig), y=Inf, vjust=1) +
  scale_fill_identity() +
  facet_wrap(~symbol) +
  Rotate_x_labels +
  theme(axis.text.x = element_text(size=5), legend.position=&#39;none&#39;) +
  labs(title=&quot;Expression changes in genes of interest&quot;, y=&quot;log2FC&quot;, caption=str_wrap(&quot;P&lt;0.05 = *\nP&lt;0.001 = !\nColor and  fill transparency based on PC1 &amp; PC2 to distinguish risdiplam/branaplam axes&quot;, 30))</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-40-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&quot;../code/scratch/scratch.pdf&quot;, height=8, width=12)</code></pre>
<p>Ok, based on this I’m not sure there are any astoundingly specific DE
changes with WC05 or WB12… Sure in some ways it is unique becase it may
be risdiplam like, and in some other ways it may be more branaplam like,
and there are some changes in effect sizes, but no totally new DE
discoveries.</p>
<p>I think it might be better to just report this to Jingxin before
trying to make a list of things to verify by qPCR… That doesn’t seem to
interesting to me given these results.</p>
<p>Well, actually, let’s at least do the analgous analysis but for
splicing. And if that looks at all promising, then I can consider doing
a more careful differential splicing or differential expression to
directly compare the WC05 or WB12 treatments to more risdiplam like
treatments or more branaplam like treatments to find what makes them
unique.</p>
<p><strong>Still to do that…</strong>*</p>
<p>Next thing I am curious about is following up on the WC04 effect at
the -4 and -3 position. It seemed to have a preference for CGGA|GT and
GGGA|GT that was not found in either risdiplam or branaplam like
molecules. First let me verify this is really a believable effect and
not just noise due to sampling a very low number of introns or
something.</p>
<p>First let’s see how many introns that median effect was based on and
if it is significant…</p>
<p>For sake of not making the plot too big, let’s just include a few
hand picked dinucleotides based on previous plot.</p>
<pre class="r"><code>leafcutter.effects_and_sig %&gt;%
  filter(UpstreamOfDonor2BaseSeq == &quot;GA&quot;) %&gt;%
  mutate(NNGAGT = substr(seq, 1, 2)) %&gt;%
  filter(NNGAGT %in% c(&quot;AA&quot;,&quot;AT&quot;, &quot;CA&quot;,&quot;CG&quot;, &quot;GG&quot;, &quot;TA&quot;, &quot;TG&quot;, &quot;TT&quot;)) %&gt;%
  group_by(NNGAGT, treatment) %&gt;%
  summarise(
    medLogef = median(logef),
    n=n(),
    MannWhitneyP=format.pval(wilcox.test(logef, alternative=&quot;greater&quot;)$p.value, 2)) %&gt;%
  left_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  ggplot(aes(y=treatment, x=NNGAGT, fill=medLogef)) +
  geom_raster() +
  geom_text(color=&#39;red&#39;, size=3, aes(label = paste(n, MannWhitneyP, sep=&#39;|&#39;))) +
  scale_fill_viridis_c() +
  theme(axis.text.y = element_text(size=6, colour = XaxisColors))</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-42-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-42-1">
Past versions of unnamed-chunk-42-1.png
</button>
</p>
<div id="fig-unnamed-chunk-42-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-42-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Oh ok, actually those weird C04 effects are because that was the
treatment where one of the replicates was a failed sample, that I have
been excluding from some (but mistakenly not all) analyses.</p>
<p>Let’s also plot the effects for that new bulge (guUaag) that I
hypothesized in previous experiments is affected by these molecules
(albeit to a lesser extent than gA|gu bulge).</p>
<pre class="r"><code>leafcutter.effects_and_sig %&gt;%
  dplyr::select(treatment, intron, seq, logef) %&gt;%
  pivot_wider(names_from = &quot;treatment&quot;, values_from = &quot;logef&quot;) %&gt;%
  mutate(
    `1_ncag|guaagn` = str_detect(seq, &quot;^\\wCAGGTAAG\\w&quot;),
    `2_nagA|guaagn` = str_detect(seq, &quot;^\\wAGAGTAAG\\w&quot;),
    `3_nagG|guaagn` = str_detect(seq, &quot;^\\wAGGGTAAG\\w&quot;),
    `4_nagU|guaagn` = str_detect(seq, &quot;^\\wAGTGTAAG\\w&quot;),
    # `5.1_uagA|guaagn` = str_detect(seq, &quot;^TAGAGTAAG\\w&quot;),
    # `5.2_aagA|guaagn` = str_detect(seq, &quot;^AAGAGTAAG\\w&quot;),
    `5.3_cagA|guaagn` = str_detect(seq, &quot;^CAGAGTAAG\\w&quot;),
    `5.4_augA|guaagn` = str_detect(seq, &quot;^ATGAGTAAG\\w&quot;),
    `6_nnng|guUaag` = str_detect(seq, &quot;^\\w\\w\\wGGTTAAG&quot;),
    `7_nnng|guAaag` = str_detect(seq, &quot;^\\w\\w\\wGGTAAAG&quot;),
  ) %&gt;%
  # filter_at(vars(matches(&quot;^\\d.+_.+$&quot;,perl=T)),any_vars(. == T)) %&gt;%
  gather(key=&quot;Pattern&quot;, value=&quot;IsMatch&quot;, matches(&quot;^\\d.+&quot;,perl=T)) %&gt;%
  # pull(Pattern) %&gt;% unique()
  filter(IsMatch) %&gt;%
  gather(key=&quot;treatment&quot;, value=&quot;logef&quot;, matches(&quot;^W.+&quot;)) %&gt;%
  group_by(Pattern, treatment) %&gt;%
  summarise(medLogef = median(logef, na.rm = T),
            p = format.pval(wilcox.test(logef, alternative=&quot;greater&quot;)$p.value, 2),
            n = sum(IsMatch)) %&gt;%
  ungroup() %&gt;%
  separate(Pattern, into = c(&quot;PlotOrder&quot;, &quot;Pattern&quot;), sep = &quot;_&quot;, convert=T) %&gt;%
  arrange(PlotOrder) %&gt;%
  left_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  ggplot(aes(y=treatment, x=Pattern, fill=medLogef)) +
  geom_raster() +
  geom_text(color=&#39;red&#39;, size=2, aes(label = paste(n, p, sep=&#39;|&#39;))) +
  scale_fill_gradient2(name=str_wrap(&quot;Median dose-response effect&quot;, 10)) +
  Rotate_x_labels +
  theme(axis.text.y = element_text(size=6, colour = XaxisColors))</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-43-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-43-1">
Past versions of unnamed-chunk-43-1.png
</button>
</p>
<div id="fig-unnamed-chunk-43-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-43-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, there really isn’t anything to be said of that. We just lack
power to say anything about the hypothesized effect on the guUaag bulge.
We Again, we can’t even really see the effects at the GT|GT introns
which I previously saw at high concentrations and have been documented
in other lit.</p>
<p>Ok, I think the last thing I want to try is to find the
C05/B12-specific splicing changes by doing a direct contrast with
branaplam-like and risdiplam-like molecules..</p>
<p>Let’s replot the PCA, and highlight the samples I proposed picking
for leafcutter contrasts. I’ll do 3 contrasts for each sample of
interest, labelled RisdiLike_High, RisdiLike_Med, and BranadLike_Med.
I’ll write out the leafcutter groups files for those contrasts, perform
the contrast (in the snakemake), and then explore results in later code
chunks:</p>
<pre class="r"><code>dir.create(&quot;../output/CustomLeafcutterGroupFiles&quot;)

Samples_of_interest &lt;- c(&quot;WD08&quot;, &quot;WC05&quot;, &quot;WB12&quot;)
RisdiLike_Med_Samples &lt;- c(&quot;WB06&quot;, &quot;WB09&quot;, &quot;WB10&quot;)
RisdiLike_High_Samples &lt;- c(&quot;WD05&quot;, &quot;WD07&quot;, &quot;WB09&quot;)
BranaLike_Med_Samples &lt;- c(&quot;WE04&quot;, &quot;WE02&quot;, &quot;WD10&quot;)


NewColors %&gt;%
  dplyr::select(treatment, NewColor, alpha) %&gt;%
  inner_join(pca.results.to.plot) %&gt;%
ggplot(aes(x=PC1, y=PC2, color=NewColor)) +
  geom_point(size=3, aes(alpha=alpha)) +
  geom_text_repel(data = . %&gt;%
            filter(treatment %in% c(Samples_of_interest, RisdiLike_Med_Samples)),
          aes(label=treatment)) +
  # geom_point(data = pca.results.to.plot_AdditionProjections) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nAll samples&quot;, caption=&quot;Samples in RisdiLike_Med contrast labelled&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-44-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>NewColors %&gt;%
  dplyr::select(treatment, NewColor, alpha) %&gt;%
  inner_join(pca.results.to.plot) %&gt;%
ggplot(aes(x=PC1, y=PC2, color=NewColor)) +
  geom_point(size=3, aes(alpha=alpha)) +
  geom_text_repel(data = . %&gt;%
            filter(treatment %in% c(Samples_of_interest, RisdiLike_High_Samples)),
          aes(label=treatment)) +
  # geom_point(data = pca.results.to.plot_AdditionProjections) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nAll samples&quot;, caption=&quot;Samples in RisdiLike_High contrast labelled&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-44-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>NewColors %&gt;%
  dplyr::select(treatment, NewColor, alpha) %&gt;%
  inner_join(pca.results.to.plot) %&gt;%
ggplot(aes(x=PC1, y=PC2, color=NewColor)) +
  geom_point(size=3, aes(alpha=alpha)) +
  geom_text_repel(data = . %&gt;%
            filter(treatment %in% c(Samples_of_interest, BranaLike_Med_Samples)),
          aes(label=treatment)) +
  # geom_point(data = pca.results.to.plot_AdditionProjections) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nAll samples&quot;, caption=&quot;Samples in BranaLike_Med contrast labelled&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-44-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ControlSamples &lt;- data.frame(BranaLike_Med_Samples, RisdiLike_High_Samples, RisdiLike_Med_Samples) %&gt;%
  gather(key=&quot;ContrastName&quot;, value = &quot;treatment&quot;) %&gt;%
  mutate(Group = &quot;Control&quot;) %&gt;%
  left_join(FullMetadata) %&gt;%
  dplyr::select(old.sample.name, Group, ContrastName)

for (treatment in Samples_of_interest){
  data.frame(treatment=treatment, ContrastName = c(&quot;BranaLike_Med_Samples&quot;, &quot;RisdiLike_High_Samples&quot;, &quot;RisdiLike_Med_Samples&quot;)) %&gt;%
    left_join(FullMetadata) %&gt;%
    dplyr::select(old.sample.name, ContrastName) %&gt;%
    mutate(Group = &quot;Treatment&quot;) %&gt;%
    bind_rows(ControlSamples) %&gt;%
    arrange(ContrastName, Group, old.sample.name) %&gt;%
    group_by(ContrastName) %&gt;%
    group_walk(~ write_tsv(.x, paste0(&quot;../output/CustomLeafcutterGroupFiles/Contrast_&quot;, treatment, &quot;_vs_&quot;,.y$ContrastName, &quot;groups.tsv&quot;), col_names = F))
}</code></pre>
<p>And then I added rules to the snakemake pipeline to run these
contrasts:</p>
<pre class="bash"><code>snakemake GatherCustomContrasts</code></pre>
<p>Now let’s explore the results..</p>
<pre class="r"><code>leafcutter.effects.CustomContrasts &lt;- list.files(path=&quot;../code/SplicingAnalysis/Exp52_CustomContrasts&quot;, pattern=&quot;^.+_effect_sizes.txt&quot;, full.names=T) %&gt;%
  setNames(str_replace(., &quot;../code/SplicingAnalysis/Exp52_CustomContrasts/(.+?)_effect_sizes.txt&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(fread) %&gt;%
  bind_rows(.id=&quot;Contrast&quot;) %&gt;%
  mutate(name = str_replace(intron, &quot;(.+?):(.+?):(.+?):clu.+([+-])&quot;, &quot;\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(cluster = str_replace(intron, &quot;(.+?):.+?:.+?:(clu.+[+-])&quot;, &quot;\\1:\\2&quot;))

leafcutter.sig.CustomContrasts &lt;-list.files(path=&quot;../code/SplicingAnalysis/Exp52_CustomContrasts&quot;, pattern=&quot;^.+_cluster_significance.txt&quot;, full.names=T) %&gt;%
  setNames(str_replace(., &quot;../code/SplicingAnalysis/Exp52_CustomContrasts/(.+?)_cluster_significance.txt&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(fread) %&gt;%
  bind_rows(.id=&quot;Contrast&quot;)

leafcutter.effects.Merged.CustomContrasts &lt;- leafcutter.effects.CustomContrasts %&gt;%
  inner_join(leafcutter.sig.CustomContrasts, by=c(&#39;cluster&#39;, &#39;Contrast&#39;)) %&gt;%
  left_join(Introns.annotations, by=&quot;name&quot;) %&gt;%
  mutate(
    TestTreatment = str_replace(Contrast,&quot;^Contrast_(.+?)_vs_(.+?)_Samplesgroups$&quot;, &quot;\\1&quot;),
    ControlTreatment = str_replace(Contrast,&quot;^Contrast_(.+?)_vs_(.+?)_Samplesgroups$&quot;, &quot;\\2&quot;),
    UpstreamOfDonor2BaseSeq = substr(seq, 3, 4)
  )

leafcutter.effects.Merged.CustomContrasts$TestTreatment %&gt;% unique()</code></pre>
<pre><code>[1] &quot;WB12&quot; &quot;WC05&quot; &quot;WD08&quot;</code></pre>
<pre class="r"><code>leafcutter.effects.Merged.CustomContrasts %&gt;%
  group_by(cluster, TestTreatment, ControlTreatment) %&gt;%
  mutate(ContainsGAGT = any(UpstreamOfDonor2BaseSeq==&quot;GA&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(p.adjust &lt; 0.1) %&gt;%
  distinct(cluster, TestTreatment, ControlTreatment, .keep_all=T) %&gt;%
  count(ContainsGAGT, TestTreatment, ControlTreatment) %&gt;%
  ggplot(aes(x=ControlTreatment, y=n, fill=ContainsGAGT)) +
  geom_col(position=&quot;stack&quot;) +
  facet_wrap(~TestTreatment, scales=&quot;free&quot;) +
  theme(legend.position=&#39;bottom&#39;) +
  Rotate_x_labels +
  labs(title=str_wrap(&quot;contrasts vs brana-like and risdi-like treatments&quot;, 40), y=&quot;Number significant clusters, FDR&lt;10%&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-46-1">
Past versions of unnamed-chunk-46-1.png
</button>
</p>
<div id="fig-unnamed-chunk-46-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-46-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, so there are fair number of significant differences. That was
totally expected. but of course I am particulary interseted in the GAGT
introns that are <em>more activated</em> in C05/D08/B12</p>
<pre class="r"><code>leafcutter.effects.Merged.CustomContrasts %&gt;%
  group_by(cluster, TestTreatment, ControlTreatment) %&gt;%
  mutate(ContainsGAGT = any(UpstreamOfDonor2BaseSeq==&quot;GA&quot;)) %&gt;%
  ungroup() %&gt;%
  filter(p.adjust &lt; 0.1) %&gt;%
  filter(ContainsGAGT) %&gt;%
  mutate(sign = sign(logef)) %&gt;%
  count(sign, TestTreatment, ControlTreatment) %&gt;%
  ggplot(aes(x=ControlTreatment, y=n, fill=as.factor(sign))) +
  geom_col(position=&quot;stack&quot;) +
  facet_wrap(~TestTreatment, scales=&quot;free&quot;) +
  theme(legend.position=&#39;bottom&#39;) +
  Rotate_x_labels +
  labs(title=str_wrap(&quot;contrasts vs brana-like and risdi-like treatments&quot;, 40), y=&quot;Number significant GAGT introns, FDR&lt;10%&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-47-1">
Past versions of unnamed-chunk-47-1.png
</button>
</p>
<div id="fig-unnamed-chunk-47-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-47-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, about the same number of upregulated vs downregulated introns in
each contrast. What I am really after I should say are the introns that
are upregulated in all contrasts. So I’ll filter for GA|GT introns that
are FDR&lt;10% significant in at least one contrast, and nominally
significant in all contrasts (P&lt;0.05), and with a effect size
direction favoring upregulation in the treatment of interest.</p>
<pre class="r"><code>IntronsActivatedByTreatmentsOfInterest &lt;- leafcutter.effects.Merged.CustomContrasts %&gt;%
  filter(UpstreamOfDonor2BaseSeq == &quot;GA&quot;) %&gt;%
  group_by(intron, TestTreatment) %&gt;%
  filter(any(p.adjust &lt; 0.1)) %&gt;%
  filter(all(p &lt; 0.05)) %&gt;%
  filter(all(logef &gt; 0)) %&gt;%
  filter(all(deltapsi &gt; 0.05)) %&gt;%
  ungroup()

IntronsActivatedByTreatmentsOfInterest %&gt;%
  dplyr::select(intron, Treatment, Control, TestTreatment, ControlTreatment ) %&gt;%
  group_by(TestTreatment, intron) %&gt;%
  mutate(MeanTreatment = mean(Treatment)) %&gt;%
  distinct(intron, TestTreatment, ControlTreatment, .keep_all=T) %&gt;%
  dplyr::select(-Treatment) %&gt;%
  pivot_wider(names_from = &quot;ControlTreatment&quot;, values_from=&quot;Control&quot;) %&gt;%
  gather(key=&quot;Group&quot;, value=&quot;IntronicPSI&quot;, -c(&quot;intron&quot;, &quot;TestTreatment&quot;)) %&gt;%
  mutate(Group= relevel(factor(Group), ref=&quot;MeanTreatment&quot;)) %&gt;%
  mutate(label = paste(TestTreatment, intron, sep=&#39;\n&#39;)) %&gt;%
  ggplot(aes(x=Group, y=IntronicPSI, fill=Group)) +
  geom_col() +
  facet_wrap(~label, scales=&quot;free&quot;) +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        strip.text.x = element_text(size = 5)) +
  labs(title=&quot;GA-GT introns most specific to samples of interest&quot;, caption=str_wrap(&quot;DeltaPSI&gt;5% in all contrasts, FDR&lt;10% in at least one contrast, P&lt;0.05 in all contrasts&quot;, 40))</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-48-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-48-1">
Past versions of unnamed-chunk-48-1.png
</button>
</p>
<div id="fig-unnamed-chunk-48-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-48-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ggsave(&quot;../code/scratch/scratch.pdf&quot;, height=6, width=12)</code></pre>
<p>Ok these are the kinds of introns I’m looking for, that Jingxin can
consider validating as specific to these three molecules of interest.
Let’s look the original data for each sample for these sets of introns…
And then double check in the genome browser.</p>
<pre class="r"><code>IntronsActivatedByTreatmentsOfInterest.vec &lt;- IntronsActivatedByTreatmentsOfInterest %&gt;% pull(intron) %&gt;% unique()

AllSamplesInContrasts &lt;- c(Samples_of_interest, RisdiLike_Med_Samples, RisdiLike_High_Samples, BranaLike_Med_Samples)


PSI %&gt;%
  filter(junc %in% IntronsActivatedByTreatmentsOfInterest.vec) %&gt;%
  dplyr::select(junc, contains(&quot;NewMolecule&quot;)) %&gt;%
  gather(key = &quot;leafcutter.name&quot;, value=&quot;IntronicPSI&quot;, -junc) %&gt;%
  left_join(FullMetadata, by=&quot;leafcutter.name&quot;) %&gt;%
  inner_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot; | treatment %in% c(&quot;WC10&quot;, &quot;WC11&quot;, &quot;WC12&quot;), 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  filter(treatment %in% AllSamplesInContrasts) %&gt;%
  # sample_n_of(5, junc) %&gt;%
  ggplot(aes(x=treatment, y=IntronicPSI, color=NewColor)) +
  geom_text(aes(alpha=alpha, label=treatment), size=3) +
  facet_wrap(~junc, scales=&quot;free&quot;) +
  scale_color_identity() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        strip.text.x = element_text(size = 5),
        legend.position=&#39;none&#39;) +
  labs(title=&quot;Candidate treatment-of-interest-specific effects&quot;, x=&quot;Treatments, roughly in order&quot;, y=&quot;Intronic PSI&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-49-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-49-1">
Past versions of unnamed-chunk-49-1.png
</button>
</p>
<div id="fig-unnamed-chunk-49-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-49-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, I guess that seems reasonable based on the previous plots. I
don’t think there is a bug in my code.</p>
<p>Ok, let’s include all samples in these plots now, to see to what
extent these candidates are just specific to the molecule of
interest.</p>
<pre class="r"><code>PSI %&gt;%
  filter(junc %in% IntronsActivatedByTreatmentsOfInterest.vec) %&gt;%
  dplyr::select(junc, contains(&quot;NewMolecule&quot;)) %&gt;%
  gather(key = &quot;leafcutter.name&quot;, value=&quot;IntronicPSI&quot;, -junc) %&gt;%
  left_join(FullMetadata, by=&quot;leafcutter.name&quot;) %&gt;%
  inner_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot; | treatment %in% c(&quot;WC10&quot;, &quot;WC11&quot;, &quot;WC12&quot;), 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  # sample_n_of(5, junc) %&gt;%
  ggplot(aes(x=treatment, y=IntronicPSI, color=NewColor)) +
  geom_text(aes(alpha=alpha, label=treatment), size=3) +
  facet_wrap(~junc, scales=&quot;free&quot;) +
  scale_color_identity() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        strip.text.x = element_text(size = 5),
        legend.position=&#39;none&#39;) +
    labs(title=&quot;Candidate treatment-of-interest-specific effects&quot;, x=&quot;Treatments, roughly in order&quot;, y=&quot;Intronic PSI&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-50-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-50-1">
Past versions of unnamed-chunk-50-1.png
</button>
</p>
<div id="fig-unnamed-chunk-50-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/e996344ad0d4841a485c9f6f500e259daacdc96b/docs/figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-50-1.png" target="_blank">e996344</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-02-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ok, there are a few D08 specific effects, but perhaps that isn’t
surprising given previous plots that show so many effects in this
particular treatment.</p>
<p>In any case, as promised, I’ll write out some of the most promising
things, including FHOD3, as junctions worthy of validation… Maybe the
better approach for identifying the worthy candidate list this is just
to forget these specific contrasts, and look at the DE plots I made, and
intersect with potential GA|GT intron drivers…</p>
<pre class="r"><code>GenesToHighlight &lt;- c(&quot;HSPA1L&quot;, &quot;CRYL1&quot;, &quot;FHOD3&quot;, &quot;ZFYVE26&quot;)

diff.expression %&gt;%
  filter(symbol %in% GenesToHighlight) %&gt;%
  left_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  mutate(Sig = case_when(
    PValue &lt; 0.001 ~ &quot;!&quot;,
    PValue &lt; 0.05 ~ &quot;*&quot;,
    TRUE ~ NA_character_
  )) %&gt;%
  ggplot(aes(x=treatment, y=logFC, fill=NewColor)) +
  geom_col(aes(alpha=alpha)) +
  geom_text(aes(label=Sig), y=Inf, vjust=1) +
  scale_fill_identity() +
  facet_wrap(~symbol) +
  Rotate_x_labels +
  theme(axis.text.x = element_text(size=5), legend.position=&#39;none&#39;) +
  labs(title=&quot;Expression changes in genes of interest&quot;, y=&quot;log2FC&quot;, caption=str_wrap(&quot;P&lt;0.05 = *\nP&lt;0.001 = !\nColor and  fill transparency based on PC1 &amp; PC2 to distinguish risdiplam/branaplam axes&quot;, 30))</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-51-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>These 4 genes are examples of how these red treatments may sometimes
be more branaplam like on some genes, and more risdiplam like on others.
Now let’s try and find the GA-GT introns drivers for these genes…</p>
<p>For convenience let’s look from the data table I wrote out from
estimating EC50s from the dose response experiment</p>
<pre class="r"><code>GA.GT.introns_FromDoseResponseExperiment &lt;- read_tsv(&quot;../output/EC50Estimtes.FromPSI.txt.gz&quot;)

GA.GT.introns_FromDoseResponseExperiment %&gt;%
  separate_rows(gene_names, sep=&quot;,&quot;) %&gt;%
  filter(gene_names %in% GenesToHighlight)</code></pre>
<pre><code># A tibble: 1 × 34
  junc        `#Chrom`  start    end gid   strand.y seq   Donor.score gene_names
  &lt;chr&gt;       &lt;chr&gt;     &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt;       &lt;dbl&gt; &lt;chr&gt;     
1 chr13:2040… chr13    2.04e7 2.04e7 chr1… -        GAGA…        5.45 CRYL1     
# … with 25 more variables: gene_ids &lt;chr&gt;, SpliceDonor &lt;chr&gt;,
#   UpstreamSpliceAcceptor &lt;chr&gt;, IntronType &lt;chr&gt;, Steepness &lt;dbl&gt;,
#   LowerLimit &lt;dbl&gt;, UpperLimit &lt;dbl&gt;, ED50_Branaplam &lt;dbl&gt;, ED50_C2C5 &lt;dbl&gt;,
#   ED50_Risdiplam &lt;dbl&gt;, spearman.coef.Branaplam &lt;dbl&gt;,
#   spearman.coef.C2C5 &lt;dbl&gt;, spearman.coef.Risdiplam &lt;dbl&gt;,
#   EC.Ratio.Test.Estimate_Branaplam.C2C5 &lt;dbl&gt;,
#   EC.Ratio.Test.Estimate_Branaplam.Risdiplam &lt;dbl&gt;, …</code></pre>
<p>Ok actually that didn’t find much… And according to that previous
data, CRYL1 GA-GT intron as actually more risdiplam-specific… weird. Ok,
let’s just go back to trying to find the GA-GT intron from the
leafcutter results…</p>
<pre class="r"><code>leafcutter.effects_and_sig %&gt;%
  filter(UpstreamOfDonor2BaseSeq==&quot;GA&quot;) %&gt;%
  separate_rows(gene_names, sep=&quot;,&quot;) %&gt;%
  filter(gene_names %in% GenesToHighlight) %&gt;%
  distinct(intron, gene_names) %&gt;%
  kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">intron</th>
<th align="left">gene_names</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">chr13:20404741:20405904:clu_31084_-</td>
<td align="left">CRYL1</td>
</tr>
<tr class="even">
<td align="left">chr18:36742468:36742737:clu_40538_+</td>
<td align="left">FHOD3</td>
</tr>
</tbody>
</table>
<pre class="r"><code>leafcutter.effects_and_sig %&gt;%
  filter(UpstreamOfDonor2BaseSeq==&quot;GA&quot;) %&gt;%
  separate_rows(gene_names, sep=&quot;,&quot;) %&gt;%
  filter(gene_names %in% GenesToHighlight) %&gt;%
  # distinct(intron, .keep_all=T) %&gt;%
  dplyr::select(intron, treatment_PSI, DMSO_PSI, treatment, gene_names) %&gt;%
  group_by(intron) %&gt;%
  mutate(DMSO_PSI = mean(DMSO_PSI)) %&gt;%
  ungroup() %&gt;%
  pivot_wider(names_from=&quot;treatment&quot;, values_from=&quot;treatment_PSI&quot;, names_glue=&quot;{treatment}_PSI&quot;) %&gt;%
  dplyr::select(intron, gene_names, everything()) %&gt;%
  gather(key=&quot;treatment&quot;, value=&quot;IntronPSI&quot;, -c(&quot;gene_names&quot;, &quot;intron&quot;)) %&gt;%
  mutate(treatment = str_replace(treatment, &quot;^(.+?)_PSI&quot;, &quot;\\1&quot;)) %&gt;%
  mutate(label=paste(intron, gene_names, sep=&#39;\n&#39;)) %&gt;%
  left_join(NewColors) %&gt;%
  mutate(PC2 = if_else(NewColor == &quot;#e34a33&quot;, 0, PC2)) %&gt;%
  mutate(treatment = fct_reorder(treatment, PC2 * alpha)) %&gt;%
  ggplot(aes(x=treatment, y=IntronPSI, fill=NewColor)) +
  geom_col(aes(alpha=alpha)) +
  scale_fill_identity() +
  facet_wrap(~label, scales=&quot;free&quot;) +
  Rotate_x_labels +
  theme(legend.position=&#39;none&#39;, axis.text.x = element_text(size=5))</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-53-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, so I recovered the same CRYL1 intron, and also the FHOD3 intron.
Still couldn’t find the intron for the other two genes… I think I am
satisfied enough just to say these two introns might be good to validate
molecule-specific effects, and also probably any number of the
risdiplam-specific or branaplam-specific introns that I previously
shared from the dose-response data.</p>
<p>Also, looking at the genome browser of these two introns listed above
it becomes obvious what the exon coordinates are if you were to design
qPCR probes for validation… FHOD3 is particularly interesting… Much like
I previously observed in HTT, the actual intron with the GA-GT 5’ss
seems to be mostly retained in the treatment - it is the upstream
cryptic intron that more obviously gets excised in the treatment. This
suggests that the treatment may aid in U1 binding and activation of exon
definition without necessarily activating the 5’ss bound to the
molecule. This may also somewhat explain my previous observation that
these molecules induce much cryptic 3’ss that aren’t in a leafcutter
cluster with a cryptic 5’ss.</p>
<p>Here is an screenshot of what I mean.</p>
<p>[<img src="assets/FHOD3_Example" alt="FHOD3" />]</p>
<div id="pca-on-subset-of-scaffolds" class="section level3">
<h3>PCA on subset of scaffolds</h3>
<p>Jingxin shared a powerpoint in which he color coded samples based on
their scaffolds… Using these and some of my custom observations, I wrote
factors for each treatment. I’m going to save that data replot PCA a few
different ways for exploration.</p>
<pre class="r"><code>TreatmentFactors &lt;- read_csv(&quot;../code/DataNotToCommit/CompoundsAndManualAnnotationFactors.csv&quot;) %&gt;%
  mutate(treatment = paste0(&quot;W&quot;, Well))</code></pre>
<p>First let me replot the PCA the way I have been plotting it
previously with colors consistent with previous plots</p>
<pre class="r"><code>pca.results.to.plot %&gt;%
  left_join(
    NewColors %&gt;%
      dplyr::select(NewColor, alpha, treatment)) %&gt;%
ggplot(aes(x=PC1, y=PC2, color=NewColor, alpha=alpha)) +
  geom_text(size=2, aes(label=treatment)) +
  scale_color_identity() +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nAll samples&quot;, caption=&quot;Colored consistent in other plots&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-55-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca.results.to.plot %&gt;%
  left_join(TreatmentFactors) %&gt;%
ggplot(aes(x=PC1, y=PC2, color=Factor)) +
  geom_text(size=2, aes(label=treatment)) +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nAll samples&quot;, caption=&quot;Colored by structural groupings from Jingxins&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-55-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca.results.to.plot %&gt;%
  left_join(TreatmentFactors) %&gt;%
ggplot(aes(x=PC3, y=PC4, color=Factor)) +
  geom_text(size=2, aes(label=treatment)) +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nAll samples&quot;, caption=&quot;Colored by structural groupings from Jingxins&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-55-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, one thing I want to try is to repeat the PCA but only exclude the
branaplam-based molecules</p>
<pre class="r"><code>NonBranaplamBasedMolecules &lt;- TreatmentFactors %&gt;%
  filter(!Factor==&quot;G&quot;) %&gt;% pull(treatment)

SamplesToInclude &lt;- 
  FullMetadata %&gt;%
  filter(Experiment %in% c(&quot;Single high dose LCL&quot;)) %&gt;%
  filter(!old.sample.name == &quot;NewMolecule.C04-2&quot;) %&gt;%
  filter(treatment %in% c(&quot;DMSO&quot;, NonBranaplamBasedMolecules)) %&gt;%
  pull(leafcutter.name)
  

pca.results &lt;- PSI.IntronsToInclude %&gt;%
  dplyr::select(junc, all_of(SamplesToInclude)) %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% prcomp(scale=T)


pca.results.to.plot &lt;- pca.results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;leafcutter.name&quot;) %&gt;%
  dplyr::select(leafcutter.name, PC1:PC6) %&gt;%
  left_join(FullMetadata %&gt;%
              dplyr::select(leafcutter.name, treatment)) %&gt;%
  left_join(TreatmentFactors, by=&quot;treatment&quot;)

pca.results.to.plot %&gt;%
  mutate(ToHighlight = if_else(
    treatment %in% c(&quot;WB12&quot;, &quot;WC05&quot;, &quot;WD08&quot;),
    1,
    0.5
    )) %&gt;%
ggplot(aes(x=PC1, y=PC2, color=Factor, alpha=ToHighlight)) +
  geom_text(size=3, aes(label=treatment)) +
  scale_alpha_continuous(range=c(0.5,1)) +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nNon branaplam skeletons&quot;, caption=&quot;Colored by structural groupings from Jingxins&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-56-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca.results.to.plot %&gt;%
  mutate(ToHighlight = if_else(
    treatment %in% c(&quot;WB12&quot;, &quot;WC05&quot;, &quot;WD08&quot;),
    1,
    0.5
    )) %&gt;%
ggplot(aes(x=PC3, y=PC4, color=Factor, alpha=ToHighlight)) +
  geom_text(size=3, aes(label=treatment)) +
  scale_alpha_continuous(range=c(0.5,1)) +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nNon branaplam skeletons&quot;, caption=&quot;Colored by structural groupings from Jingxin&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-56-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca.results.to.plot %&gt;%
  mutate(ToHighlight = if_else(
    treatment %in% c(&quot;WB12&quot;, &quot;WC05&quot;, &quot;WD08&quot;),
    1,
    0.5
    )) %&gt;%
ggplot(aes(x=PC5, y=PC6, color=Factor, alpha=ToHighlight)) +
  geom_text(size=3, aes(label=treatment)) +
  scale_alpha_continuous(range=c(0.5,1)) +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nNon branaplam skeletons&quot;, caption=&quot;Colored by structural groupings from Jingxin&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-56-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>And let’s do the same for branaplam based molecules</p>
<pre class="r"><code>BranaplamBasedMolecules &lt;- TreatmentFactors %&gt;%
  filter(Factor==&quot;G&quot;) %&gt;% pull(treatment)

SamplesToInclude &lt;- 
  FullMetadata %&gt;%
  filter(Experiment %in% c(&quot;Single high dose LCL&quot;)) %&gt;%
  filter(!old.sample.name == &quot;NewMolecule.C04-2&quot;) %&gt;%
  filter(treatment %in% c(&quot;DMSO&quot;, BranaplamBasedMolecules)) %&gt;%
  pull(leafcutter.name)
  

pca.results &lt;- PSI.IntronsToInclude %&gt;%
  dplyr::select(junc, all_of(SamplesToInclude)) %&gt;%
  column_to_rownames(&quot;junc&quot;) %&gt;%
  scale() %&gt;% t() %&gt;% prcomp(scale=T)


pca.results.to.plot &lt;- pca.results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;leafcutter.name&quot;) %&gt;%
  dplyr::select(leafcutter.name, PC1:PC6) %&gt;%
  left_join(FullMetadata %&gt;%
              dplyr::select(leafcutter.name, treatment)) %&gt;%
  left_join(TreatmentFactors, by=&quot;treatment&quot;)

pca.results.to.plot %&gt;%
  mutate(ToHighlight = if_else(
    treatment %in% c(&quot;WB12&quot;, &quot;WC05&quot;, &quot;WD08&quot;),
    1,
    0.5
    )) %&gt;%
ggplot(aes(x=PC1, y=PC2, color=Factor, alpha=ToHighlight)) +
  geom_text(size=3, aes(label=treatment)) +
  scale_alpha_continuous(range=c(0.5,1)) +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nNon branaplam skeletons&quot;, caption=&quot;Colored by structural groupings from Jingxins&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-57-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca.results.to.plot %&gt;%
  mutate(ToHighlight = if_else(
    treatment %in% c(&quot;WB12&quot;, &quot;WC05&quot;, &quot;WD08&quot;),
    1,
    0.5
    )) %&gt;%
ggplot(aes(x=PC3, y=PC4, color=Factor, alpha=ToHighlight)) +
  geom_text(size=3, aes(label=treatment)) +
  scale_alpha_continuous(range=c(0.5,1)) +
  theme_bw() +
  labs(title = &quot;PCA using 632 upregulated GAGT introns\nNon branaplam skeletons&quot;, caption=&quot;Colored by structural groupings from Jingxin&quot;)</code></pre>
<p><img src="figure/20220221_ExploreExpOf52.Rmd/unnamed-chunk-57-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>I’m re-questioning whether C05 and B12 are special in any worthwhile
way</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] knitr_1.39         ggrepel_0.9.1      gplots_3.1.3       Heatplus_3.4.0    
 [5] qvalue_2.28.0      edgeR_3.38.4       limma_3.52.4       data.table_1.14.2 
 [9] RColorBrewer_1.1-3 forcats_0.5.1      stringr_1.4.0      dplyr_1.0.9       
[13] purrr_0.3.4        readr_2.1.2        tidyr_1.2.0        tibble_3.1.7      
[17] ggplot2_3.3.6      tidyverse_1.3.1   

loaded via a namespace (and not attached):
 [1] bitops_1.0-7       fs_1.5.2           bit64_4.0.5        lubridate_1.8.0   
 [5] httr_1.4.3         rprojroot_2.0.3    tools_4.2.0        backports_1.4.1   
 [9] bslib_0.3.1        utf8_1.2.2         R6_2.5.1           KernSmooth_2.23-20
[13] DBI_1.1.2          colorspace_2.0-3   withr_2.5.0        tidyselect_1.1.2  
[17] bit_4.0.4          compiler_4.2.0     git2r_0.30.1       cli_3.3.0         
[21] rvest_1.0.2        xml2_1.3.3         labeling_0.4.2     sass_0.4.1        
[25] caTools_1.18.2     scales_1.2.0       digest_0.6.29      R.utils_2.11.0    
[29] rmarkdown_2.14     pkgconfig_2.0.3    htmltools_0.5.2    dbplyr_2.1.1      
[33] fastmap_1.1.0      highr_0.9          rlang_1.0.2        readxl_1.4.0      
[37] rstudioapi_0.13    jquerylib_0.1.4    generics_0.1.2     farver_2.1.0      
[41] jsonlite_1.8.0     gtools_3.9.2       vroom_1.5.7        R.oo_1.24.0       
[45] magrittr_2.0.3     Rcpp_1.0.8.3       munsell_0.5.0      fansi_1.0.3       
[49] R.methodsS3_1.8.1  lifecycle_1.0.1    stringi_1.7.6      whisker_0.4       
[53] yaml_2.3.5         plyr_1.8.7         grid_4.2.0         parallel_4.2.0    
[57] promises_1.2.0.1   crayon_1.5.1       lattice_0.20-45    haven_2.5.0       
[61] splines_4.2.0      hms_1.1.1          locfit_1.5-9.7     pillar_1.7.0      
[65] reshape2_1.4.4     reprex_2.0.1       glue_1.6.2         evaluate_0.15     
[69] modelr_0.1.8       vctrs_0.4.1        tzdb_0.3.0         httpuv_1.6.5      
[73] cellranger_1.1.0   gtable_0.3.0       assertthat_0.2.1   xfun_0.30         
[77] broom_0.8.0        later_1.3.0        viridisLite_0.4.0  workflowr_1.7.0   
[81] ellipsis_0.3.2    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
