<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-11-02" />

<title>2023-11-02_3MoleculesOfInterestLeafcutterDs</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">small molecular splicing RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">2023-11-02_3MoleculesOfInterestLeafcutterDs</h1>
<h4 class="date">2023-11-02</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#intro" id="toc-intro">Intro</a></li>
<li><a href="#analysis" id="toc-analysis">Analysis</a></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-11-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>20211209_JingxinRNAseq/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed19900924code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(19900924)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed19900924code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(19900924)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunSmallMoleculeRNASeqtree3c2aa401dcb108044b2382821578f792b0b42897targetblank3c2aa40a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/tree/3c2aa401dcb108044b2382821578f792b0b42897" target="_blank">3c2aa40</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunSmallMoleculeRNASeqtree3c2aa401dcb108044b2382821578f792b0b42897targetblank3c2aa40a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/tree/3c2aa401dcb108044b2382821578f792b0b42897" target="_blank">3c2aa40</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/20220707_TitrationSeries_DE_testing.nb.html
    Ignored:    code/.DS_Store
    Ignored:    code/._DOCK7.pdf
    Ignored:    code/._DOCK7_DMSO1.pdf
    Ignored:    code/._DOCK7_SM2_1.pdf
    Ignored:    code/._FKTN_DMSO_1.pdf
    Ignored:    code/._FKTN_SM2_1.pdf
    Ignored:    code/._MAPT.pdf
    Ignored:    code/._PKD1_DMSO_1.pdf
    Ignored:    code/._PKD1_SM2_1.pdf
    Ignored:    code/.snakemake/
    Ignored:    code/1KG_HighCoverageCalls.samplelist.txt
    Ignored:    code/5ssSeqs.tab
    Ignored:    code/Alignments/
    Ignored:    code/Branaplam_Risdiplam_specific_introns.bed.gz
    Ignored:    code/Branaplam_Risdiplam_specific_introns.bed.gz.tbi
    Ignored:    code/ChemCLIP/
    Ignored:    code/ClinVar/
    Ignored:    code/DE_testing/
    Ignored:    code/DE_tests.mat.counts.gz
    Ignored:    code/DE_tests.txt.gz
    Ignored:    code/DataNotToCommit/
    Ignored:    code/DonorMotifSearches/
    Ignored:    code/DoseResponseData/
    Ignored:    code/Fastq/
    Ignored:    code/FastqFastp/
    Ignored:    code/FivePrimeSpliceSites.txt
    Ignored:    code/FragLenths/
    Ignored:    code/Ishigami/
    Ignored:    code/Meme/
    Ignored:    code/Multiqc/
    Ignored:    code/OMIM/
    Ignored:    code/OldBigWigs/
    Ignored:    code/PhyloP/
    Ignored:    code/QC/
    Ignored:    code/ReferenceGenomes/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/Session2.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TracksSession
    Ignored:    code/bigwigs/
    Ignored:    code/featureCounts/
    Ignored:    code/figs/
    Ignored:    code/fimo_out/
    Ignored:    code/geena/
    Ignored:    code/hg38ToMm39.over.chain.gz
    Ignored:    code/igv_session.template.xml
    Ignored:    code/igv_session.xml
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/rstudio-server.job
    Ignored:    code/rules/.ExpMoleculesOfInterest.smk.swp
    Ignored:    code/scratch/
    Ignored:    code/temp/
    Ignored:    code/test.txt.gz
    Ignored:    code/testPlottingWithMyScript.ForJingxin.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin2.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin3.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin4.sh
    Ignored:    code/testPlottingWithMyScript.sh
    Ignored:    code/tracks.with_chRNA.RisdiBrana.xml
    Ignored:    code/tracks.with_chRNA.RisdiBranaWithExtras.xml
    Ignored:    code/tracks.xml
    Ignored:    data/~$52CompoundsTempPlateLayoutForPipettingConvenience.xlsx

Untracked files:
    Untracked:  analysis/2023-11-02_3MoleculesOfInterestLeafcutterDs.Rmd
    Untracked:  code/config/samples.3MoleculesOfInterestExperiment.Contrasts.tsv
    Untracked:  code/scripts/Exp202310_MakeLeafcutterContrasts.R

Unstaged changes:
    Modified:   analysis/2023-10-31_Explore3MoleculesOfInterest.Rmd
    Modified:   code/rules/ExpMoleculesOfInterest.smk
    Modified:   code/rules/common.smk
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>I’ve processed each of three doses for each of three molecules as a
leafcutter contrast with 5 DMSO samples… Previously in the firboblast
experiment, I noticed a correlation between effect size and different
types of NN in NNGA|GT introns… I will similarly look for that here with
these three new molecules.</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<p>load libs</p>
<pre class="r"><code>library(tidyverse)
library(RColorBrewer)
library(data.table)
library(gplots)
library(ggrepel)

# define some useful funcs
sample_n_of &lt;- function(data, size, ...) {
  dots &lt;- quos(...)
  
  group_ids &lt;- data %&gt;% 
    group_by(!!! dots) %&gt;% 
    group_indices()
  
  sampled_groups &lt;- sample(unique(group_ids), size)
  
  data %&gt;% 
    filter(group_ids %in% sampled_groups)
}


# Set theme
theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family=&quot;Helvetica&quot;)))

# I use layer a lot, to rotate long x-axis labels
Rotate_x_labels &lt;- theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))</code></pre>
<p>Read in data</p>
<pre class="r"><code>juncs &lt;- fread(&quot;../code/SplicingAnalysis/FullSpliceSiteAnnotations/JuncfilesMerged.annotated.basic.bed.gz&quot;) %&gt;%
  mutate(Junc = paste(chrom, start, end, strand, sep=&quot;_&quot;))
donors &lt;- fread(&quot;../code/SplicingAnalysis/FullSpliceSiteAnnotations/JuncfilesMerged.annotated.basic.bed.5ss.tab.gz&quot;, col.names = c(&quot;Junc.Donor&quot;, &quot;seq&quot;, &quot;DonorScore&quot;)) %&gt;%
  separate(Junc.Donor, into=c(&quot;Junc&quot;, &quot;Donor&quot;), sep=&quot;::&quot;)

Cluster_sig &lt;- 
  Sys.glob(&quot;../code/SplicingAnalysis/Exp202310_3Molecules_Contrasts/*_cluster_significance.txt&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/SplicingAnalysis/Exp202310_3Molecules_Contrasts/(.+?)_cluster_significance.txt&quot;, &quot;\\1&quot;)) %&gt;%
  append(
    Sys.glob(&quot;../code/SplicingAnalysis/leafcutter/differential_splicing/*_cluster_significance.txt&quot;) %&gt;%
      grep(&quot;ExpOf52&quot;, ., invert=T, value=T) %&gt;%
      setNames(str_replace(., &quot;../code/SplicingAnalysis/leafcutter/differential_splicing/(.+?)_cluster_significance.txt&quot;, &quot;\\1&quot;))
  ) %&gt;%
  lapply(fread) %&gt;%
  bind_rows(.id=&quot;Contrast&quot;)

effect_sizes &lt;- Sys.glob(&quot;../code/SplicingAnalysis/Exp202310_3Molecules_Contrasts/*_effect_sizes.txt&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/SplicingAnalysis/Exp202310_3Molecules_Contrasts/(.+?)_effect_sizes.txt&quot;, &quot;\\1&quot;)) %&gt;%
  append(
    Sys.glob(&quot;../code/SplicingAnalysis/leafcutter/differential_splicing/*_effect_sizes.txt&quot;) %&gt;%
      grep(&quot;ExpOf52&quot;, ., invert=T, value=T) %&gt;%
      setNames(str_replace(., &quot;../code/SplicingAnalysis/leafcutter/differential_splicing/(.+?)_effect_sizes.txt&quot;, &quot;\\1&quot;))
  ) %&gt;%
  lapply(fread, col.names=c(&quot;intron&quot;, &quot;logef&quot;, &quot;DMSO&quot;, &quot;treatment&quot;, &quot;deltapsi&quot;)) %&gt;%
  bind_rows(.id=&quot;Contrast&quot;) %&gt;%
  mutate(Junc = str_replace(intron, &quot;^(.+?):(.+?):(.+?):clu_.+?_([+-])$&quot;, &quot;\\1_\\2_\\3_\\4&quot;)) %&gt;%
  mutate(cluster = str_replace(intron, &quot;^(.+?):.+?:.+?:(clu_.+?_[+-])$&quot;, &quot;\\1:\\2&quot;)) %&gt;%
  inner_join(
    inner_join(juncs, donors)
  ) %&gt;%
  inner_join(Cluster_sig) %&gt;%
  mutate(CellType = if_else(str_detect(Contrast, &quot;_&quot;), &quot;LCL&quot;, &quot;Fibroblast&quot;)) %&gt;%
  mutate(logef = if_else(CellType==&quot;Fibroblast&quot;, -1*logef, logef)) %&gt;%
  mutate(logef = if_else(CellType==&quot;Fibroblast&quot;, -1*deltapsi, deltapsi)) %&gt;%
  mutate(NN__GU = str_extract(seq, &quot;^\\w{2}&quot;)) %&gt;%
  mutate(NNGU = substr(seq, 3, 4)) </code></pre>
<pre class="r"><code>effect_sizes %&gt;%
  filter(p.adjust &lt; 0.01) %&gt;%
  ggplot(aes(x=logef, color=Contrast)) +
  stat_ecdf() +
  geom_vline(xintercept = 0, linetype=&quot;dashed&quot;) +
  geom_vline(
    data = . %&gt;%
      group_by(NNGU, Contrast) %&gt;%
      summarise(med=median(logef, na.rm=T)),
    aes(xintercept=med, color=Contrast),
    linetype=&#39;dashed&#39;
  ) +
  coord_cartesian(xlim=c(-0.5, 0.5)) +
  ylab(&quot;CumulativeFraction&quot;) +
  facet_wrap(~NNGU) +
  theme_light() +
  ggtitle(&quot;Effect sizes for NN|GT introns, grouped by NN&quot;) +
  labs(caption=&quot;Only included introns in FDR&lt;0.01 significant clusters&quot;)</code></pre>
<p><img src="figure/2023-11-02_3MoleculesOfInterestLeafcutterDs.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>effect_sizes %&gt;%
  # filter(p.adjust &lt; 0.01) %&gt;%
  filter(NNGU==&quot;GA&quot;) %&gt;%
  ggplot(aes(x=logef, color=Contrast)) +
  stat_ecdf() +
  geom_vline(xintercept = 0, linetype=&quot;dashed&quot;) +
  geom_vline(
    data = . %&gt;%
      group_by(NN__GU, Contrast) %&gt;%
      summarise(med=median(logef, na.rm=T)),
    aes(xintercept=med, color=Contrast),
    linetype=&#39;dashed&#39;
  ) +
  coord_cartesian(xlim=c(-0.5, 0.5)) +
  ylab(&quot;CumulativeFraction&quot;) +
  facet_wrap(~NN__GU) +
  theme_light() +
  ggtitle(&quot;Effect sizes for NNGA|GT introns, grouped by NN&quot;) +
  labs(caption=&quot;All tested GA|GU&quot;)</code></pre>
<p><img src="figure/2023-11-02_3MoleculesOfInterestLeafcutterDs.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Make same plot with less samples… There’s a lot of clutter…</p>
<pre class="r"><code>Colors &lt;- effect_sizes %&gt;%
  distinct(Contrast) %&gt;%
  mutate(Colors = recode(Contrast, SMSM27_3000=&quot;#3182bd&quot;, SMSM27_600=&quot;#9ecae1&quot;, SMSM27_120=&quot;#deebf7&quot;, SMSM32_120=&quot;#fee0d2&quot;, SMSM32_3000=&quot;#de2d26&quot;, SMSM32_600=&quot;#fc9272&quot;, SMSM6_10000=&quot;#a65628&quot;, SMSM6_2000=&quot;#a65628&quot;, SMSM6_50000=&quot;#a65628&quot;, A10=&quot;#7fc97f&quot;, C2=&quot;#beaed4&quot;, CUOME=&quot;#fdc086&quot;, SM2=&quot;#ffff99&quot;)) %&gt;%
  dplyr::select(Contrast, Colors) %&gt;% deframe()

Colors.Subset &lt;- effect_sizes %&gt;%
  distinct(Contrast) %&gt;%
  filter(Contrast %in% c(&quot;SM2&quot;, &quot;C2&quot;, &quot;SMSM27_3000&quot;, &quot;SMSM27_600&quot;, &quot;SMSM32_3000&quot;, &quot;SMSM32_600&quot;, &quot;SMSM6_50000&quot;)) %&gt;%
  mutate(Colors = recode(Contrast, SMSM6_50000=&quot;#b15928&quot;, SMSM27_3000=&quot;#e31a1c&quot;, SMSM27_600=&quot;#fb9a99&quot;, SMSM32_3000=&quot;#ff7f00&quot;, SMSM32_600=&quot;#fdbf6f&quot;, C2=&quot;#377eb8&quot;, SM2=&quot;#4daf4a&quot;)) %&gt;%
  dplyr::select(Contrast, Colors) %&gt;% deframe()

effect_sizes %&gt;%
  # filter(p.adjust &lt; 0.01) %&gt;%
  filter(Contrast %in% names(Colors.Subset)) %&gt;%
  filter(NNGU==&quot;GA&quot;) %&gt;%
  group_by(NN__GU) %&gt;%
  mutate(Unique_Elements = n_distinct(intron)) %&gt;%   # Now summarise with unique elements per group
  ungroup() %&gt;%
  mutate(FacetGroup = str_glue(&quot;{NN__GU}; n={Unique_Elements}&quot;)) %&gt;%
  ggplot(aes(x=logef, color=Contrast)) +
  stat_ecdf() +
  geom_vline(xintercept = 0, linetype=&quot;dashed&quot;) +
  geom_vline(
    data = . %&gt;%
      group_by(FacetGroup, Contrast) %&gt;%
      summarise(med=median(logef, na.rm=T)),
    aes(xintercept=med, color=Contrast),
    linetype=&#39;dashed&#39;
  ) +
  scale_color_manual(values=Colors.Subset, drop=T) +
  coord_cartesian(xlim=c(-0.1, 0.1)) +
  ylab(&quot;CumulativeFraction&quot;) +
  facet_wrap(~FacetGroup) +
  theme_light() +
  ggtitle(&quot;Effect sizes for NNGA|GT introns, grouped by NN&quot;) +
  labs(caption=&quot;All leafcutter-tested GA|GU&quot;)</code></pre>
<p><img src="figure/2023-11-02_3MoleculesOfInterestLeafcutterDs.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>So first recall this plot I have made in the past, where the x-axis
is the spearman dose:PSI correlation coefficient, sort of a proxy for
effects on different NN sub-categories of NNGA|GU introns. branaplam has
stronger effects on GA, CA, and TA… but AA looks more or less the same
between risdiplam/branaplam… AT maybe looks a bit risdiplam specific…
This btw is the same sequence in that risdiplam-specific sequence in
HSD17B4 which has two poison exons. In any case, now look at the second
plot, where the x-axis is the logef (from leafcutter) for introns with
these different groups… I also included some data from the fibroblast
experiment: SM2 (colored green, because it is a branaplam scaffold) and
C2 (colored blue because it is a risdiplam scaffold). Again just looking
at SM2 and C2, the relative effects between classes are mostly
consistent with the dose titration experiment with risdiplam and
branaplam… Anyway, now looking at SMSM27 and SMSM32 (the red and
orange), the effects look stronger than C2 for the GA and CA facets…
Again, its tricky to compare, because the effective genome-wide dose
isn’t exactly the same between experiments, and the cell type is
different… But I figure that by looking across different facets you can
sort of control for those things. So from that, I do feel in some ways
these molecules do have some slight branaplam-like-character moreso than
risdiplam.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggrepel_0.9.1      gplots_3.1.3       data.table_1.14.2  RColorBrewer_1.1-3
 [5] forcats_0.5.1      stringr_1.4.0      dplyr_1.0.9        purrr_0.3.4       
 [9] readr_2.1.2        tidyr_1.2.0        tibble_3.1.7       ggplot2_3.3.6     
[13] tidyverse_1.3.1   

loaded via a namespace (and not attached):
 [1] httr_1.4.3         sass_0.4.1         jsonlite_1.8.0     R.utils_2.11.0    
 [5] modelr_0.1.8       gtools_3.9.2       bslib_0.3.1        assertthat_0.2.1  
 [9] highr_0.9          cellranger_1.1.0   yaml_2.3.5         pillar_1.7.0      
[13] backports_1.4.1    glue_1.6.2         digest_0.6.29      promises_1.2.0.1  
[17] rvest_1.0.2        colorspace_2.0-3   htmltools_0.5.2    httpuv_1.6.5      
[21] R.oo_1.24.0        pkgconfig_2.0.3    broom_0.8.0        haven_2.5.0       
[25] scales_1.2.0       later_1.3.0        tzdb_0.3.0         git2r_0.30.1      
[29] generics_0.1.2     farver_2.1.0       ellipsis_0.3.2     withr_2.5.0       
[33] cli_3.3.0          magrittr_2.0.3     crayon_1.5.1       readxl_1.4.0      
[37] evaluate_0.15      R.methodsS3_1.8.1  fs_1.5.2           fansi_1.0.3       
[41] xml2_1.3.3         tools_4.2.0        hms_1.1.1          lifecycle_1.0.1   
[45] munsell_0.5.0      reprex_2.0.1       compiler_4.2.0     jquerylib_0.1.4   
[49] caTools_1.18.2     rlang_1.0.2        grid_4.2.0         rstudioapi_0.13   
[53] bitops_1.0-7       labeling_0.4.2     rmarkdown_2.14     gtable_0.3.0      
[57] DBI_1.1.2          R6_2.5.1           lubridate_1.8.0    knitr_1.39        
[61] fastmap_1.1.0      utf8_1.2.2         workflowr_1.7.0    rprojroot_2.0.3   
[65] KernSmooth_2.23-20 stringi_1.7.6      Rcpp_1.0.8.3       vctrs_0.4.1       
[69] dbplyr_2.1.1       tidyselect_1.1.2   xfun_0.30         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
