<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-10-26" />

<title>2023-10-26_ExploreDrmModels</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">small molecular splicing RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">2023-10-26_ExploreDrmModels</h1>
<h4 class="date">2023-10-26</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#load-libs" id="toc-load-libs">Load libs</a></li>
<li><a href="#intro" id="toc-intro">Intro</a></li>
<li><a href="#analysis" id="toc-analysis">Analysis</a></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-10-26
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>20211209_JingxinRNAseq/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed19900924code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(19900924)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed19900924code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(19900924)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunSmallMoleculeRNASeqtreec3a42e31ee9d4e949fa6f9ea4e1a9773bac73efetargetblankc3a42e3a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/tree/c3a42e31ee9d4e949fa6f9ea4e1a9773bac73efe" target="_blank">c3a42e3</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunSmallMoleculeRNASeqtreec3a42e31ee9d4e949fa6f9ea4e1a9773bac73efetargetblankc3a42e3a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/tree/c3a42e31ee9d4e949fa6f9ea4e1a9773bac73efe" target="_blank">c3a42e3</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/20220707_TitrationSeries_DE_testing.nb.html
    Ignored:    code/.DS_Store
    Ignored:    code/._DOCK7.pdf
    Ignored:    code/._DOCK7_DMSO1.pdf
    Ignored:    code/._DOCK7_SM2_1.pdf
    Ignored:    code/._FKTN_DMSO_1.pdf
    Ignored:    code/._FKTN_SM2_1.pdf
    Ignored:    code/._MAPT.pdf
    Ignored:    code/._PKD1_DMSO_1.pdf
    Ignored:    code/._PKD1_SM2_1.pdf
    Ignored:    code/.snakemake/
    Ignored:    code/1KG_HighCoverageCalls.samplelist.txt
    Ignored:    code/5ssSeqs.tab
    Ignored:    code/Alignments/
    Ignored:    code/Branaplam_Risdiplam_specific_introns.bed.gz
    Ignored:    code/Branaplam_Risdiplam_specific_introns.bed.gz.tbi
    Ignored:    code/ChemCLIP/
    Ignored:    code/ClinVar/
    Ignored:    code/DE_testing/
    Ignored:    code/DE_tests.mat.counts.gz
    Ignored:    code/DE_tests.txt.gz
    Ignored:    code/DataNotToCommit/
    Ignored:    code/DonorMotifSearches/
    Ignored:    code/DoseResponseData/
    Ignored:    code/Fastq/
    Ignored:    code/FastqFastp/
    Ignored:    code/FivePrimeSpliceSites.txt
    Ignored:    code/FragLenths/
    Ignored:    code/Ishigami/
    Ignored:    code/Meme/
    Ignored:    code/Multiqc/
    Ignored:    code/OMIM/
    Ignored:    code/OldBigWigs/
    Ignored:    code/PhyloP/
    Ignored:    code/QC/
    Ignored:    code/ReferenceGenomes/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/Session2.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TracksSession
    Ignored:    code/bigwigs/
    Ignored:    code/featureCounts/
    Ignored:    code/figs/
    Ignored:    code/fimo_out/
    Ignored:    code/geena/
    Ignored:    code/hg38ToMm39.over.chain.gz
    Ignored:    code/igv_session.template.xml
    Ignored:    code/igv_session.xml
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/rstudio-server.job
    Ignored:    code/rules/.RNASeqProcessing.smk.swp
    Ignored:    code/scratch/
    Ignored:    code/test.txt.gz
    Ignored:    code/testPlottingWithMyScript.ForJingxin.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin2.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin3.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin4.sh
    Ignored:    code/testPlottingWithMyScript.sh
    Ignored:    code/tracks.with_chRNA.RisdiBrana.xml
    Ignored:    code/tracks.with_chRNA.RisdiBranaWithExtras.xml
    Ignored:    code/tracks.xml
    Ignored:    data/~$52CompoundsTempPlateLayoutForPipettingConvenience.xlsx

Untracked files:
    Untracked:  analysis/2023-10-26_ExploreDrmModels.Rmd
    Untracked:  analysis/20230322_CandidateGUUAAG_bulges.Rmd
    Untracked:  analysis/20230418_ExploreIshigamiGEO.Rmd
    Untracked:  analysis/20230502_PCAPlotsForYang.Rmd
    Untracked:  analysis/20230915_TopHitsAcrossDoses.Rmd
    Untracked:  analysis/20230927_AGT_MechanismSearch.Rmd
    Untracked:  analysis/20231006_SearchForCausalJuncs.Rmd
    Untracked:  analysis/20231013_UnbiasedMotifSearch.Rmd
    Untracked:  code/envs/r_2.yml
    Untracked:  code/rules/ExploreIshigami.smk
    Untracked:  code/scripts/Fit_GeneExpressionModels.R
    Untracked:  code/scripts/Fit_SplicingModels.R
    Untracked:  code/scripts/MakeDonorSitesForMeme.R
    Untracked:  code/scripts/TidyAndSpearmanCorTitrationData_2.R
    Untracked:  output/SperamanCoeffs.MaxJuncForEach.GAGTDonors.bed
    Untracked:  output/tracks.with_chRNA.RisdiBranaWithExtras.xml

Unstaged changes:
    Modified:   analysis/20221108_MakeFigsFor10minPresentation.Rmd
    Modified:   code/Snakefile
    Modified:   code/rules/BptMotifSeach.smk
    Modified:   code/rules/RNASeqProcessing.smk
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-libs" class="section level2">
<h2>Load libs</h2>
<pre class="r"><code>library(tidyverse)
library(RColorBrewer)
library(data.table)
library(drc)
library(ggrepel)

theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family=&quot;Helvetica&quot;)))</code></pre>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>There are many ways to fit models with <code>drc::drm</code>… See <a
href="https://rstats4ag.org/dose-response-curves.html">this
tutorial</a>…</p>
<p>I have noticed some funny things, in that depending on exactly how
you call the <code>drm</code> function, the lower and upper limits
estimates are sometimes switched (with corresponding sign change in
slope parameter), and sometimes you get wildly different estimates… I
kind of know what I am looking for inuitively, and I think I have an
idea of how exactly I want to call <code>drm</code> to fit dose response
curves to gene expression. I’m going to demonstrate some of these
nuances here… But first for some terminology…. the 4 parameter log
logistic model has for parameters: b, c, d, and e…</p>
<ul>
<li>b is the “slope” parameter</li>
<li>c is the lower limit parameter. This may correspond to the limit
when dose=0, or the limit when dose=Infinity.</li>
<li>d is the upper limit parameter. Again, this is just the “other”
limit compared to c… The same model can be specificied by swapping c and
d, and also changing the sign of b</li>
<li>e is the ED50 parameter.</li>
</ul>
<p>When I fit dose response curves to gene expression, I figured the
following details would make the model most useful and easily
interpretable:</p>
<p>Response is in units of log2FC relative to DMSO… So I will use the
<span class="math inline">\(log2CPM_d - mean(log2CPM_{DMSO})\)</span> as
the response metric (where <span
class="math inline">\(log2CPM_d\)</span> denotes the log2
CountsPerMillion for a gene at dose <span
class="math inline">\(d\)</span>). Thus, one of the limit parameters
should be fixed at 0. For best modelling, it makes sense to limit
parameters when reasonable, so I will also fit each treatment
simultaneously such that I can limit the other limit parameter to be the
same amongst all 3 treatments. The ED50 and slope parameter can freely
vary between treatments, as I want to capture examples like HTT
manual/eyeball interpretation of the dose response data looks like the
slope is similar between all three treatments, but the ED50 is shifted.
And in the HSD17B4 example, the slope looks clearly different between
treatments, and the underlying reason makes sense (in that at the
splicing level, there are either 2 or 1 poison exons depending on the
treatment, and the gene expression response is logically sort of like
the complement of the product of the poison exon dose response curves).
Also, for simplicity/interpretability, rather than use the actual
nanomolar dose for each treatment, I will rescale the doses of branaplam
and C2C5 to be in units that are roughly functionally equivalent to the
nanomolar dose for risdiplam. For example, since C2C5 is about 10x more
potent than risdiplam, I will use 10*nanomolarDose for C2C5.</p>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<p>Below, I will use some example data and show what happens when I try
to fit these models different ways. In theory, a lot of these models
should converge to the same thing, but for whatever reason they
don’t.</p>
<pre class="r"><code>f_in &lt;-&quot;../code/DoseResponseData/LCL/TidyExpressionDoseData_logFCTransformedAndAllDMSORepsInEachSeries.txt.gz&quot;

sample_n_of &lt;- function(data, size, ...) {
  dots &lt;- quos(...)

  group_ids &lt;- data %&gt;%
    group_by(!!! dots) %&gt;%
    group_indices()

  sampled_groups &lt;- sample(unique(group_ids), size)

  data %&gt;%
    filter(group_ids %in% sampled_groups)
}

expression.dat &lt;- fread(f_in) %&gt;%
  group_by(treatment) %&gt;%
  mutate(doseRank = dense_rank(dose.nM)) %&gt;%
  ungroup() %&gt;%
  as_tibble() %&gt;%
  mutate(doseInRisdiscale = case_when(
    treatment == &quot;C2C5&quot; ~ dose.nM * 10,
    treatment == &quot;Branaplam&quot; ~ dose.nM * sqrt(10),
    TRUE ~ dose.nM
  ))</code></pre>
<p>Plot dose response data for some genes of intrest</p>
<pre class="r"><code>GenesToHighlight &lt;- c(&quot;STAT1&quot;, &quot;HTT&quot;, &quot;MYB&quot;,  &quot;TRIM11&quot;, &quot;TRAFD1&quot;, &quot;VEGFA&quot;, &quot;FBXW11&quot;, &quot;HSD17B4&quot;)

expression.dat %&gt;%
  mutate(treatment = factor(treatment)) %&gt;%
  filter(hgnc_symbol %in% GenesToHighlight) %&gt;%
  ggplot(aes(x=doseRank, y=log2FC, color=treatment)) +
  geom_point() +
  geom_line() +
  facet_wrap(~hgnc_symbol, scale=&quot;free_y&quot;)</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now fit models for each of these in three ways, which should all in
effect be identical:</p>
<pre class="r"><code>for (gene in GenesToHighlight){
  print(&quot;#######################################################################&quot;)
  data &lt;- expression.dat %&gt;%
    mutate(treatment = factor(treatment)) %&gt;%
    filter(hgnc_symbol == gene)
  print(paste(&quot;Fitting&quot;, gene, &quot;model with LL.4, fix upper limit&quot;))
  tryCatch(expr={
    fit.LL.4.Fixed.d &lt;- drm(formula = log2FC ~ doseInRisdiscale,
                 data = data,
                 fct = LL.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;), fixed=c(NA,NA,0,NA)),
                 curveid = treatment,
                 pmodels=data.frame(treatment, 1, treatment, treatment),
                 robust = &quot;mean&quot;)
    message(&quot;Successfully fitted model:&quot;)
    plot(fit.LL.4.Fixed.d)
    summary(fit.LL.4.Fixed.d)
  },
  error=function(e){
      cat(&quot;ERROR:&quot;, gene, &quot; model LL.4 fixed upper limit\n&quot;, conditionMessage(e))
    }
  )

  print(paste(&quot;Fitting&quot;, gene, &quot;model with LL.4, fix lower limit&quot;))
  tryCatch(expr={
    fit.LL.4.Fixed.c &lt;- drm(formula = log2FC ~ doseInRisdiscale,
                 data = data,
                 fct = LL.4(names=c(&quot;Steepness&quot;, &quot;LowerLimit&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;), fixed=c(NA,0,NA,NA)),
                 curveid = treatment,
                 pmodels=data.frame(treatment, treatment, 1, treatment),
                 robust = &quot;mean&quot;)
    message(&quot;Successfully fitted model.&quot;)
    plot(fit.LL.4.Fixed.c)
    summary(fit.LL.4.Fixed.c)
  },
  error=function(e){
      cat(&quot;ERROR:&quot;, gene, &quot; model LL.4 fixed lower limit\n&quot;, conditionMessage(e))
    }
  )
  
  print(paste(&quot;Fitting&quot;, gene, &quot;model with LL.3&quot;))
  tryCatch(expr={
    fit.LL.3 &lt;- drm(formula = log2FC ~ doseInRisdiscale,
                 data = data,
                 fct = LL.3(names=c(&quot;Steepness&quot;, &quot;UpperLimit&quot;, &quot;ED50&quot;), fixed=c(NA,NA,NA)),
                 curveid = treatment,
                 pmodels=data.frame(treatment, 1, treatment),
                 robust = &quot;mean&quot;)
    message(&quot;Successfully fitted model.&quot;)
    plot(fit.LL.3)
    summary(fit.LL.3)
  },
  error=function(e){
      cat(&quot;ERROR:&quot;, gene, &quot; model LL.3\n&quot;, conditionMessage(e))
    }
  )
}</code></pre>
<pre><code>[1] &quot;#######################################################################&quot;
[1] &quot;Fitting STAT1 model with LL.4, fix upper limit&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Fitting STAT1 model with LL.4, fix lower limit&quot;
Error in optim(startVec, opfct, hessian = TRUE, method = optMethod, control = list(maxit = maxIt,  : 
  non-finite finite-difference value [7]
ERROR: STAT1  model LL.4 fixed lower limit
 Convergence failed[1] &quot;Fitting STAT1 model with LL.3&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;#######################################################################&quot;
[1] &quot;Fitting HTT model with LL.4, fix upper limit&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Fitting HTT model with LL.4, fix lower limit&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Fitting HTT model with LL.3&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;#######################################################################&quot;
[1] &quot;Fitting MYB model with LL.4, fix upper limit&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Fitting MYB model with LL.4, fix lower limit&quot;
Error in optim(startVec, opfct, hessian = TRUE, method = optMethod, control = list(maxit = maxIt,  : 
  non-finite finite-difference value [7]
ERROR: MYB  model LL.4 fixed lower limit
 Convergence failed[1] &quot;Fitting MYB model with LL.3&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;#######################################################################&quot;
[1] &quot;Fitting TRIM11 model with LL.4, fix upper limit&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Fitting TRIM11 model with LL.4, fix lower limit&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Fitting TRIM11 model with LL.3&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;#######################################################################&quot;
[1] &quot;Fitting TRAFD1 model with LL.4, fix upper limit&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Fitting TRAFD1 model with LL.4, fix lower limit&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Fitting TRAFD1 model with LL.3&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-13.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;#######################################################################&quot;
[1] &quot;Fitting VEGFA model with LL.4, fix upper limit&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Fitting VEGFA model with LL.4, fix lower limit&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-15.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Fitting VEGFA model with LL.3&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;#######################################################################&quot;
[1] &quot;Fitting FBXW11 model with LL.4, fix upper limit&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-17.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Fitting FBXW11 model with LL.4, fix lower limit&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Fitting FBXW11 model with LL.3&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-19.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;#######################################################################&quot;
[1] &quot;Fitting HSD17B4 model with LL.4, fix upper limit&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-20.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;Fitting HSD17B4 model with LL.4, fix lower limit&quot;
Error in optim(startVec, opfct, hessian = TRUE, method = optMethod, control = list(maxit = maxIt,  : 
  non-finite finite-difference value [7]
ERROR: HSD17B4  model LL.4 fixed lower limit
 Convergence failed[1] &quot;Fitting HSD17B4 model with LL.3&quot;</code></pre>
<p><img src="figure/2023-10-26_ExploreDrmModels.Rmd/unnamed-chunk-4-21.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggrepel_0.9.1      drc_3.0-1          MASS_7.3-56        data.table_1.14.2 
 [5] RColorBrewer_1.1-3 forcats_0.5.1      stringr_1.4.0      dplyr_1.0.9       
 [9] purrr_0.3.4        readr_2.1.2        tidyr_1.2.0        tibble_3.1.7      
[13] ggplot2_3.3.6      tidyverse_1.3.1   

loaded via a namespace (and not attached):
 [1] fs_1.5.2          lubridate_1.8.0   httr_1.4.3        rprojroot_2.0.3  
 [5] tools_4.2.0       backports_1.4.1   bslib_0.3.1       utf8_1.2.2       
 [9] R6_2.5.1          DBI_1.1.2         colorspace_2.0-3  withr_2.5.0      
[13] tidyselect_1.1.2  compiler_4.2.0    git2r_0.30.1      cli_3.3.0        
[17] rvest_1.0.2       xml2_1.3.3        sandwich_3.0-1    labeling_0.4.2   
[21] sass_0.4.1        scales_1.2.0      mvtnorm_1.1-3     digest_0.6.29    
[25] rmarkdown_2.14    R.utils_2.11.0    pkgconfig_2.0.3   htmltools_0.5.2  
[29] plotrix_3.8-2     highr_0.9         dbplyr_2.1.1      fastmap_1.1.0    
[33] rlang_1.0.2       readxl_1.4.0      rstudioapi_0.13   farver_2.1.0     
[37] jquerylib_0.1.4   generics_0.1.2    zoo_1.8-10        jsonlite_1.8.0   
[41] gtools_3.9.2      R.oo_1.24.0       car_3.1-1         magrittr_2.0.3   
[45] Matrix_1.5-3      Rcpp_1.0.8.3      munsell_0.5.0     fansi_1.0.3      
[49] abind_1.4-5       lifecycle_1.0.1   R.methodsS3_1.8.1 stringi_1.7.6    
[53] multcomp_1.4-19   yaml_2.3.5        carData_3.0-5     grid_4.2.0       
[57] promises_1.2.0.1  crayon_1.5.1      lattice_0.20-45   haven_2.5.0      
[61] splines_4.2.0     hms_1.1.1         knitr_1.39        pillar_1.7.0     
[65] codetools_0.2-18  reprex_2.0.1      glue_1.6.2        evaluate_0.15    
[69] modelr_0.1.8      vctrs_0.4.1       tzdb_0.3.0        httpuv_1.6.5     
[73] cellranger_1.1.0  gtable_0.3.0      assertthat_0.2.1  xfun_0.30        
[77] broom_0.8.0       later_1.3.0       survival_3.3-1    workflowr_1.7.0  
[81] TH.data_1.1-1     ellipsis_0.3.2   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
