<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-03-06" />

<title>Check introns with unique specificities</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">small molecular splicing RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Check introns with unique
specificities</h1>
<h4 class="date">2023-03-06</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#recheck-list-of-risdibrana-specific-introns"
id="toc-recheck-list-of-risdibrana-specific-introns">Recheck list of
risdi/brana-specific introns</a></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-07
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>20211209_JingxinRNAseq/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed19900924code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(19900924)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed19900924code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(19900924)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkunSmallMoleculeRNASeqtreeea36e65d3d24bcefa896febc6457716cfc0c3f13targetblankea36e65a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/tree/ea36e65d3d24bcefa896febc6457716cfc0c3f13" target="_blank">ea36e65</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkunSmallMoleculeRNASeqtreeea36e65d3d24bcefa896febc6457716cfc0c3f13targetblankea36e65a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/tree/ea36e65d3d24bcefa896febc6457716cfc0c3f13" target="_blank">ea36e65</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/20220707_TitrationSeries_DE_testing.nb.html
    Ignored:    code/.DS_Store
    Ignored:    code/._DOCK7.pdf
    Ignored:    code/._DOCK7_DMSO1.pdf
    Ignored:    code/._DOCK7_SM2_1.pdf
    Ignored:    code/._FKTN_DMSO_1.pdf
    Ignored:    code/._FKTN_SM2_1.pdf
    Ignored:    code/._MAPT.pdf
    Ignored:    code/._PKD1_DMSO_1.pdf
    Ignored:    code/._PKD1_SM2_1.pdf
    Ignored:    code/.snakemake/
    Ignored:    code/1KG_HighCoverageCalls.samplelist.txt
    Ignored:    code/5ssSeqs.tab
    Ignored:    code/Alignments/
    Ignored:    code/Branaplam_Risdiplam_specific_introns.bed.gz
    Ignored:    code/Branaplam_Risdiplam_specific_introns.bed.gz.tbi
    Ignored:    code/ChemCLIP/
    Ignored:    code/ClinVar/
    Ignored:    code/DE_testing/
    Ignored:    code/DE_tests.mat.counts.gz
    Ignored:    code/DE_tests.txt.gz
    Ignored:    code/DataNotToCommit/
    Ignored:    code/DoseResponseData/
    Ignored:    code/Fastq/
    Ignored:    code/FastqFastp/
    Ignored:    code/FragLenths/
    Ignored:    code/Meme/
    Ignored:    code/Multiqc/
    Ignored:    code/OMIM/
    Ignored:    code/OldBigWigs/
    Ignored:    code/PhyloP/
    Ignored:    code/QC/
    Ignored:    code/ReferenceGenomes/
    Ignored:    code/Session.vim
    Ignored:    code/Session2.vim
    Ignored:    code/SplicingAnalysis/
    Ignored:    code/TracksSession
    Ignored:    code/bigwigs/
    Ignored:    code/featureCounts/
    Ignored:    code/figs/
    Ignored:    code/geena/
    Ignored:    code/hg38ToMm39.over.chain.gz
    Ignored:    code/igv_session.template.xml
    Ignored:    code/igv_session.xml
    Ignored:    code/log
    Ignored:    code/logs/
    Ignored:    code/rstudio-server.job
    Ignored:    code/scratch/
    Ignored:    code/test.txt.gz
    Ignored:    code/testPlottingWithMyScript.ForJingxin.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin2.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin3.sh
    Ignored:    code/testPlottingWithMyScript.ForJingxin4.sh
    Ignored:    code/testPlottingWithMyScript.sh
    Ignored:    code/tracks.xml
    Ignored:    data/~$52CompoundsTempPlateLayoutForPipettingConvenience.xlsx
    Ignored:    output/._PioritizedIntronTargets.pdf

Unstaged changes:
    Modified:   analysis/20230306_CheckIntronsWithUniqueSpecifities.Rmd
    Modified:   code/scripts/GenometracksByGenotype

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/20230306_CheckIntronsWithUniqueSpecifities.Rmd</code>)
and HTML
(<code>docs/20230306_CheckIntronsWithUniqueSpecifities.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/ea36e65d3d24bcefa896febc6457716cfc0c3f13/analysis/20230306_CheckIntronsWithUniqueSpecifities.Rmd" target="_blank">ea36e65</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-07
</td>
<td>
added igv session
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/bfairkun/SmallMoleculeRNASeq/ea36e65d3d24bcefa896febc6457716cfc0c3f13/docs/20230306_CheckIntronsWithUniqueSpecifities.html" target="_blank">ea36e65</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-07
</td>
<td>
added igv session
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)

library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<div id="recheck-list-of-risdibrana-specific-introns"
class="section level3">
<h3>Recheck list of risdi/brana-specific introns</h3>
<p>I previously made a list of risdiplam/branaplam/C2C5-specific introns
based on the dose response experiment. Because the different molecules
have different genome-wide effective potencies (ie, 100nM C2C5 is
different than 100nM risdiplam, despite them having generally similar
specificity profiles), I tested for a difference in the EC50 of each
intron for each traitment pair, relative to the genome-wide median. You
can see that in a <a href="20220913_TidyDataForJingxin.html">previous
notebook</a> I plotted the dose-response datapoints for a random sample
of 20 of these significant molecule-specific effects, and they are
generally believable effects from those.</p>
<pre class="r"><code>GA.GT.Introns &lt;- read_tsv(&quot;../output/EC50Estimtes.FromPSI.txt.gz&quot;)

ColorKey &lt;- c(&quot;Branaplam-specific&quot;=&quot;#005A32&quot;, &quot;Risdiplam-specific&quot;=&quot;#084594&quot;, &quot;Non significant&quot;=&quot;#969696&quot;)

P1.dat &lt;- GA.GT.Introns %&gt;%
  mutate(Color = case_when(
    EC.Ratio.Test.Estimate.FDR_Branaplam.Risdiplam &gt; 0.05 ~ &quot;Non significant&quot;,
    ECRatio.ComparedToGenomewideMedian_Branaplam.Risdiplam &gt; 1 ~ &quot;Risdiplam-specific&quot;,
    ECRatio.ComparedToGenomewideMedian_Branaplam.Risdiplam &lt; 1 ~ &quot;Branaplam-specific&quot;
  ))

ggplot(P1.dat, aes(x=ED50_Branaplam, y=ED50_Risdiplam, color=Color)) +
  geom_point() +
  scale_x_continuous(trans=&#39;log10&#39;, limits = c(1, 1E7)) +
  scale_y_continuous(trans=&#39;log10&#39;, limits = c(1, 1E7)) +
  scale_color_manual(values=ColorKey) +
  labs(x= &quot;Branaplam\nED50 (nanomolar)&quot;, y= &quot;Risdiplam\nED50 (nanomolar)&quot;, color=NULL) +
  theme(legend.position=&#39;bottom&#39;) +
  theme_bw()</code></pre>
<p><img src="figure/20230306_CheckIntronsWithUniqueSpecifities.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/bfairkun/SmallMoleculeRNASeq/blob/ea36e65d3d24bcefa896febc6457716cfc0c3f13/docs/figure/20230306_CheckIntronsWithUniqueSpecifities.Rmd/unnamed-chunk-1-1.png" target="_blank">ea36e65</a>
</td>
<td>
Benjmain Fair
</td>
<td>
2023-03-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>P1.dat %&gt;%
  count(Color)</code></pre>
<pre><code># A tibble: 4 × 2
  Color                  n
  &lt;chr&gt;              &lt;int&gt;
1 Branaplam-specific   240
2 Non significant      200
3 Risdiplam-specific   161
4 &lt;NA&gt;                2878</code></pre>
<p>Ok, so each point is a branaplam-specific or risdiplam-specific GA-GT
intron. There are 200 non-significant points, 161 risdiplam-specific
points, and 240 branaplam-specific points. I think it will be useful for
Jingxin to be able to easily visualize these before validating them…
Some introns that change PSI from 1% to 3% might be very significant,
but not the best for validation. Furthermore, not all of these events
might be involved in poison exon inclusion. Some of them might just be
alt 5’ss or something. And these details might inform which are the best
candidates for validation by flanking PCR primers. Also, when looking at
the data, some of the ‘risdiplam-specific’ introns I called are just not
believable effects. Like the dose-response data points might just be too
noisy. I did a reasonable job of filtering these out, and trying to make
the FDR estimate reasonably calibrated… Or like there are some examples
where it appears ‘risdiplam-specific’ but it is because the cryptic
event is only just barely visible at the highest dose of risdiplam, and
not visible in any of the branaplam doses (so it just requires a really
really high dose to see the effect). So for all these reasons, it will
be best to check the raw coverage data first… So I’ll write the out this
list as a bedfile for easy viewing in IGV, a save a session that has the
bedfile loaded, along with bigwigs for the dose response experiment so
hopefully it will be easy to view.</p>
<pre class="r"><code>P1.dat %&gt;%
  dplyr::select(chrom=`#Chrom`, start, end, junc, score=ECRatio.ComparedToGenomewideMedian_Branaplam.Risdiplam, strand=strand.y, Color) %&gt;%
  filter(!is.na(score)) %&gt;%
  mutate(thickStart = start, thickEnd=end, Color=recode(Color, !!!ColorKey)) %&gt;%
  mutate(RgbCol = apply(col2rgb(Color), 2, paste, collapse=&#39;,&#39;)) %&gt;%
  dplyr::select(chrom:strand, thickStart, thickEnd, RgbCol) %&gt;%
  arrange(chrom, start, end) %&gt;%
  write_tsv(&quot;../code/Branaplam_Risdiplam_specific_introns.bed&quot;, col_names = F)</code></pre>
<pre class="bash"><code>bgzip ../code/Branaplam_Risdiplam_specific_introns.bed
tabix -p bed ../code/Branaplam_Risdiplam_specific_introns.bed.gz</code></pre>
<p>So to view, you will need to use the globus link I shared with
Jingxin to download the <code>code</code> directory, or at minimum
download the <code>code/bigwigs</code> folder of bigwig files, the
<code>code/tracks.xml</code> IGV session, and the
<code>code/Branaplam_Risdiplam_specific_introns.bed.gz</code> and
<code>code/Branaplam_Risdiplam_specific_introns.bed.gz.tbi</code>. Then,
from IGV, go to File &gt;&gt; OpenSession and open the tracks.xml. The
tracks.xml uses relative filepaths so you need to preserve the relative
filepaths of the thing you downloaded (that is, they need to all be in
the same folder, such as a folder named <code>code</code>).</p>
<p>Now you should be able to select the
<code>Branaplam_Risdiplam_specific_introns.bed</code> track, and Ctl-F
or Ctl-B to jump to the next GA-GT introns, colored according to
significance. Browse around to find coordinates of potential splice
events for validation.</p>
<p>Here are screenshots of a few examples…</p>
<div class="figure">
<img src="assets/BranaSpecificExample1.png" alt="" />
<p class="caption">BranaSpecific1</p>
</div>
<div class="figure">
<img src="assets/BranaSpecificExample2.png" alt="" />
<p class="caption">BranaSpecific2</p>
</div>
<div class="figure">
<img src="assets/RisdiSpecificExample1.png" alt="" />
<p class="caption">RisdiSpecific1</p>
</div>
<p>In general, I’m noticing there a quite a few believable
branaplam-specific examples but the risdlam-specific are for the most
part very small specificity effect sizes. That is, there are very few
obviously risdiplam-specific events, compared to branaplam-specific
events.</p>
<p>Let’s list the top risdi-specific introns in the following table:</p>
<pre class="r"><code>P1.dat %&gt;%
  arrange(desc(ECRatio.ComparedToGenomewideMedian_Branaplam.Risdiplam)) %&gt;%
  head() %&gt;%
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="4%" />
<col width="0%" />
<col width="1%" />
<col width="1%" />
<col width="2%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="2%" />
<col width="1%" />
<col width="2%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="2%" />
<col width="2%" />
<col width="2%" />
<col width="4%" />
<col width="4%" />
<col width="4%" />
<col width="5%" />
<col width="6%" />
<col width="5%" />
<col width="4%" />
<col width="5%" />
<col width="4%" />
<col width="4%" />
<col width="5%" />
<col width="4%" />
<col width="2%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">junc</th>
<th align="left">#Chrom</th>
<th align="right">start</th>
<th align="right">end</th>
<th align="left">gid</th>
<th align="left">strand.y</th>
<th align="left">seq</th>
<th align="right">Donor.score</th>
<th align="left">gene_names</th>
<th align="left">gene_ids</th>
<th align="left">SpliceDonor</th>
<th align="left">UpstreamSpliceAcceptor</th>
<th align="left">IntronType</th>
<th align="right">Steepness</th>
<th align="right">LowerLimit</th>
<th align="right">UpperLimit</th>
<th align="right">ED50_Branaplam</th>
<th align="right">ED50_C2C5</th>
<th align="right">ED50_Risdiplam</th>
<th align="right">spearman.coef.Branaplam</th>
<th align="right">spearman.coef.C2C5</th>
<th align="right">spearman.coef.Risdiplam</th>
<th align="right">EC.Ratio.Test.Estimate_Branaplam.C2C5</th>
<th align="right">EC.Ratio.Test.Estimate_Branaplam.Risdiplam</th>
<th align="right">EC.Ratio.Test.Estimate_C2C5.Risdiplam</th>
<th align="right">ECRatio.ComparedToGenomewideMedian_Branaplam.C2C5</th>
<th
align="right">ECRatio.ComparedToGenomewideMedian_Branaplam.Risdiplam</th>
<th align="right">ECRatio.ComparedToGenomewideMedian_C2C5.Risdiplam</th>
<th align="right">EC.Ratio.Test.Estimate.P_Branaplam.C2C5</th>
<th align="right">EC.Ratio.Test.Estimate.P_Branaplam.Risdiplam</th>
<th align="right">EC.Ratio.Test.Estimate.P_C2C5.Risdiplam</th>
<th align="right">EC.Ratio.Test.Estimate.FDR_Branaplam.C2C5</th>
<th align="right">EC.Ratio.Test.Estimate.FDR_Branaplam.Risdiplam</th>
<th align="right">EC.Ratio.Test.Estimate.FDR_C2C5.Risdiplam</th>
<th align="left">Color</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">chr2:72179461:72182894:clu_4568_-</td>
<td align="left">chr2</td>
<td align="right">72179461</td>
<td align="right">72182894</td>
<td align="left">chr2_clu_4568_-</td>
<td align="left">-</td>
<td align="left">GAGAGTAAGTA</td>
<td align="right">6.005894</td>
<td align="left">EXOC6B</td>
<td align="left">ENSG00000144036.16</td>
<td align="left">chr2.72182894.-</td>
<td align="left">chr2.72182903.-</td>
<td align="left">Annotated</td>
<td align="right">-2.7570544</td>
<td align="right">3.1361062</td>
<td align="right">31.384932</td>
<td align="right">2.495499e+09</td>
<td align="right">248.12468</td>
<td align="right">785.0629479</td>
<td align="right">0.2277339</td>
<td align="right">0.5594024</td>
<td align="right">0.9139077</td>
<td align="right">1.005744e+07</td>
<td align="right">3178724.1044</td>
<td align="right">0.3160571</td>
<td align="right">7.482724e+06</td>
<td align="right">25547101.480</td>
<td align="right">3.0610287</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="left">Risdiplam-specific</td>
</tr>
<tr class="even">
<td align="left">chr13:20404741:20405904:clu_25569_-</td>
<td align="left">chr13</td>
<td align="right">20404741</td>
<td align="right">20405904</td>
<td align="left">chr13_clu_25569_-</td>
<td align="left">-</td>
<td align="left">GAGAGTAAGTC</td>
<td align="right">5.446634</td>
<td align="left">CRYL1</td>
<td align="left">ENSG00000165475.15</td>
<td align="left">chr13.20405904.-</td>
<td align="left">chr13.20405997.-</td>
<td align="left">Alt 5’ss</td>
<td align="right">-2.8191629</td>
<td align="right">3.3431713</td>
<td align="right">48.302244</td>
<td align="right">1.641993e+09</td>
<td align="right">1400.00395</td>
<td align="right">4071.6117233</td>
<td align="right">0.1740777</td>
<td align="right">0.6024948</td>
<td align="right">0.9128709</td>
<td align="right">1.172849e+06</td>
<td align="right">403278.4047</td>
<td align="right">0.3438452</td>
<td align="right">8.725985e+05</td>
<td align="right">3241109.952</td>
<td align="right">3.3301579</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="left">Risdiplam-specific</td>
</tr>
<tr class="odd">
<td align="left">chr9:108931201:108933271:clu_18237_-</td>
<td align="left">chr9</td>
<td align="right">108931201</td>
<td align="right">108933271</td>
<td align="left">chr9_clu_18237_-</td>
<td align="left">-</td>
<td align="left">AAGAGTAAGCA</td>
<td align="right">4.905988</td>
<td align="left">ELP1</td>
<td align="left">ENSG00000070061.15</td>
<td align="left">chr9.108933271.-</td>
<td align="left">chr9.108933405.-</td>
<td align="left">Alt 5’ss</td>
<td align="right">-9.6753286</td>
<td align="right">0.5638888</td>
<td align="right">7.339398</td>
<td align="right">2.540583e+07</td>
<td align="right">33.32162</td>
<td align="right">895.8671341</td>
<td align="right">0.1926687</td>
<td align="right">0.9052038</td>
<td align="right">0.7822166</td>
<td align="right">7.624427e+05</td>
<td align="right">28358.9229</td>
<td align="right">0.0371948</td>
<td align="right">5.672567e+05</td>
<td align="right">227917.950</td>
<td align="right">0.3602338</td>
<td align="right">0.0000021</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000023</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="left">Risdiplam-specific</td>
</tr>
<tr class="even">
<td align="left">chr18:10773629:10774006:clu_33760_-</td>
<td align="left">chr18</td>
<td align="right">10773629</td>
<td align="right">10774006</td>
<td align="left">chr18_clu_33760_-</td>
<td align="left">-</td>
<td align="left">ATGAGTAAGTC</td>
<td align="right">4.559120</td>
<td align="left">PIEZO2</td>
<td align="left">ENSG00000154864.12</td>
<td align="left">chr18.10774006.-</td>
<td align="left">chr18.10774036.-</td>
<td align="left">Annotated</td>
<td align="right">-2.2089456</td>
<td align="right">0.0687500</td>
<td align="right">51.453519</td>
<td align="right">1.001587e+07</td>
<td align="right">143.33677</td>
<td align="right">1424.8728605</td>
<td align="right">0.2738613</td>
<td align="right">0.9128709</td>
<td align="right">0.8416254</td>
<td align="right">6.987650e+04</td>
<td align="right">7029.3086</td>
<td align="right">0.1005962</td>
<td align="right">5.198805e+04</td>
<td align="right">56493.880</td>
<td align="right">0.9742791</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.4229854</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.6497405</td>
<td align="left">Risdiplam-specific</td>
</tr>
<tr class="odd">
<td align="left">chr19:10330526:10332870:clu_34319_-</td>
<td align="left">chr19</td>
<td align="right">10330526</td>
<td align="right">10332870</td>
<td align="left">chr19_clu_34319_-</td>
<td align="left">-</td>
<td align="left">GAGAGTACGCC</td>
<td align="right">2.214523</td>
<td align="left">RAVER1</td>
<td align="left">ENSG00000161847.14</td>
<td align="left">chr19.10332870.-</td>
<td align="left">chr19.10332909.-</td>
<td align="left">Alt 5’ss</td>
<td align="right">-0.0000251</td>
<td align="right">0.1685537</td>
<td align="right">3.589164</td>
<td align="right">1.413434e+03</td>
<td align="right">22.81686</td>
<td align="right">0.7759248</td>
<td align="right">0.9831921</td>
<td align="right">-0.2712254</td>
<td align="right">-0.5135291</td>
<td align="right">6.194690e+01</td>
<td align="right">1821.6118</td>
<td align="right">29.4060210</td>
<td align="right">4.608844e+01</td>
<td align="right">14640.120</td>
<td align="right">284.7988165</td>
<td align="right">0.9969948</td>
<td align="right">0.9965823</td>
<td align="right">0.9983501</td>
<td align="right">0.4873920</td>
<td align="right">0.4546814</td>
<td align="right">1.0000000</td>
<td align="left">Non significant</td>
</tr>
<tr class="even">
<td align="left">chr4:41014398:41032584:clu_10225_-</td>
<td align="left">chr4</td>
<td align="right">41014398</td>
<td align="right">41032584</td>
<td align="left">chr4_clu_10225_-</td>
<td align="left">-</td>
<td align="left">AGGAGTAGGTG</td>
<td align="right">5.043083</td>
<td align="left">APBB2</td>
<td align="left">ENSG00000163697.17</td>
<td align="left">chr4.41032584.-</td>
<td align="left">chr4.41032626.-</td>
<td align="left">Alt 5’ss</td>
<td align="right">-3.0030040</td>
<td align="right">0.0355949</td>
<td align="right">1.682290</td>
<td align="right">3.820296e+05</td>
<td align="right">22.66113</td>
<td align="right">452.1565311</td>
<td align="right">0.3195048</td>
<td align="right">0.8580987</td>
<td align="right">0.9082951</td>
<td align="right">1.685837e+04</td>
<td align="right">844.9056</td>
<td align="right">0.0501179</td>
<td align="right">1.254261e+04</td>
<td align="right">6790.426</td>
<td align="right">0.4853942</td>
<td align="right">0.0010952</td>
<td align="right">0.0000000</td>
<td align="right">0.0006567</td>
<td align="right">0.0009786</td>
<td align="right">0.0000000</td>
<td align="right">0.0025582</td>
<td align="left">Risdiplam-specific</td>
</tr>
</tbody>
</table>
<p>After looking at some of the top risdiplam-specific hits from the
list above, here is perhaps the most clear example:</p>
<div class="figure">
<img src="assets/RisdiSpecificExample2.png" alt="" />
<p class="caption">RisdiSpecific2</p>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.9     purrr_0.3.4    
[5] readr_2.1.2     tidyr_1.2.0     tibble_3.1.7    ggplot2_3.3.6  
[9] tidyverse_1.3.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     lubridate_1.8.0  assertthat_0.2.1 rprojroot_2.0.3 
 [5] digest_0.6.29    utf8_1.2.2       R6_2.5.1         cellranger_1.1.0
 [9] backports_1.4.1  reprex_2.0.1     evaluate_0.15    highr_0.9       
[13] httr_1.4.3       pillar_1.7.0     rlang_1.0.2      readxl_1.4.0    
[17] rstudioapi_0.13  whisker_0.4      jquerylib_0.1.4  rmarkdown_2.14  
[21] bit_4.0.4        munsell_0.5.0    broom_0.8.0      compiler_4.2.0  
[25] httpuv_1.6.5     modelr_0.1.8     xfun_0.30        pkgconfig_2.0.3 
[29] htmltools_0.5.2  tidyselect_1.1.2 workflowr_1.7.0  fansi_1.0.3     
[33] crayon_1.5.1     tzdb_0.3.0       dbplyr_2.1.1     withr_2.5.0     
[37] later_1.3.0      grid_4.2.0       jsonlite_1.8.0   gtable_0.3.0    
[41] lifecycle_1.0.1  DBI_1.1.2        git2r_0.30.1     magrittr_2.0.3  
[45] scales_1.2.0     vroom_1.5.7      cli_3.3.0        stringi_1.7.6   
[49] farver_2.1.0     fs_1.5.2         promises_1.2.0.1 xml2_1.3.3      
[53] bslib_0.3.1      ellipsis_0.3.2   generics_0.1.2   vctrs_0.4.1     
[57] tools_4.2.0      bit64_4.0.5      glue_1.6.2       hms_1.1.1       
[61] parallel_4.2.0   fastmap_1.1.0    yaml_2.3.5       colorspace_2.0-3
[65] rvest_1.0.2      knitr_1.39       haven_2.5.0      sass_0.4.1      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
