---
title: "MakeBigwigTsv"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=F, message=F)
```

## Intro

I have made a lot of bigwigs for RNA-seq files for this project. I also have a [helper script](https://github.com/bfairkun/GenometracksByGenotype/blob/master/AggregateBigwigsForPlotting.py) for use with [pygenometracks](https://pygenometracks.readthedocs.io/en/latest/), that requires a tab seperated file list of the bigwigs to plot things easily... Here I will make that list:

## Code

```{r}
library(tidyverse)
library(RColorBrewer)

StrandedBigwigList <- Sys.glob(paste0(usethis::proj_path(), "/code/bigwigs/stranded/*.bw"))

UnstrandedBigwigList <- Sys.glob(paste0(usethis::proj_path(), "/code/bigwigs/unstranded/*.bw"))

```


Now read in sample metadata:

```{r}

chRNA.seq.experiment <- read_tsv("../data/chRNA_sampleLibraryKey.tsv") %>%
  mutate(OldSampleID = paste0("chRNA_", `#`)) %>%
  mutate(CellType = "LCL") %>%
  mutate(Treatment = str_replace(X3, "^(.+?) .+", "\\1")) %>%
  mutate(Treatment = str_replace_all(Treatment, "-", "")) %>%
  mutate(QualitativeDose = str_replace(X3, "^.+? (.+?) .+", "\\1")) %>%
  mutate(Replicate = str_replace(X3, "^.+ (.+)$", "\\1")) %>%
  mutate(dose.nM = case_when(
    Treatment == "Risdiplam" & QualitativeDose == "high" ~ 3160,
    Treatment == "Risdiplam" & QualitativeDose == "low" ~ 100,
    Treatment == "Branaplam" & QualitativeDose == "high" ~ 1000,
    Treatment == "Branaplam" & QualitativeDose == "low" ~ 31.6,
    Treatment == "C2C5" & QualitativeDose == "high" ~ 316,
    Treatment == "C2C5" & QualitativeDose == "low" ~ 10,
    TRUE ~ NA_real_
  )) %>%
  mutate(LibType = "chRNA") %>%
  mutate(SampleID = paste(Treatment, dose.nM, CellType, LibType, Replicate, sep = "_")) %>%
  mutate(Group_label = paste(Treatment, dose.nM, CellType, LibType, sep = "_")) %>%
  dplyr::select(OldSampleID, Group_label, SampleID)



TitrationExp <- data.frame(C2C5=c(1000, 316, 100, 31.6, 10, 3.16, 1, 0.316),
           Branaplam=c(3160, 1000, 316, 100, 31.6, 10, 3.16, 1),
           Risdiplam=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16)) %>%
  mutate(TitrationPoint = 1:8) %>%
  gather(key="Treatment", value="dose.nM", -TitrationPoint) %>%
  add_row(TitrationPoint=1, Treatment="DMSO", dose.nM = NA_real_) %>%
  add_row(TitrationPoint=2, Treatment="DMSO", dose.nM = NA_real_) %>%
  add_row(TitrationPoint=3, Treatment="DMSO", dose.nM = NA_real_) %>%
  mutate(LibType = "polyA", CellType="LCL") %>%
  mutate(TreatmentShort = recode(Treatment, !!!c("Risdiplam"="Ris", "Branaplam"="Bran"))) %>%
  mutate(OldSampleID = paste0("TitrationExp", TreatmentShort, "-", TitrationPoint)) %>%
  mutate(Group_label = paste(Treatment, dose.nM, CellType, LibType, sep = "_")) %>%
  mutate(Replicate = case_when(
    Treatment == "DMSO" ~ TitrationPoint,
    TRUE ~ 1
  )) %>%
  mutate(SampleID =  paste(Treatment, dose.nM, CellType, LibType, Replicate,sep = "_")) %>%
  dplyr::select(OldSampleID, Group_label, SampleID)

FibroblastExp <- read_tsv("../code/config/samples.tsv") %>%
  separate(sample, into=c("Treatment", "Replicate"), sep="-", remove=F) %>%
  mutate(LibType = "polyA", CellType="Fibroblast", dose.nM = NA_real_) %>%
  mutate(SampleID =  paste(Treatment, dose.nM, CellType, LibType, Replicate,sep = "_")) %>%
  mutate(Group_label = paste(Treatment, dose.nM, CellType, LibType, sep = "_")) %>%
  dplyr::select(OldSampleID=sample, Group_label, SampleID)

AllSamples <- bind_rows(chRNA.seq.experiment, TitrationExp, FibroblastExp)
```

Before I make the final bigwig tsv files, I want to make leafcutter junction excision ratio summary files for plotting sashimi:

```{r}

leafcutter_count_numers <-"../code/SplicingAnalysis/leafcutter_all_samples/leafcutter_perind_numers.counts.gz"

Count.Table.mat <- read.table(leafcutter_count_numers, sep = ' ', nrows = Inf) %>%
  as.matrix()

ClusterMax.mat <- Count.Table.mat %>%
    as.data.frame() %>%
    rownames_to_column("junc") %>%
    mutate(cluster=str_replace(junc, "^(.+?):.+?:.+?:(.+)$", "\\1_\\2")) %>%
    group_by(cluster) %>%
    mutate(across(where(is.numeric), sum)) %>%
    ungroup() %>%
    dplyr::select(junc, everything(), -cluster) %>%
    column_to_rownames("junc") %>%
    as.matrix()


PSI.df <- (Count.Table.mat / as.numeric(ClusterMax.mat) * 100) %>%
    signif(digits=2) %>%
    as.data.frame()


bigwigs.df <- data.frame(BigwigFilepath = c(StrandedBigwigList, UnstrandedBigwigList)) %>%
  mutate(OldSampleID = str_replace(BigwigFilepath, ".+/bigwigs/.+?/(.+?)\\..+$", "\\1")) %>%
  left_join(AllSamples, by="OldSampleID") %>%
  mutate(StrandDummy = str_replace(BigwigFilepath, ".+\\.(.+?)\\.bw", "\\1")) %>%
  mutate(Strand = case_when(
    StrandDummy == "minus" ~ "-",
    StrandDummy == "plus" ~ "+",
    TRUE ~ "."
  )) %>%
  mutate(Group_label = case_when(
    Strand == "." ~ paste(Group_label, "u", sep="_"),
    TRUE ~ Group_label
  ))


ColNameKey <- bigwigs.df %>%
      dplyr::select(SampleID, OldSampleID) %>%
      mutate(OldSampleID = str_replace_all(OldSampleID, "-", ".")) %>%
  distinct(SampleID, .keep_all=T) %>%
  deframe()

length(ColNameKey)

#Write out PSI table
PSI.df %>%
  rownames_to_column("junc") %>%
  dplyr::rename(!!!ColNameKey) %>%
  separate(junc, into=c("#Chrom", "start", "end", "cluster"), convert=T, remove=F, sep=":") %>%
  mutate(gid = paste(`#Chrom`, cluster, sep="_" )) %>%
  mutate(strand = str_extract(cluster, "[+-]")) %>%
  dplyr::select(`#Chrom`, start, end, junc, gid, strand, everything(), -cluster) %>%
  arrange(`#Chrom`, start, end) %>%
  write_tsv("../code/SplicingAnalysis/leafcutter_all_samples/PSI.table.bed")

#Also write out junction count table
Count.Table.mat %>%
  as.data.frame() %>%
  rownames_to_column("junc") %>%
  dplyr::rename(!!!ColNameKey) %>%
  separate(junc, into=c("#Chrom", "start", "end", "cluster"), convert=T, remove=F, sep=":") %>%
  mutate(gid = paste(`#Chrom`, cluster, sep="_" )) %>%
  mutate(strand = str_extract(cluster, "[+-]")) %>%
  dplyr::select(`#Chrom`, start, end, junc, gid, strand, everything(), -cluster) %>%
  arrange(`#Chrom`, start, end) %>%
  write_tsv("../code/SplicingAnalysis/leafcutter_all_samples/JuncCounts.table.bed")




```


Then remember to bgzip and tabix index the bed file for memory efficient reading

```{bash, eval=F}
bgzip ../code/SplicingAnalysis/leafcutter_all_samples/PSI.table.bed
tabix -p bed ../code/SplicingAnalysis/leafcutter_all_samples/PSI.table.bed.gz
```

Now let's write out the bigwig list and groups files.

```{r}
bigwigs.df %>%
  dplyr::select(SampleID, BigwigFilepath, Group_label, Strand ) %>%
  write_tsv("../code/bigwigs/BigwigList.tsv", col_names = F)

bigwigs.df %>%
  distinct(Group_label) %>%
  separate(Group_label, into=c("Treatment", "dose.nM", "CellType", "LibType", "IsUnstranded"), sep="_", convert=T, remove=F ) %>%
  group_by(Treatment) %>%
  mutate(doseRank = dense_rank(dose.nM)) %>%
  rowwise() %>%
  mutate(Color = case_when(
    Treatment == "Branaplam" ~ brewer.pal(n=8, name="Greens")[doseRank],
    Treatment == "C2C5" ~ brewer.pal(n=8, name="Purples")[doseRank],
    Treatment == "Risdiplam" ~ brewer.pal(n=8, name="Blues")[doseRank],
    Treatment == "DMSO" ~ "#969696",
    Treatment == "A10" ~ "#7fc97f",
    Treatment == "C2" ~ "#beaed4",
    Treatment == "CUOME" ~ "#fdc086",
    Treatment == "SM2" ~ "#ffff99"
  )) %>%
  arrange(CellType, LibType, Treatment, dose.nM) %>%
  ungroup() %>%
  mutate(BedgzFile = (paste0(usethis::proj_path(), "/code/SplicingAnalysis/leafcutter_all_samples/PSI.table.bed.gz"))) %>%
  mutate(IsDMSO = Treatment == "DMSO") %>%
  arrange(LibType, CellType, desc(IsDMSO), Treatment, dose.nM, Group_label) %>%
  dplyr::select(Group_label, Color, BedgzFile) %>%
  mutate(Supergroup = case_when(
    str_detect(Group_label, "_LCL_") ~ str_replace(Group_label, "^(.+?)_.+?_(.+?_.+)$", "\\1_\\2"),
    TRUE ~ Group_label
  )) %>%
  write_tsv("../code/bigwigs/BigwigList.groups.tsv", col_names =F)
```

