---
title: "20220922_Explore_SubparRNAseq"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message=F)
```

## Intro

Jingxin's lab has made many compounds (ie >50) from one or two scaffolds, and is interseted in using RNA-seq to readout splicing effects. Obviously at this scale, with limited resources, and with low prior knowledge about their splice-modulating activity, it doesn't make sense to do a full titration experiment with RNA-seq for each compound. We are more likely to settle with a single dose, at a smaller read depth, and try to learn what we can about the activity and specificity of these compounds. Here I will explore our existing data just to gain some intuitions on the power of RNA-seq, keeping in mind what it would be like if we only had one replicate at lower read coverage. Eventually I may do a more careful power analysis, involving sub-sampling reads from our existing data, and re-analyzing them to simulate potential study designs.


## Analysis

First let's read in library size in terms of mapped reads.

```{r}
library(tidyverse)
library(gplots)
library(RColorBrewer)

counts <- read_tsv("../output/QC/ReadCountsAndJunctionsPerSamples.tsv", col_names = c("fn", "ChromosomalReads"))

sample.list <- read_tsv("../code/bigwigs/BigwigList.tsv",
                        col_names = c("SampleName", "bigwig", "group", "strand")) %>%
  filter(strand==".") %>%
  dplyr::select(-strand) %>%
  mutate(old.sample.name = str_replace(bigwig, "/project2/yangili1/bjf79/20211209_JingxinRNAseq/code/bigwigs/unstranded/(.+?).bw", "\\1")) %>%
  separate(SampleName, into=c("treatment", "dose.nM", "cell.type", "libType", "rep"), convert=T, remove=F, sep="_") %>%
  left_join(
    read_tsv("../code/bigwigs/BigwigList.groups.tsv", col_names = c("group", "color", "bed", "supergroup")),
    by="group"
  )

TreatmentColorsForLabels <- sample.list %>%
  group_by(treatment) %>%
  filter(dose.nM == max(dose.nM) | treatment == "DMSO") %>%
  ungroup() %>%
  distinct(treatment, .keep_all=T) %>%
  arrange(dose.nM) %>%
  mutate(vjust=row_number()*1.2)

TreatmentColorsForLabelsKey <- TreatmentColorsForLabels %>%
  filter(!treatment=="DMSO") %>%
  dplyr::select(treatment, color) %>% deframe()

TreatmentColorsLabels.Layer <- geom_text(
    data = TreatmentColorsForLabels, 
    aes(label=treatment, color=color, vjust=vjust),
    y=Inf, x=Inf, hjust=1.05
  )

counts.plot.dat <- counts %>%
  mutate(old.sample.name = str_replace(fn, "Alignments/STAR_Align/(.+?)/Aligned.sortedByCoord.out.bam", "\\1")) %>%
  inner_join(sample.list) %>%
  arrange(cell.type, libType, desc(treatment=="DMSO"), treatment, dose.nM, rep) %>%
  mutate(SampleName = factor(SampleName, levels=SampleName)) %>%
  mutate(Experiment = case_when(
    libType == "chRNA" ~ "nascent RNA profiling",
    cell.type == "LCL" & libType == "polyA" ~ "dose response experiment",
    cell.type == "Fibroblast" ~ "Initial experiment in fibroblast"
  )) %>%
  mutate(label = paste0(treatment,"; ", dose.nM))

counts.plot.labels <- counts.plot.dat %>%
  dplyr::select(SampleName, label) %>% deframe()

ReadsPerDataset <- ggplot(counts.plot.dat, aes(x=SampleName, y=ChromosomalReads/2E6, fill=color)) +
  geom_col() +
  scale_fill_identity() +
  scale_x_discrete(name="Sample; treatment_nanomolar-dose", label=counts.plot.labels) +
  facet_grid(cols = vars(Experiment), scales="free", space = "free_x") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size=5)) +
  theme(strip.text.x = element_text(size = 8)) +
  labs(title="RNA-seq datasets", y="Read count (M)")
ReadsPerDataset
```

```{r, eval=F}
ggsave("../code/figs/Original/ReadsPerSample.pdf", ReadsPerDataset, height=3, width=7)
```


Now let's read in the table of a few hundred GA-GT splice sites that I previously shared with Jingxin, along with some other splice count tables... I want to see how rare these GA-GT splice sites are in our data, to get a sense how many reads we need to detect them.

```{r}

GA.GT.Introns <- read_tsv("../output/EC50Estimtes.FromPSI.txt.gz")


all.samples.PSI <- read_tsv("../code/SplicingAnalysis/leafcutter_all_samples/PSI.table.bed.gz")
all.samples.junccounts <- read_tsv("../code/SplicingAnalysis/leafcutter_all_samples/JuncCounts.table.bed.gz")

all.samples.5ss <- read_tsv("../code/SplicingAnalysis/FullSpliceSiteAnnotations/JuncfilesMerged.annotated.basic.bed.5ss.tab.gz", col_names = c("intron", "seq", "score")) %>%
  mutate(intron = str_replace(intron, "^(.+?)::.+$", "\\1")) %>%
  separate(intron, into=c("chrom", "start", "stop", "strand"), sep="_", convert=T, remove=F)

all.samples.intron.annotations <- read_tsv("../code/SplicingAnalysis/FullSpliceSiteAnnotations/JuncfilesMerged.annotated.basic.bed.gz")

All.GA.GT <- all.samples.5ss %>%
  filter(str_detect(seq, "^\\w{2}GAGT")) %>%
  inner_join(all.samples.intron.annotations %>%
               rename(stop=end) %>%
               dplyr::select(-score)) %>%
  left_join(
    GA.GT.Introns %>%
      dplyr::select(chrom=`#Chrom`, start, stop=end, strand=strand.y, LowerLimit, UpperLimit, UpstreamSpliceAcceptor, spearman.coef.Branaplam, spearman.coef.C2C5,spearman.coef.Risdiplam)
  ) %>%
  mutate(GAGT.set = case_when(
    !is.na(LowerLimit) ~ "Juncs modelled in LCL",
    !is.na(spearman.coef.Branaplam) ~ "Juncs counted in LCL",
    TRUE ~ "Other juncs observed across datasets"
  )) %>%
  inner_join(
    all.samples.junccounts %>%
      dplyr::select(junc, contains("LCL")) %>%
      dplyr::select(junc, contains("polyA")) %>%
      mutate(intron = str_replace(junc, "^(chr.+?):(.+?):(.+?):clu.+?([+-])$", "\\1_\\2_\\3_\\4"))
  )

GA.GT.CPM.Plot.dat <- All.GA.GT %>%
  dplyr::select(intron, GAGT.set, contains("LCL")) %>%
  add_count(GAGT.set) %>%
  gather("SampleName", "JuncCounts", contains("LCL")) %>%
  group_by(SampleName, n, GAGT.set) %>%
  summarise(sumJuncCounts = sum(JuncCounts)) %>%
  inner_join(counts.plot.dat) %>%
  mutate(CPM=sumJuncCounts/(ChromosomalReads/1E6)) %>%
  mutate(FacetLabel = paste0(GAGT.set, ";n=",n)) %>%
  arrange(cell.type, libType, desc(treatment=="DMSO"), treatment, dose.nM, rep, GAGT.set)

GA.GT.CPM.Plot.dat$SampleName <- factor(GA.GT.CPM.Plot.dat$SampleName, levels=unique(GA.GT.CPM.Plot.dat$SampleName))

ggplot(GA.GT.CPM.Plot.dat, aes(x=SampleName, y=CPM, fill=color, color=color)) +
  geom_col() +
  scale_fill_identity() +
  scale_color_identity() +
  scale_x_discrete(name="Sample; treatment_nanomolar-dose", label=counts.plot.labels) +
  facet_wrap(~FacetLabel, scales="free_y") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size=5)) +
  theme(strip.text.x = element_text(size = 6)) +
  labs(title="DetectionOf GA|GT introns in aggregate", y="CountsPerMillion")

```

Ok, aggregating the set of 600 or so junctions that I modelled I see clear dose-dependent differences between DMSO and treatments, with CountsPerMillions (expression of junction reads) in the hundreds... Even with a single sample at a fraction of the read depth, i think this would be easily detectable.


Perhaps another way to consider what it would be like to analyze data with just one replicate, is to look at delta-PSI, based on either 1 sample, or the average of 3 samples, from the fibroblast data...


```{r}
cluster_sig_files <- list.files("../code/SplicingAnalysis/leafcutter/differential_splicing", "*_cluster_significance.txt", full.names = T)
effect_sizes_files <- list.files("../code/SplicingAnalysis/leafcutter/differential_splicing", "*_effect_sizes.txt", full.names = T)
treatments <- str_replace(cluster_sig_files, ".+/(.+?)_cluster_significance.txt", "\\1")

cluster.sig <- map(cluster_sig_files, read_tsv) %>%
  set_names(cluster_sig_files) %>%
  bind_rows(.id="f") %>%
  mutate(treatment = str_replace(f, ".+/(.+?)_cluster_significance.txt", "\\1")) %>%
  select(-f)

effect_sizes <- map(effect_sizes_files, read_tsv) %>%
  set_names(effect_sizes_files) %>%
  bind_rows(.id="f") %>%
  mutate(treatment = str_replace(f, ".+/(.+?)_effect_sizes.txt", "\\1")) %>%
  select(-f) %>%
  unite("psi_treatment", treatments, sep=" ", na.rm=T) %>%
  mutate(psi_treatment = as.numeric(psi_treatment),
         cluster = str_replace(intron, "(.+?:).+:(.+?)", "\\1\\2"))


#Based on all introns
all.samples.PSI %>%
  dplyr::select(1:6, contains("Fibroblast")) %>%
  dplyr::select(-c(1:6)) %>%
  drop_na() %>%
  as.matrix() %>%
  cor() %>%
  heatmap.2(trace='none')

#Based on all GA|GT introns
all.samples.PSI %>%
  dplyr::select(1:6, contains("Fibroblast")) %>%
  inner_join(dplyr::select(All.GA.GT, `#Chrom`=chrom, start, end=stop, strand)) %>%
  dplyr::select(-c(1:6)) %>%
  drop_na() %>%
  as.matrix() %>%
  cor() %>%
  heatmap.2(trace='none')

```

Lastly, I think way to plot it that Yang is interested in is to average the 3 replicates for treatment, calculate delta-psi (from DMSO averaged across three replicates), and compare to the delta-psi from just a single replicate. Let's include one GA-GT introns for this...

```{r, fig.height=7, fig.width=7}
DMSO.mean.PSI <- all.samples.PSI %>%
  dplyr::select(1:6, contains("Fibroblast")) %>%
  inner_join(dplyr::select(All.GA.GT, `#Chrom`=chrom, start, end=stop, strand)) %>%
  drop_na() %>%
  dplyr::select(junc, contains("Fibroblast")) %>%
  gather("SampleName", "PSI", contains("Fibroblast")) %>%
  inner_join(sample.list) %>%
  filter(treatment == "DMSO") %>%
  group_by(junc) %>%
  summarise(meanPSI.DMSO = mean(PSI)) %>%
  ungroup()

treament.mean.PSI <- all.samples.PSI %>%
  dplyr::select(1:6, contains("Fibroblast")) %>%
  inner_join(dplyr::select(All.GA.GT, `#Chrom`=chrom, start, end=stop, strand)) %>%
  drop_na() %>%
  dplyr::select(junc, contains("Fibroblast")) %>%
  gather("SampleName", "PSI", contains("Fibroblast")) %>%
  inner_join(sample.list) %>%
  filter(!treatment == "DMSO") %>%
  group_by(treatment, junc) %>%
  summarise(PSI = mean(PSI)) %>%
  ungroup() %>%
  mutate(SampleName = paste(treatment, "mean", sep="_")) %>%
  mutate(PSI.Calculation = "Average across 3 treatment replicates")
  
Compare3vs1Replicate.Dat.To.Plot <- all.samples.PSI %>%
  dplyr::select(1:6, contains("Fibroblast")) %>%
  inner_join(dplyr::select(All.GA.GT, `#Chrom`=chrom, start, end=stop, strand)) %>%
  drop_na() %>%
  dplyr::select(junc, contains("Fibroblast")) %>%
  gather("SampleName", "PSI", contains("Fibroblast")) %>%
  inner_join(sample.list) %>%
  filter(!treatment == "DMSO") %>%
  dplyr::select(treatment, SampleName, junc, PSI) %>%
  mutate(PSI.Calculation = "Single treatment replicate") %>%
  bind_rows(treament.mean.PSI) %>%
  inner_join(DMSO.mean.PSI) %>%
  mutate(DeltaPSI = PSI-meanPSI.DMSO)


ggplot(Compare3vs1Replicate.Dat.To.Plot, aes(x=DeltaPSI, color=treatment)) +
  stat_ecdf(aes(linetype=PSI.Calculation)) +
  facet_wrap(~treatment) +
  theme_bw() +
  coord_cartesian(xlim=c(-10, 10)) +
  labs(y="ecdf", title='effects of GA|GT introns')
```

Ok, so the mean genomewide distribution (plotted as ecdf) of the delta-PSI of GA.GT introns looks visually indistinguishable whether you average 3 replicates or if you just use one replicate.

Let's now plot these DeltaPSIs again (comparing deltaPSI calculated from one replicate, versus average from three replicates) as a correlation matrix, with clustering the rows and columns...

```{r}

Compare3vs1Replicate.Dat.To.Plot.As.matrix <- Compare3vs1Replicate.Dat.To.Plot %>%
  dplyr::select(SampleName, junc, DeltaPSI) %>%
  pivot_wider(names_from = "SampleName", values_from="DeltaPSI") %>%
  column_to_rownames("junc")

ConvertToColorVector <- function(Vector, PalletteString="Set1"){
  ConversionKey <- setNames(brewer.pal(length(unique(Vector)), PalletteString), unique(Vector))[1:length(unique(Vector))]
  ColorVector <- recode(Vector, !!!ConversionKey)
  return( list(Key=ConversionKey, ColorVector=ColorVector))
}

Rowcols <- colnames(Compare3vs1Replicate.Dat.To.Plot.As.matrix) %>%
  str_replace("^(.+?)_.+", "\\1") %>% ConvertToColorVector()
Colcols <- colnames(Compare3vs1Replicate.Dat.To.Plot.As.matrix) %>%
  str_detect("mean") %>% as.character() %>% ConvertToColorVector(PalletteString="Set2")

# Row colors are treatments
# Column colors are 1 or 3 replicates
cor(Compare3vs1Replicate.Dat.To.Plot.As.matrix) %>%
  heatmap.2(trace='none', ColSideColors = Colcols$ColorVector, RowSideColors = Rowcols$ColorVector)


```


Let's plot deltaPSI from 1 vs 3 averaged replicates for some individual introns. A random sample of GA|GT introns. I always find looking at a random sample of introns to be helpful, to gain some intuition.


```{r}
sample_n_of <- function(data, size, ...) {
  dots <- quos(...)
  
  group_ids <- data %>% 
    group_by(!!! dots) %>% 
    group_indices()
  
  sampled_groups <- sample(unique(group_ids), size)
  
  data %>% 
    filter(group_ids %in% sampled_groups)
}

set.seed(0)
Compare3vs1Replicate.Dat.To.Plot %>%
  sample_n_of(20, junc) %>%
  ggplot(aes(y=DeltaPSI, x=SampleName, color=PSI.Calculation, fill=treatment)) +
  geom_col() +
  facet_wrap(~junc,scales="free_y") +
  theme_bw() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
  labs(title="20 random GA|GT introns", y="DeltaPSI (versus DMSO from 3 replicates)", color="Treatment PSI from")

```

Hmm.. so as expected, individual intron measurements can be quite noisy, and I still haven't figured out how or if I will want to do any differential splicing tests at the individual intron level.

Let's go back to thinking about characterizing samples on a global level... I wonder if it will be useful to plot new samples in a previously determined priniciple component space that is more interpretable. For example, let's go back to the dose response experiment, calculate principal components, and overlay the fibroblast samples in the same principle component axes. Keep in mind the experiments were done in different cell types, so I'm not sure what to expect here, but let's just see...

```{r}
#First compute principle components from dose response data
pca.results <- all.samples.PSI %>%
  inner_join(
    dplyr::select(All.GA.GT, `#Chrom`=chrom, start, end=stop, strand)
    ) %>%
  drop_na() %>%
  dplyr::select(1:6, contains("LCL")) %>%
  dplyr::select(1:6, contains("polyA")) %>%
  dplyr::select(junc, contains("LCL")) %>%
  column_to_rownames("junc") %>%
  scale() %>% t() %>% prcomp(scale=T)

summary(pca.results)

# Then plot original samples
pca.results$x %>%
  as.data.frame() %>%
  rownames_to_column("SampleName") %>%
  dplyr::select(SampleName, PC1, PC2, PC3) %>%
  left_join(sample.list, by="SampleName") %>%
  ggplot(aes(x=PC1, y=PC2, color=color)) +
  TreatmentColorsLabels.Layer +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = "PCA using all GA|GT introns", x="PC1 (52% variance explained)", y="PC2 (15% variance explained)")

pca.results$x %>%
  as.data.frame() %>%
  rownames_to_column("SampleName") %>%
  dplyr::select(SampleName, PC1:PC6) %>%
  left_join(sample.list, by="SampleName") %>%
  ggplot(aes(x=PC3, y=PC4, color=color)) +
  TreatmentColorsLabels.Layer +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = "PCA using all GA|GT introns")

pca.results$x %>%
  as.data.frame() %>%
  rownames_to_column("SampleName") %>%
  dplyr::select(SampleName, PC1:PC6) %>%
  left_join(sample.list, by="SampleName") %>%
  ggplot(aes(x=PC5, y=PC6, color=color)) +
  TreatmentColorsLabels.Layer +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = "PCA using all GA|GT introns")


```


Hmm, it looks like PC5 does a decent job at seperating C2C5 from risidiplam. Let's plot just PC2 and PC5...

```{r}
pca.results$x %>%
  as.data.frame() %>%
  rownames_to_column("SampleName") %>%
  dplyr::select(SampleName, PC1:PC6) %>%
  left_join(sample.list, by="SampleName") %>%
  ggplot(aes(x=PC2, y=PC5, color=color)) +
  TreatmentColorsLabels.Layer +
  geom_point(size=3) +
  scale_color_identity() +
  theme_bw() +
  labs(title = "PCA using all GA|GT introns")
```



Now let's try plotting the fibroblast data in the old PC space. From visually looking at a few loci in the genome browser, I already know that SM2 is a lot like branaplam, so I'll look for that effect...


```{r}

FirboblastDat <- all.samples.PSI %>%
  inner_join(
    dplyr::select(All.GA.GT, `#Chrom`=chrom, start, end=stop, strand)
    ) %>%
  drop_na() %>%
  dplyr::select(junc, contains("Fibroblast")) %>%
  column_to_rownames("junc") %>%
  scale() %>% t()

Fibroblastdat.projected <- predict(pca.results, FirboblastDat)

TreatmentColorsForLabels.FibroblastColorsAdded <-
sample.list %>%
  group_by(treatment) %>%
  filter(dose.nM == max(dose.nM) | treatment == "DMSO" | cell.type=="Fibroblast") %>%
  ungroup() %>%
  distinct(treatment, .keep_all=T) %>%
  arrange(dose.nM) %>%
  mutate(vjust=row_number()*1.2)

TreatmentColorsLabels.Layer.FibroblastColorsAdded <- geom_text(
    data = TreatmentColorsForLabels.FibroblastColorsAdded, 
    aes(label=treatment, color=color, vjust=vjust),
    y=Inf, x=Inf, hjust=1.05
  )

bind_rows(
    Fibroblastdat.projected %>% as.data.frame(),
    pca.results$x %>% as.data.frame()
  ) %>%
  rownames_to_column("SampleName") %>%
  dplyr::select(SampleName, PC1:PC6) %>%
  left_join(sample.list, by="SampleName") %>%
  ggplot(aes(x=PC1, y=PC2, color=color, shape=cell.type)) +
  geom_point(size=3) +
  scale_color_identity() +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  theme_bw() +
  labs(title = "PCA using all GA|GT introns", x="PC1 (52% variance explained)", y="PC2 (15% variance explained)")


bind_rows(
    Fibroblastdat.projected %>% as.data.frame(),
    pca.results$x %>% as.data.frame()
  ) %>%
  rownames_to_column("SampleName") %>%
  dplyr::select(SampleName, PC1:PC6) %>%
  left_join(sample.list, by="SampleName") %>%
  ggplot(aes(x=PC2, y=PC5, color=color, shape=cell.type)) +
  geom_point(size=3) +
  scale_color_identity() +
  TreatmentColorsLabels.Layer.FibroblastColorsAdded +
  theme_bw() +
  labs(title = "PCA using all GA|GT introns")



```

Ok, I rather like looking at the data this way, and each point is a single sample, and each point is placed nicely in a place I expect and can somewhat interpret in a few of these principle components.

## Conclusion

though I haven't worked out the details of how I would want to perform an analysis with one RNA-seq replicate, I'm confident there is enough genome-wide information to at the very least discriminate "active" from inactive compounds from 1 replicate...

We might not need to do a formal statistical test (ie leafcutter) for each splice event to get a decent profile of each treatment. I could probably consider doing simpler things like looking at meta-intron features (the sum of junction counts many GA|GT introns), or if I wanted to get a broad sense of specificity, plotting things as I have done in the same PC space as previously calculated will at least distinguish the branaplam-vs-risdiplam dimension. Though, a better understanding of specificity (particularly for new dimensions that might not easily be described in the risdiplam-branaplam axis) may require more unbiased analysis which will no doubt benefit from first doing splice-event-level statistical tests.
