---
title: "20220919_chRNA_IntronRetention"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message=F)
```


## Intro

I have quantified intron retention with splice-q package for each sample. I want to investigate the hypothesis that some of the introns affected by small molecules have unique signatures as measured by intron retention or splicing in chRNA dataset. In particular, from looking at the HTT locus below, I notice a few things:

1. The drug-induced exon is in a host-intron with a bit more intron retention compared to the surrounding introns in untreated (DMSO) samples. Perhaps introns with slower splicing are a bit predisposed to new drug-induced splicing.
1. The drug induced exon is spliced-in at a higher rate (relative to the skipping event) in the chRNA, presumably since NMD quickly degrades the exon-included isoform post-transcriptionally.
1. The intron upstream of the drug-induced exon seems to be spliced more efficiently in treated samples, relative to the downstream intron which actually contains the target 5' GA|GT splice site (black line in screenshot) that interacts with the small molecule. This is reminiscent of early exon definition experiments wherein strengthening a downstream 5'ss enhances splicing of the upstream intron.

![IGVScreenshot](assets/ScreenShot_HTT.png)

![HTT](assets/ForJingxin_HTT.png)

In this notebook, I will explore whether these observations are representative of drug-induced exons genomewide.
First, read in the data...

```{r}
library(tidyverse)
library(gplots)

#Read in sample metadata
sample.list <- read_tsv("../code/bigwigs/BigwigList.tsv",
                        col_names = c("SampleName", "bigwig", "group", "strand")) %>%
  filter(strand==".") %>%
  mutate(old.sample.name = str_replace(bigwig, "/project2/yangili1/bjf79/20211209_JingxinRNAseq/code/bigwigs/unstranded/(.+?).bw", "\\1")) %>%
  separate(SampleName, into=c("treatment", "dose.nM", "cell.type", "libType", "rep"), convert=T, remove=F, sep="_") %>%
  left_join(
    read_tsv("../code/bigwigs/BigwigList.groups.tsv", col_names = c("group", "color", "bed", "supergroup")),
    by="group"
  )

SampleNameConversionKey <- sample.list %>%
  dplyr::select(SampleName, old.sample.name) %>%
  deframe()

GA.GT.Introns <- read_tsv("../output/EC50Estimtes.FromPSI.txt.gz")

read.SpliceQ.Table <- function(fn){
  read_tsv(fn) %>%
  rename_at(vars(-Intron), ~str_replace(.x, ".+?Merged/(.+?)\\.txt\\.gz", "\\1")) %>%
  rename(!!!SampleNameConversionKey) %>%
  separate(Intron, into=c("Chrom", "start", "stop", "strand"), convert=T, sep="_") %>%
  mutate(stop = stop+1) %>%
  mutate(Chrom=paste0("chr", Chrom)) %>%
  dplyr::select(Chrom:strand, everything())
}

SpliceQ.IER <- read.SpliceQ.Table("../code/SplicingAnalysis/SpliceQ/MergedTable.IER.txt.gz")
SpliceQ.SE <- read.SpliceQ.Table("../code/SplicingAnalysis/SpliceQ/MergedTable.SE.txt.gz")
  

all.samples.PSI <- read_tsv("../code/SplicingAnalysis/leafcutter_all_samples/PSI.table.bed.gz")

all.samples.5ss <- read_tsv("../code/SplicingAnalysis/FullSpliceSiteAnnotations/JuncfilesMerged.annotated.basic.bed.5ss.tab.gz", col_names = c("intron", "seq", "score")) %>%
  mutate(intron = str_replace(intron, "^(.+?)::.+$", "\\1")) %>%
  separate(intron, into=c("chrom", "start", "stop", "strand"), sep="_", convert=T, remove=F)

all.samples.intron.annotations <- read_tsv("../code/SplicingAnalysis/FullSpliceSiteAnnotations/JuncfilesMerged.annotated.basic.bed.gz")

```


### effect sizes in chRNA versus polyA

Let's check the first hypothesis... that chRNA will generally show larger effect sizes between ratio of included versus skipped reads.

```{r}
GA.GT.Introns.DrugInducedCassetteExons <- GA.GT.Introns %>%
  filter(!is.na(Steepness)) %>%
  filter(UpperLimit > LowerLimit) %>%
  filter(!is.na(UpstreamSpliceAcceptor)) %>%
  dplyr::rename(strand=strand.y)


chRNA.Comparison.Samples <- sample.list %>%
  filter(cell.type=="LCL") %>%
  group_by(dose.nM, treatment) %>%
  filter(any(libType=="chRNA"))


IntronsToQuantifyInDrugInducedCassetteExons <- all.samples.PSI %>%
  filter(gid %in% GA.GT.Introns$gid) %>%
  mutate(SpliceAcceptor = case_when(
    strand == "+" ~ paste(`#Chrom`, end, strand, sep="."),
    strand == "-" ~ paste(`#Chrom`, start-2, strand, sep=".")
  )) %>%
  select(1:6, SpliceAcceptor, contains("LCL")) %>%
  gather(key="Sample", value="PSI", contains("LCL")) %>%
  group_by(junc, gid, strand, SpliceAcceptor) %>%
  drop_na() %>%
  summarise(mean=mean(PSI, na.rm = F)) %>%
  group_by(SpliceAcceptor) %>%
  filter(mean == max(mean)) %>%
  ungroup() %>%
  dplyr::select(UpstreamJunc = junc, SpliceAcceptor) %>%
  inner_join(GA.GT.Introns.DrugInducedCassetteExons, by=c("SpliceAcceptor"="UpstreamSpliceAcceptor")) %>%
  rename(DownstreamJunc = junc) %>%
  mutate(SkippingJunc = case_when(
    strand == "+" ~ str_replace(paste(UpstreamJunc, DownstreamJunc), "^(chr.+?):(.+?):(.+?):(clu.+?) (chr.+?):(.+?):(.+?):clu.+$", "\\1:\\2:\\7:\\4"),
    strand == "-" ~ str_replace(paste(UpstreamJunc, DownstreamJunc), "^(chr.+?):(.+?):(.+?):(clu.+?) (chr.+?):(.+?):(.+?):clu.+$", "\\1:\\6:\\3:\\4")
  )) %>%
  dplyr::select(UpstreamJunc,DownstreamJunc, SkippingJunc ) %>%
  gather(key="IntronType", value="junc") %>%
  distinct(.keep_all = T) %>%
  mutate(gid = str_replace(junc, ".+?:(clu.+$)", "\\1")) %>%
  add_count(gid) %>%
  filter(n==3)

chRNA.Comparison.PSI.Inclusion.tidy <- IntronsToQuantifyInDrugInducedCassetteExons %>%
  dplyr::select(-n, -gid) %>%
  inner_join(
    all.samples.PSI %>%
      select(1:6, chRNA.Comparison.Samples$SampleName),
    by="junc"
  ) %>%
  gather("Sample", "PSI", contains("LCL")) %>%
  pivot_wider(names_from = "IntronType", values_from="PSI", id_cols=c("gid", "Sample")) %>%
  unnest() %>%
  mutate(InclusionPSI=UpstreamJunc+DownstreamJunc) %>%
  separate(Sample, into=c("treatment", "dose.nM", "cell.type", "libType", "rep"), sep="_", convert=T) %>%
  group_by("treatment", "dose.nM") %>%
  filter(any(libType == "chRNA")) %>%
  ungroup() %>%
  group_by(gid, treatment, dose.nM, cell.type, libType) %>%
  summarise_at(vars(UpstreamJunc:InclusionPSI), mean, na.rm=T) %>%
  mutate(TreatedOrControl = if_else(treatment=="DMSO", "Control", "Treatment")) %>%
  ungroup()


inner_join(
    chRNA.Comparison.PSI.Inclusion.tidy %>%
      filter(TreatedOrControl == "Treatment") %>%
      dplyr::select(-TreatedOrControl),
    chRNA.Comparison.PSI.Inclusion.tidy %>%
      filter(TreatedOrControl == "Control") %>%
      dplyr::select(-TreatedOrControl, -treatment, -dose.nM),
    by=c("gid", "cell.type", "libType"),
    suffix=c(".treatment", ".control")) %>%
  inner_join(
  sample.list %>%
    filter(cell.type=="LCL") %>%
    distinct(treatment, dose.nM, color)) %>%
ggplot(aes(x=InclusionPSI.treatment-InclusionPSI.control, color=color)) +
  stat_ecdf(aes(linetype=libType)) +
  scale_color_identity() +
  facet_wrap(~treatment) +
  theme_bw() +
  labs(title="Comparing DeltaPSI in drug-induced exons (treatment - DMSO)", y="ecdf",
       caption="Darker color is higher dose\nPSI_treatment-PSIcontrol")


inner_join(
    chRNA.Comparison.PSI.Inclusion.tidy %>%
      filter(TreatedOrControl == "Treatment") %>%
      dplyr::select(-TreatedOrControl),
    chRNA.Comparison.PSI.Inclusion.tidy %>%
      filter(TreatedOrControl == "Control") %>%
      dplyr::select(-TreatedOrControl, -treatment, -dose.nM),
    by=c("gid", "cell.type", "libType"),
    suffix=c(".treatment", ".control")) %>%
  inner_join(
  sample.list %>%
    filter(cell.type=="LCL") %>%
    distinct(treatment, dose.nM, color)) %>%
ggplot(aes(x=InclusionPSI.treatment/InclusionPSI.control, color=color)) +
  stat_ecdf(aes(linetype=libType)) +
  scale_color_identity() +
  scale_x_continuous(trans="log2") +
  facet_wrap(~treatment) +
  theme_bw() +
  labs(title="Comparing DeltaPSI in drug-induced exons (treatment / DMSO)", y="ecdf",
       caption="Darker color is higher dose\nPSI_treatment/PSIcontrol")
```

Ok, so while we can see these exons have a generally positive deltaPSI (as expected, by design of how i filtered the data), and the deltaPSI is more positive at higher dose, it doesn't seem that the deltaPSI is any bigger in chRNA. And how you choose to the process the data makes all the difference (Difference in PSI, versus Ratio).

```{r}
inner_join(
    chRNA.Comparison.PSI.Inclusion.tidy %>%
      filter(TreatedOrControl == "Treatment") %>%
      dplyr::select(-TreatedOrControl),
    chRNA.Comparison.PSI.Inclusion.tidy %>%
      filter(TreatedOrControl == "Control") %>%
      dplyr::select(-TreatedOrControl, -treatment, -dose.nM),
    by=c("gid", "cell.type", "libType"),
    suffix=c(".treatment", ".control")) %>%
  inner_join(
  sample.list %>%
    filter(cell.type=="LCL") %>%
    distinct(treatment, dose.nM, color)) %>%
ggplot(aes(x=(InclusionPSI.treatment/SkippingJunc.treatment)/(InclusionPSI.control/SkippingJunc.control), color=color)) +
  stat_ecdf(aes(linetype=libType)) +
  scale_color_identity() +
  scale_x_continuous(trans="log2") +
  facet_wrap(~treatment) +
  theme_bw() +
  labs(title="Comparing DeltaPSI in drug-induced exons (treatment / DMSO)", y="ecdf",
       caption="Darker color is higher dose\nPSI_treatment/PSIcontrol")
```


```{r}
AddPsuedocount <- function(x){
  return(x+0.01)
}

dat.to.plot <- inner_join(
    chRNA.Comparison.PSI.Inclusion.tidy %>%
      filter(TreatedOrControl == "Treatment") %>%
      dplyr::select(-TreatedOrControl),
    chRNA.Comparison.PSI.Inclusion.tidy %>%
      filter(TreatedOrControl == "Control") %>%
      dplyr::select(-TreatedOrControl, -treatment, -dose.nM),
    by=c("gid", "cell.type", "libType"),
    suffix=c(".treatment", ".control")) %>%
  inner_join(
  sample.list %>%
    filter(cell.type=="LCL") %>%
    distinct(treatment, dose.nM, color)) %>%
  #Add pseudocount to PSI
  mutate_at(vars(UpstreamJunc.treatment:InclusionPSI.control), AddPsuedocount) %>%
  mutate(SkippingToInclusionRatio = (InclusionPSI.treatment/SkippingJunc.treatment)/(InclusionPSI.control/SkippingJunc.control)) %>%
  dplyr::select(-(UpstreamJunc.treatment:InclusionPSI.control)) %>%
  pivot_wider(names_from="libType", values_from="SkippingToInclusionRatio")

ggplot(dat.to.plot, aes(x=(chRNA/polyA), color=color)) +
  geom_vline(xintercept=1) +
  stat_ecdf() +
  scale_color_identity() +
  scale_x_continuous(trans="log10") +
  facet_wrap(~treatment) +
  theme_bw() +
  labs(title="effect sizes don't appear stronger in chRNA", y="ecdf",
       x="DrugInducedSkippingRatio_chRNA/DrugInducedSkippingRatio_polyA",
       caption="DrugInducedSkippingRatio = (PSI_included_treatment/PSI_skip_treatment)/(PSI_included_control/PSI_skip_control)")
```

And to verify that I did the analysis reasonably, let's look at the HTT skipping event, where I know (from manually inspecting the screenshot at the top with the sashimi plots of the raw PSI) that chRNA should have a higher DrugInducedSkipping Ratio in branaplam

```{r}
GA.GT.Introns.DrugInducedCassetteExons %>%
  filter(gene_names == "HTT")

dat.to.plot %>%
  filter(gid=="chr4_clu_9322_+")
  
```

...as expected. Overall I don't think drug-induced exons are actually that more apparent in chRNA. Later I will peruse a large handful of sashimi plots to make verify. Now let's move on to looking at the natural splicing rate of introns that host the drug-induced exons...


### Host intron inclusion of drug-induced introns 

First let's look at the correlation matrix based on IntronExcisionRatio and SpliceEfficiency metrics (see [SpliceQ](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04282-6) paper) for the GA|GT introns that I previously fit dose-response curves to... Since 

```{r, fig.height=10, fig.width=10}

# GA.GT.Introns.DrugInducedCassetteExons

# chRNA.Comparison.PSI.Inclusion.tidy

HostIntronsToQuantifyInDrugInducedCassetteExons <- IntronsToQuantifyInDrugInducedCassetteExons %>%
  pivot_wider(names_from="IntronType", values_from="junc") %>%
  separate(SkippingJunc, into=c("Chrom", "start","stop","cluster"), sep=":", convert=T) %>%
  mutate(strand = str_extract(cluster, "[+-]"))

SpliceQ.IER %>%
  dplyr::select(1:4, matches("DMSO_NA_LCL_chRNA_.+")) %>%
  gather("Sample", "IER", contains("DMSO")) %>%
  group_by(Chrom, start, stop, strand) %>%
  summarise(IER.mean=mean(IER)) %>%
  inner_join(all.samples.intron.annotations, by=c("Chrom"="chrom", "start", "stop"="end", "strand")) %>%
  left_join(HostIntronsToQuantifyInDrugInducedCassetteExons) %>%
  group_by(gene_ids) %>%
  filter(any(!is.na(UpstreamJunc))) %>%
  ungroup() %>%
  ggplot(aes(x=IER.mean)) +
  stat_ecdf() +
  geom_vline(
    data = . %>% filter(!is.na(UpstreamJunc)),
    aes(xintercept=IER.mean)
  ) +
  facet_wrap(~gene_names) +
  theme_bw() +
  labs(y="ecdf", x="IntronExcisionRatio", title="IER distribution of host introns, compared to other introns in host gene")
  

```

Hmmm, It's not obvious if there is a trend... I was hypothesizing that introns hosting the cassette exon would have have lower intron excision ratios, indicating less efficient splicing. Let's plot this a few different ways:

```{r}
SpliceQ.IER %>%
  dplyr::select(1:4, matches("DMSO_NA_LCL_chRNA_.+")) %>%
  gather("Sample", "IER", contains("DMSO")) %>%
  group_by(Chrom, start, stop, strand) %>%
  summarise(IER.mean=mean(IER)) %>%
  inner_join(all.samples.intron.annotations, by=c("Chrom"="chrom", "start", "stop"="end", "strand")) %>%
  left_join(HostIntronsToQuantifyInDrugInducedCassetteExons) %>%
  group_by(gene_ids) %>%
  filter(any(!is.na(UpstreamJunc))) %>%
  ungroup() %>%
  mutate(IsHostIntronOfDrugInducedCassetteExon = if_else(is.na(UpstreamJunc), "Control Intron", "Host Intron")) %>%
  # group_by(gene_names, IsHostIntronOfDrugInducedCassetteExon) %>%
  # sample_n(1) %>%
  ggplot(aes(x=IER.mean, color=IsHostIntronOfDrugInducedCassetteExon)) +
  stat_ecdf() +
  theme_bw() +
  labs(y="ecdf", x="IntronExcisionRatio", caption=str_wrap("Comparing introns hosting the drug-induced cassette exons to all introns within the same set of genes", 50))
```

The intron excision ratio metric is basically the length-normalized intronic coverage, relative to the flanking exons... So if there is a cassette exon that is spliced in at an appreciable rate in DMSO samples, that might explain this. The splice efficiency metric, based only on exon-exon junction and exon-intron, intron-exon reads should be less sensitive to this. let's check that the effect is there too.

```{r}
SpliceQ.SE %>%
  dplyr::select(1:4, matches("DMSO_NA_LCL_chRNA_.+")) %>%
  gather("Sample", "IER", contains("DMSO")) %>%
  group_by(Chrom, start, stop, strand) %>%
  summarise(IER.mean=mean(IER)) %>%
  inner_join(all.samples.intron.annotations, by=c("Chrom"="chrom", "start", "stop"="end", "strand")) %>%
  left_join(HostIntronsToQuantifyInDrugInducedCassetteExons) %>%
  group_by(gene_ids) %>%
  filter(any(!is.na(UpstreamJunc))) %>%
  ungroup() %>%
  mutate(IsHostIntronOfDrugInducedCassetteExon = if_else(is.na(UpstreamJunc), "Control Intron", "Host Intron")) %>%
  # group_by(gene_names, IsHostIntronOfDrugInducedCassetteExon) %>%
  # sample_n(1) %>%
  ggplot(aes(x=IER.mean, color=IsHostIntronOfDrugInducedCassetteExon)) +
  stat_ecdf() +
  theme_bw() +
  labs(y="ecdf", x="SpicingEfficiency", caption=str_wrap("Comparing introns hosting the drug-induced cassette exons to all introns within the same set of genes", 50))


```



Hm, here there is some very modest effect, but I will get to doing a more proper test for significance later.


Finally I want to get to the question about whether that exon-definition effect was apparent, wherein the upstream intron has more efficient splicing... I think I'll do this later, as it requires me to quantify intron retention in unannotated introns, which isn't in the existing intron retention quantifications that I made using splice-q. For now I think I can get some intuitions by looking at a number of loci with pygenometracks...

### Plotting with pygenometracks

Let's plot all of the drug induced exons with high confidence dose-response effects in a way that might make clear what is happening in the chromatin-associated RNA...

```{r, eval=F}

dir.create("../code/scratch/20220926_Plot_chRNA")

NewBigwigSampleList <- bind_rows(
  sample.list %>%
    filter(cell.type=="LCL" & libType=="chRNA" & treatment == "DMSO") %>%
    mutate(Treatment.topair=c("Risdiplam", "Branaplam", "C2C5")) %>%
    expand(SampleName, Treatment.topair) %>%
    inner_join(sample.list) %>%
    mutate(treatment = paste(treatment,Treatment.topair, sep=".")) %>%
    mutate(NewGroup = paste(treatment,dose.nM, cell.type, libType, "u", sep="_")) %>%
    mutate(SampleName = paste(treatment,dose.nM, cell.type, libType, rep, sep="_")),
  sample.list %>%
    filter(!(cell.type=="LCL" & libType=="chRNA" & treatment=="DMSO")) %>%
    mutate(NewGroup = group)) %>%
  dplyr::select(SampleName, bigwig, group, NewGroup, strand)


groups <- read_tsv("../code/bigwigs/BigwigList.groups.tsv", col_names=c("GroupName", "Color", "PSI", "Supergroup")) %>%
  filter(str_detect(GroupName, "_u$"))

left_join(NewBigwigSampleList, groups, by=c("group"="GroupName")) %>%
  distinct(NewGroup, .keep_all=T) %>%
  dplyr::select(NewGroup, Color, PSI, Supergroup) %>%
  mutate(Supergroup = case_when(
    str_detect(NewGroup, "^DMSO\\.") ~ str_replace(NewGroup, "DMSO\\.(.+?)_.+$", "\\1_LCL_chRNA_u"),
    TRUE ~ Supergroup
  )) %>%
  separate(NewGroup, into=c("treatment", "dose.nM", "cell.type", "libType"), convert=T, remove=F, sep="_") %>%
  mutate(IsControl = treatment == "DMSO") %>%
  mutate(IsControl2 = str_detect(treatment, "DMSO")) %>%
  arrange(cell.type, libType, desc(IsControl), desc(IsControl2), treatment, dose.nM) %>%
  dplyr::select(NewGroup, Color, PSI, Supergroup) %>%
  write_tsv("../code/scratch/20220926_Plot_chRNA/.groups.tsv", col_names = F)

NewBigwigSampleList %>% dplyr::select(-group) %>%
write_tsv("../code/scratch/20220926_Plot_chRNA/.bwList.tsv", col_names = F)
  

RegionsToPlot <- all.samples.PSI %>%
  filter(gid %in% GA.GT.Introns$gid) %>%
  mutate(SpliceAcceptor = case_when(
    strand == "+" ~ paste(`#Chrom`, end, strand, sep="."),
    strand == "-" ~ paste(`#Chrom`, start-2, strand, sep=".")
  )) %>%
  select(1:6, SpliceAcceptor, contains("LCL")) %>%
  gather(key="Sample", value="PSI", contains("LCL")) %>%
  group_by(junc, gid, strand, SpliceAcceptor) %>%
  drop_na() %>%
  summarise(mean=mean(PSI, na.rm = F)) %>%
  group_by(SpliceAcceptor) %>%
  filter(mean == max(mean)) %>%
  ungroup() %>%
  dplyr::select(UpstreamJunc = junc, SpliceAcceptor) %>%
  inner_join(GA.GT.Introns.DrugInducedCassetteExons, by=c("SpliceAcceptor"="UpstreamSpliceAcceptor")) %>%
  filter(IntronType == "Alt 5'ss") %>%
  rename(DownstreamJunc = junc) %>%
  mutate(SkippingJunc = case_when(
    strand == "+" ~ str_replace(paste(UpstreamJunc, DownstreamJunc), "^(chr.+?):(.+?):(.+?):(clu.+?) (chr.+?):(.+?):(.+?):clu.+$", "\\1:\\2:\\7:\\4"),
    strand == "-" ~ str_replace(paste(UpstreamJunc, DownstreamJunc), "^(chr.+?):(.+?):(.+?):(clu.+?) (chr.+?):(.+?):(.+?):clu.+$", "\\1:\\6:\\3:\\4")
  )) %>%
  dplyr::select(UpstreamJunc,DownstreamJunc, SkippingJunc )




# Write out GA|GT splice sites, I will highlight these with a vertical bar
RegionsToPlot %>%
  separate(DownstreamJunc, into=c("chrom", "start", "end", "cluster"), convert=T, remove=F, sep=":") %>%
  mutate(strand = str_extract(cluster, "[+-]$")) %>%
  mutate(SpliceDonor = case_when(
    strand == "+" ~ start,
    strand == "-" ~ end
  )) %>%
  mutate(SpliceDonorEnd = SpliceDonor + 1) %>%
  select(`chrom`, SpliceDonor, SpliceDonorEnd) %>%
  write_tsv("../code/scratch/20220926_Plot_chRNA/.SpliceDonors.bed", col_names = F)

# write out pygenometracks ini file for GA|GT splice sites track
fileConn<-file("../code/scratch/20220926_Plot_chRNA/.SpliceDonors.ini")
writeLines(c("[vlines]","type = vlines", "file = scratch/20220926_Plot_chRNA/.SpliceDonors.bed"), con=fileConn)
close(fileConn)

# Write out sashimi filter juncs
RegionsToPlot %>%
  gather(key="type", value="junc") %>%
  separate(junc, into=c("chrom", "start", "end", "cluster"), convert=T, sep=":") %>%
  dplyr::select(chrom, start, end) %>%
  write_tsv("../code/scratch/20220926_Plot_chRNA/.SpliceJuncsToFilter.bed", col_names = F)



# Write out shell commands that use a helper script of mine for plotting the data nicely.
#--FilterJuncsByBed scratch/ToPlot.introns.bed
RegionsToPlot %>%
  left_join(
    GA.GT.Introns %>%
      dplyr::select(DownstreamJunc=junc, gene_names)) %>%
  drop_na() %>%
  mutate(gene_names=str_replace_all(gene_names, ",", "-")) %>%
  separate(SkippingJunc, into=c("chrom", "start", "end", "cluster"), convert=T, remove=F, sep=":") %>%
  mutate(min=start-400, max=end+400) %>%
  mutate(n=row_number()) %>%
  mutate(n.pad=str_pad(n,3,pad="0")) %>%
  mutate(cmd = str_glue("python scripts/GenometracksByGenotype/AggregateBigwigsForPlotting.py --GroupSettingsFile scratch/20220926_Plot_chRNA/.groups.tsv --BigwigList scratch/20220926_Plot_chRNA/.bwList.tsv --Normalization WholeGenome --Region {chrom}:{min}-{max} --BigwigListType KeyFile --FilterJuncsByBed scratch/20220926_Plot_chRNA/.SpliceJuncsToFilter.bed --OutputPrefix scratch/20220926_Plot_chRNA/. -vv --TracksTemplate scripts/GenometracksByGenotype/tracks_templates/GeneralPurposeColoredByGenotypeWithSupergroups.ini --Bed12GenesToIni scripts/GenometracksByGenotype/PremadeTracks/gencode.v26.FromGTEx.genes.bed12.gz\npyGenomeTracks --tracks <(cat scratch/20220926_Plot_chRNA/.tracks.ini scratch/20220926_Plot_chRNA/.SpliceDonors.ini) --out scratch/20220926_Plot_chRNA/{n.pad}_{gene_names}_{chrom}_{start}_{end}.png --region {chrom}:{min}-{max} --trackLabelFraction 0.15 --title {gene_names}_{chrom}_{start}\n\n")) %>%
  select(cmd) %>%
  write.table("../code/scratch/20220926_Plot_chRNA/MakePlots.sh", quote=F, row.names = F, col.names = F)
```

And run the script to plot...

```{bash, eval=F}
cd /project2/yangili1/bjf79/20211209_JingxinRNAseq/code/
bash scratch/20220926_Plot_chRNA/MakePlots.sh
```

