---
title: "2023-10-26_ExploreDrmModels"
output:
  pdf_document: default
  html_document: default
date: '2023-10-26'
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message=F)
```

## Load libs
```{r}
library(tidyverse)
library(RColorBrewer)
library(data.table)
library(drc)
library(ggrepel)

theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family="Helvetica")))
```

## Intro

There are many ways to fit models with `drc::drm`... See [this tutorial](https://rstats4ag.org/dose-response-curves.html)...

I have noticed some funny things, in that depending on exactly how you call the `drm` function, the lower and upper limits estimates are sometimes switched (with corresponding sign change in slope parameter), and sometimes you get wildly different estimates... I kind of know what I am looking for inuitively, and I think I have an idea of how exactly I want to call `drm` to fit dose response curves to gene expression. I'm going to demonstrate some of these nuances here... But first for some terminology.... the 4 parameter log logistic model has for parameters: b, c, d, and e...

- b is the "slope" parameter
- c is the lower limit parameter. This may correspond to the limit when dose=0, or the limit when dose=Infinity.
- d is the upper limit parameter. Again, this is just the "other" limit compared to c... The same model can be specificied by swapping c and d, and also changing the sign of b
- e is the ED50 parameter.

When I fit dose response curves to gene expression, I figured the following details would make the model most useful and easily interpretable:

Response is in units of log2FC relative to DMSO... So I will use the $log2CPM_d - mean(log2CPM_{DMSO})$ as the response metric (where $log2CPM_d$ denotes the log2 CountsPerMillion for a gene at dose $d$). Thus, one of the limit parameters should be fixed at 0. For best modelling, it makes sense to limit parameters when reasonable, so I will also fit each treatment simultaneously such that I can limit the other limit parameter to be the same amongst all 3 treatments. The ED50 and slope parameter can freely vary between treatments, as I want to capture examples like HTT manual/eyeball interpretation of the dose response data looks like the slope is similar between all three treatments, but the ED50 is shifted. And in the HSD17B4 example, the slope looks clearly different between treatments, and the underlying reason makes sense (in that at the splicing level, there are either 2  or 1 poison exons depending on the treatment, and the gene expression response is logically sort of like the complement of the product of the poison exon dose response curves). Also, for simplicity/interpretability, rather than use the actual nanomolar dose for each treatment, I will rescale the doses of branaplam and C2C5 to be in units that are roughly functionally equivalent to the nanomolar dose for risdiplam. For example, since C2C5 is about 10x more potent than risdiplam, I will use 10*nanomolarDose for C2C5. Also I know that the steepness parameter should be constrained... sometimes the fit converges to something really ridiculous with a unrealistically steep dose. Since b is scaled by the limits, it as actually reasonable imo to just set the same limits across all genes... something like $abs(c)<5$

## Analysis

Below, I will use some example data and show what happens when I try to fit these models different ways. In theory, a lot of these models should converge to the same thing, but for whatever reason they don't.



```{r}
f_in <-"../code/DoseResponseData/LCL/TidyExpressionDoseData_logFCTransformedAndAllDMSORepsInEachSeries.txt.gz"

sample_n_of <- function(data, size, ...) {
  dots <- quos(...)

  group_ids <- data %>%
    group_by(!!! dots) %>%
    group_indices()

  sampled_groups <- sample(unique(group_ids), size)

  data %>%
    filter(group_ids %in% sampled_groups)
}

expression.dat <- fread(f_in) %>%
  group_by(treatment) %>%
  mutate(doseRank = dense_rank(dose.nM)) %>%
  ungroup() %>%
  as_tibble() %>%
  mutate(doseInRisdiscale = case_when(
    treatment == "C2C5" ~ dose.nM * 10,
    treatment == "Branaplam" ~ dose.nM * sqrt(10),
    TRUE ~ dose.nM
  ))

expression.dat %>%
  distinct(doseInRisdiscale, treatment)
```

Plot dose response data for some genes of intrest

```{r}
GenesToHighlight <- c("STAT1", "HTT", "MYB",  "TRIM11", "TRAFD1", "VEGFA", "FBXW11", "HSD17B4")

expression.dat %>%
  mutate(treatment = factor(treatment)) %>%
  filter(hgnc_symbol %in% GenesToHighlight) %>%
  ggplot(aes(x=doseInRisdiscale, y=log2FC, color=treatment)) +
  geom_point() +
  geom_line() +
  scale_x_continuous(trans="log1p", breaks=c(10000, 1000,  100,  10,  0), labels=c("10K", "1K", "100", "10", "0")) +
  facet_wrap(~hgnc_symbol, scale="free_y")
```

Now plot some random genes...

```{r, fig.height=10, fig.width=12}
set.seed(0)
expression.dat %>%
  filter(!is.na(spearman) & !is.na(log2FC)) %>%
  add_count(Geneid) %>%
  filter(n==33) %>%
  sample_n_of(40, hgnc_symbol) %>%
  ggplot(aes(x=doseRank, y=log2FC, color=treatment)) +
  geom_point() +
  geom_line() +
  facet_wrap(~hgnc_symbol, scale="free_y")
```

Let's handpick a couple of those, to add to the gene list of interest to more carefully try fitting models...

```{r}
GenesToHighlight <- c(GenesToHighlight, c("BTG2", "B9D1", "ZIK1", "POLN", "TOMM20P4", "CRYBB1", "EIF2AK4"))
```


Now fit models for each of these in three ways, which should all in effect be identical:

## Without contraining slope

```{r, results='asis'}
SlopeAbsValueLimit <- 4.5
control <- drmc(maxIt=5000)

for (gene in GenesToHighlight){
  cat("\n## ", gene, "\n")
  data <- expression.dat %>%
    mutate(treatment = factor(treatment)) %>%
    filter(hgnc_symbol == gene)
  
  P <- ggplot(data, aes(x=doseInRisdiscale, y=log2FC, color=treatment)) +
  geom_point() +
  geom_line() +
  scale_x_continuous(trans="log1p", breaks=c(10000, 1000,  100,  10,  0), labels=c("10K", "1K", "100", "10", "0"))
  print(P)
  cat("\n")
  
  cat("\n### LL.4, fix upper limit\n\n")
  tryCatch(expr={
    fit <- drm(formula = log2FC ~ doseInRisdiscale,
                 data = data,
                 fct = LL.4(names=c("Steepness", "LowerLimit", "UpperLimit", "ED50"), fixed=c(NA,NA,0,NA)),
                 curveid = treatment,
                 pmodels=data.frame(treatment, 1, treatment, treatment),
                 control = control,
                 # lowerl = c(-5, -Inf,  -Inf),
                 # upperl = c(5, Inf,  Inf),
                 robust = "mean")
    message("\n\nSuccessfully fitted model:\n\n")
    plot(fit)
    as.data.frame(fit$coefficients) %>% knitr::kable() %>% print()
  },
  error=function(e){
      cat("ERROR:", gene, " model LL.4 fixed upper limit\n", conditionMessage(e), "\n")
    }
  )

  cat("\n### LL.4, fix lower limit\n")
  tryCatch(expr={
    fit <- drm(formula = log2FC ~ doseInRisdiscale,
                 data = data,
                 fct = LL.4(names=c("Steepness", "LowerLimit", "UpperLimit", "ED50"), fixed=c(NA,0,NA,NA)),
                 curveid = treatment,
                 pmodels=data.frame(treatment, 1, treatment, treatment),# Dunno why this is parameterized as is... but this seems to work
                 # lowerl = c(-5, -Inf,  -Inf),
                 # upperl = c(5, Inf,  Inf),
                 robust = "mean")
    message("\n\nSuccessfully fitted model:\n\n")
    plot(fit)
    as.data.frame(fit$coefficients) %>% knitr::kable() %>% print()
  },
  error=function(e){
      cat("ERROR:", gene, " model LL.4 fixed lower limit\n", conditionMessage(e), "\n")
    }
  )

  cat("\n### LL.3\n")
  tryCatch(expr={
    fit <- drm(formula = log2FC ~ doseInRisdiscale,
                 data = data,
                 fct = LL.3(names=c("Steepness", "UpperLimit", "ED50"), fixed=c(NA,NA,NA)),
                 curveid = treatment,
                 pmodels=data.frame(treatment, 1, treatment),
                 # lowerl = c(-5, -Inf,  -Inf),
                 # upperl = c(5, Inf,  Inf),
                 robust = "mean")
    message("\n\nSuccessfully fitted model:\n\n")
    plot(fit)
    as.data.frame(fit$coefficients) %>% knitr::kable() %>% print()
  },
  error=function(e){
      cat("ERROR:", gene, " model LL.3\n", conditionMessage(e), "\n")
    }
  )
  
    cat("\n### LL.4, fix upper limit, limit, limit slope\n\n")
  tryCatch(expr={
    fit <- drm(formula = log2FC ~ doseInRisdiscale,
                 data = data,
                 fct = LL.4(names=c("Steepness", "LowerLimit", "UpperLimit", "ED50"), fixed=c(NA,NA,0,NA)),
                 curveid = treatment,
                 pmodels=data.frame(treatment, 1, treatment, treatment),
                 lowerl = c(-SlopeAbsValueLimit, -Inf,  -Inf),
                 upperl = c(SlopeAbsValueLimit, Inf,  Inf),
                 robust = "mean")
    message("\n\nSuccessfully fitted model:\n\n")
    plot(fit)
    as.data.frame(fit$coefficients) %>% knitr::kable() %>% print()
  },
  error=function(e){
      cat("ERROR:", gene, " model LL.4 fixed upper limit, limit slope\n", conditionMessage(e), "\n")
    }
  )

  cat("\n### LL.4, fix lower limit, limit slope\n")
  tryCatch(expr={
    fit <- drm(formula = log2FC ~ doseInRisdiscale,
                 data = data,
                 fct = LL.4(names=c("Steepness", "LowerLimit", "UpperLimit", "ED50"), fixed=c(NA,0,NA,NA)),
                 curveid = treatment,
                 pmodels=data.frame(treatment, 1, treatment, treatment),# Dunno why this is parameterized as is... but this seems to work
                 lowerl = c(-SlopeAbsValueLimit, -Inf,  -Inf),
                 upperl = c(SlopeAbsValueLimit, Inf,  Inf),
                 robust = "mean")
    message("\n\nSuccessfully fitted model:\n\n")
    plot(fit)
    as.data.frame(fit$coefficients) %>% knitr::kable() %>% print()
  },
  error=function(e){
      cat("ERROR:", gene, " model LL.4 fixed lower limit, limit slope\n", conditionMessage(e), "\n")
    }
  )

  cat("\n### LL.3, limit slope\n")
  tryCatch(expr={
    fit <- drm(formula = log2FC ~ doseInRisdiscale,
                 data = data,
                 fct = LL.3(names=c("Steepness", "UpperLimit", "ED50"), fixed=c(NA,NA,NA)),
                 curveid = treatment,
                 pmodels=data.frame(treatment, 1, treatment),
                 lowerl = c(-SlopeAbsValueLimit, -Inf,  -Inf),
                 upperl = c(SlopeAbsValueLimit, Inf,  Inf),
                 robust = "mean")
    message("\n\nSuccessfully fitted model:\n\n")
    plot(fit)
    as.data.frame(fit$coefficients) %>% knitr::kable() %>% print()
  },
  error=function(e){
      cat("ERROR:", gene, " model LL.3, limit slope\n", conditionMessage(e), "\n")
    }
  )
}
```

