---
title: "Check introns with unique specificities"
output: html_document
date: '2023-03-06'
---

```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)

library(tidyverse)


```

### Recheck list of risdi/brana-specific introns

I previously made a list of risdiplam/branaplam/C2C5-specific introns based on the dose response experiment. Because the different molecules have different genome-wide effective potencies (ie, 100nM C2C5 is different than 100nM risdiplam, despite them having generally similar specificity profiles), I tested for a difference in the EC50 of each intron for each traitment pair, relative to the genome-wide median. You can see that in a [previous notebook](20220913_TidyDataForJingxin.html) I plotted the dose-response datapoints for a random sample of 20 of these significant molecule-specific effects, and they are generally believable effects from those.

```{r}
GA.GT.Introns <- read_tsv("../output/EC50Estimtes.FromPSI.txt.gz")

ColorKey <- c("Branaplam-specific"="#005A32", "Risdiplam-specific"="#084594", "Non significant"="#969696")

P1.dat <- GA.GT.Introns %>%
  mutate(Color = case_when(
    EC.Ratio.Test.Estimate.FDR_Branaplam.Risdiplam > 0.05 ~ "Non significant",
    ECRatio.ComparedToGenomewideMedian_Branaplam.Risdiplam > 1 ~ "Risdiplam-specific",
    ECRatio.ComparedToGenomewideMedian_Branaplam.Risdiplam < 1 ~ "Branaplam-specific"
  ))

ggplot(P1.dat, aes(x=ED50_Branaplam, y=ED50_Risdiplam, color=Color)) +
  geom_point() +
  scale_x_continuous(trans='log10', limits = c(1, 1E7)) +
  scale_y_continuous(trans='log10', limits = c(1, 1E7)) +
  scale_color_manual(values=ColorKey) +
  labs(x= "Branaplam\nED50 (nanomolar)", y= "Risdiplam\nED50 (nanomolar)", color=NULL) +
  theme(legend.position='bottom') +
  theme_bw()


P1.dat %>%
  count(Color)

```

Ok, so each point is a branaplam-specific or risdiplam-specific GA-GT intron. There are 200 non-significant points, 161 risdiplam-specific points, and 240 branaplam-specific points. I think it will be useful for Jingxin to be able to easily visualize these before validating them... Some introns that change PSI from 1% to 3% might be very significant, but not the best for validation. Furthermore, not all of these events might be involved in poison exon inclusion. Some of them might just be alt 5'ss or something. And these details might inform which are the best candidates for validation by flanking PCR primers. Also, when looking at the data, some of the 'risdiplam-specific' introns I called are just not believable effects. Like the dose-response data points might just be too noisy. I did a reasonable job of filtering these out, and trying to make the FDR estimate reasonably calibrated... Or like there are some examples where it appears 'risdiplam-specific' but it is because the cryptic event is only just barely visible at the highest dose of risdiplam, and not visible in any of the branaplam doses (so it just requires a really really high dose to see the effect). So for all these reasons, it will be best to check the raw coverage data first... So I'll write the out this list as a bedfile for easy viewing in IGV, a save a session that has the bedfile loaded, along with bigwigs for the dose response experiment so hopefully it will be easy to view.

```{r, eval=F}
P1.dat %>%
  dplyr::select(chrom=`#Chrom`, start, end, junc, score=ECRatio.ComparedToGenomewideMedian_Branaplam.Risdiplam, strand=strand.y, Color) %>%
  filter(!is.na(score)) %>%
  mutate(thickStart = start, thickEnd=end, Color=recode(Color, !!!ColorKey)) %>%
  mutate(RgbCol = apply(col2rgb(Color), 2, paste, collapse=',')) %>%
  dplyr::select(chrom:strand, thickStart, thickEnd, RgbCol) %>%
  arrange(chrom, start, end) %>%
  write_tsv("../code/Branaplam_Risdiplam_specific_introns.bed", col_names = F)
```

```{bash, eval=F}
bgzip ../code/Branaplam_Risdiplam_specific_introns.bed
tabix -p bed ../code/Branaplam_Risdiplam_specific_introns.bed.gz
```

So to view, you will need to use the globus link I shared with Jingxin to download the `code` directory, or at minimum download the `code/bigwigs` folder of bigwig files, the `code/tracks.xml` IGV session, and the `code/Branaplam_Risdiplam_specific_introns.bed.gz` and `code/Branaplam_Risdiplam_specific_introns.bed.gz.tbi`. Then, from IGV, go to File >> OpenSession and open the tracks.xml. The tracks.xml uses relative filepaths so you need to preserve the relative filepaths of the thing you downloaded (that is, they need to all be in the same folder, such as a folder named `code`).

Now you should be able to select the `Branaplam_Risdiplam_specific_introns.bed` track, and Ctl-F or Ctl-B to jump to the next GA-GT introns, colored according to significance. Browse around to find coordinates of potential splice events for validation. Note that each bigwig is already normalized signal relative to library depth. With autoscaling for the y-axis turned on (the default), it may be visually obvious to see the poison exons, but it might not be totally obvious to see if there are effects on gene expression. To better visualize effects on host gene expression, I recommend just selecting all bigwig tracks and right-click to "group autoscale". Since coverage of each track is already normalized by library size, it should be easy to visually interpret if there is a change in gene expression.

Here are screenshots of a few examples...

![BranaSpecific1](assets/BranaSpecificExample1.png)

![BranaSpecific2](assets/BranaSpecificExample2.png)

![RisdiSpecific1](assets/RisdiSpecificExample1.png)


In general, I'm noticing there a quite a few believable branaplam-specific examples but the risdlam-specific are for the most part very small specificity effect sizes. That is, there are very few obviously risdiplam-specific events, compared to branaplam-specific events.

Let's list the top risdi-specific introns in the following table:

```{r}
P1.dat %>%
  arrange(desc(ECRatio.ComparedToGenomewideMedian_Branaplam.Risdiplam)) %>%
  head() %>%
  knitr::kable()
```

After looking at some of the top risdiplam-specific hits from the list above, here is perhaps the most clear example:

![RisdiSpecific2](assets/RisdiSpecificExample2.png)

## Update, 20230308

Yang suggested I do some more exploring/filtering of the data a bit more to see if there are any introns and interesting host genes with reasonable differences in specificity between the risdiplam and branaplam treatments. So like HTT should show up here for example, but I wonder what other interesting genes may be on the list. I want to get a more useful/filtered list for Yang and Jingxin to browse


```{r}
ColorKey <- c("Branaplam-specific"="#005A32", "Risdiplam-specific"="#084594", "Non significant"="#969696")


P.histogram <- P1.dat %>%
  mutate(Color = factor(Color, levels=names(ColorKey))) %>%
  ggplot(aes(x=log2(ECRatio.ComparedToGenomewideMedian_Branaplam.Risdiplam), fill=Color)) +
  geom_histogram() +
  scale_fill_manual(values=ColorKey) +
  theme(legend.position="bottom") +
  labs(x="Specificity\nlog2(BranaplamEC50/RisdiplamEC50)", caption="EC50s normalized to median before comparing ratios")

P.histogram
```

Let's just consider stronger specificity effects, with abs(log2 ratio) > 2.

Note that I haven't filtered these GA-GT intron affects for those with effects on the host gene. So like some of these hits may not effect host gene expression.


```{r}
P.histogram +
  xlim(c(-10, 10)) +
  geom_vline(xintercept=c(-2,2))

SpecificEffects <- P1.dat %>%
  mutate(log2Ratio = log2(ECRatio.ComparedToGenomewideMedian_Branaplam.Risdiplam)) %>%
  filter(abs(log2Ratio)>2) %>%
  filter(!Color == "Non significant") %>%
  separate_rows(gene_names, gene_ids, sep=',') %>%
  dplyr::select(Color, log2Ratio, gene_names, gene_ids) %>%
  # filter(gene_names == "HTT") %>%
  arrange(log2Ratio)

head(SpecificEffects)

```

Let's also merge this with some gene expression data. For convenience, I'm just going to merge it with the spearman of the dose:response effect on expression, and also gather the FoldChange at the mid-point dose, which I think is a reasonable place to start to see some effects (the high dose in this experiment might be a bit ridiculous from the pharmacology point of view).

```{r}
gene_names <- read_tsv("../data/Genes.list.txt")

GeneExpressionData.tidy <- read_tsv("../code/DoseResponseData/LCL/TidyExpressionDoseData.txt.gz") %>%
  mutate(CPM = log2(CPM)) %>%
  mutate(ensembl_gene_id = str_replace(Geneid, "^(.+?)\\..*$", "\\1")) %>%
  left_join(gene_names)


 
```

As a data parsing side note... I can quickly get the doses for the midpoint doses by filtering on the color column. Let me remember how I coded the colors to get the appropriate doses

```{r}
GeneExpressionData.tidy %>%
  distinct(color, .keep_all=T) %>%
  ggplot(aes(x=1, y=color, fill=color)) +
  geom_raster() +
  geom_label(aes(label=color)) +
  scale_fill_identity() +
  facet_wrap(~treatment)
```

Ok, so in other words, I want to filter for color = #969696 (DMSO), or #74C476 (middle branaplam dose), or #6BAED6 (middle risdiplam dose).

```{r}

MoleculeSpecificEffects <- GeneExpressionData.tidy %>%
  filter(color %in% c("#74C476", "#6BAED6")) %>%
  filter(Geneid %in% SpecificEffects$gene_ids) %>%
  dplyr::select(Geneid, CPM, SampleName, treatment, spearman) %>%
  left_join(
    GeneExpressionData.tidy %>%
      filter(color %in% c("#969696")) %>%
      filter(Geneid %in% SpecificEffects$gene_ids) %>%
      dplyr::select(Geneid, CPM) %>%
      group_by(Geneid) %>%
      summarise(CPM_DMSO = mean(CPM))
  ) %>%
  mutate(Log2FoldChange = CPM - CPM_DMSO) %>%
  pivot_wider(names_from = c("treatment"), values_from=c("CPM", "spearman", "SampleName", "Log2FoldChange")) %>%
  inner_join(SpecificEffects, by=c("Geneid"="gene_ids")) %>%
  dplyr::rename(log2RatioOfSplicingED50= log2Ratio) %>%
  arrange(log2RatioOfSplicingED50)

MoleculeSpecificEffects %>%
  filter(gene_names == "HTT") %>%
  knitr::kable()

```


Ok that makes sense. Note that the CPM columns are actually log2CPM. We can see that HTT has a branaplam-specific effect.



Let's print the full table...

So for this table I only considered ones where there is a significant difference in splicing ED50 between risdiplam and branaplam. Technically, I renormalized EC50 to the genomewide median ED50 for GA-GT introns since risdiplam and branaplam molar concentrations aren't directly comparable even for STAT1. Then, I filtered for the stronger effects (with log2 ratio) magnitude greater than 2 (see histogram above with the 2 vertical lines). Then, in the table, I am also listing the effects on expression in branaplam and risdiplam... Because some of the GA-GT introns that get upregulated don't actually effect host gene expression, like if they don't create an NMD poison exon. Note that all the CPM values in the table are actually log2CPM. 

Then, in the table there is also a column called "spearman_Branaplam" or "spearman_Risdiplam" which is the expression dose:response speraman correlation across the whole titration series... so in the example of HTT which is on the table (and also above), you can see that there is a similarly strong negative spearman correlation coefficient in the risdiplam and branaplam, but the log2RatioOfSplicingED50 column suggests it is quite branaplam specific, and if you look at the CPM_Branaplam (100nM branaplam) column compared to the CPM_DMSO and CPM_Risdiplam (316nM risdiplam) columns, you can see that there really is only a strong expression effect in branaplam at that midpoint dose... The midpoint dose is roughly comparable to some of the relatively low doses that Jingxin used in the recent experiment of 52 molecules

```{r}
MoleculeSpecificEffects %>%
  knitr::kable()
```


Yang noted that we didn't see FOXM1 or LRKK2 on this list. Let's investigate why.

```{r}
SpecificEffects %>%
  filter(gene_names %in% c("FOXM1", "LRRK2"))

GA.GT.Introns %>%
  separate_rows(gene_names, gene_ids, sep=',') %>%
  filter(gene_names %in% c("FOXM1", "LRRK2", "HTT", "STAT1"))

GA.GT.Introns.OfInterest <- 
  GA.GT.Introns %>%
  separate_rows(gene_names, gene_ids, sep=',') %>%
  filter(gene_names %in% c("FOXM1", "LRRK2", "HTT", "STAT1")) %>%
  pull(junc)


GeneExpressionData.tidy %>%
  filter(hgnc_symbol %in% c("FOXM1", "LRRK2", "HTT", "STAT1")) %>%
  ggplot(aes(color=treatment)) +
  geom_line(aes(x=dose.nM, y=log2(CPM))) +
  scale_x_continuous(trans="log1p", limits=c(0, 10000), breaks=c(10000, 3160, 1000, 316, 100, 31.6, 10, 3.16, 1, 0.316, 0)) +
  facet_wrap(~hgnc_symbol, scales = "free_y") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size=3)) +
  labs(title="Dose response effects on gene expression genes of interest")

```

...oops,  with the x-axis in nanomolar units its kind of annoying to interpret. Let's just plot the x-axis dose rank, instead of nanomolar units, since each of the titration curves cover similar effective dosages.

```{r}

GeneExpressionData.tidy %>%
  filter(hgnc_symbol %in% c("FOXM1", "LRRK2", "HTT", "STAT1")) %>%
  group_by(treatment) %>%
  mutate(DoseRank = dense_rank(dose.nM)) %>%
  ungroup() %>%
  ggplot(aes(color=treatment)) +
  geom_line(aes(x=DoseRank, y=log2(CPM))) +
  facet_wrap(~hgnc_symbol, scales = "free_y") +
  theme_bw() +
  labs(title="Dose response effects on gene expression genes of interest")

SplicingData.tidy <- read_tsv("../code/DoseResponseData/LCL/TidySplicingDoseData.txt.gz")

SplicingData.tidy %>%
  filter(junc %in% GA.GT.Introns.OfInterest) %>%
  group_by(treatment) %>%
  mutate(DoseRank = dense_rank(dose.nM)) %>%
  ungroup() %>%
  mutate(label = paste(gene_names, junc, sep='\n')) %>%
  ggplot(aes(color=treatment)) +
  geom_line(aes(x=DoseRank, y=PSI)) +
  facet_wrap(~label, scales = "free_y") +
  theme_bw() +
  theme(strip.text.x = element_text(size = 5)) +
  labs(title="All GAGT introns in genes of interest", y="Intronic PSI")
  

```

Ok, so I can see the dose response gene expression effects on those genes. And also here are the dose response splicing effects for all the detected GAGT introns (and passed default leafcutter cluster filters)  in those genes. The intron in FOXM1 is not in the large table because it didn't pass my filters for specific splicing effects, and that seems understandable looking at the splicing data. The LRRK2 intron isn't in the long table because I didn't even attempt fitting a dose-response curve to the splicing data because I filtered for introns with spearman > 0.9 in at least one treatment. Perhaps that is a bit too strict to capture these introns that only start to show effects at the high doses.
